---
title: "CDK-mediated phosphoryations of IDRs"
author: "J.M. Valverde & G. Dubra"
output:
  github_document: 
    toc: TRUE
---

This repository provides the source code and data for all the analyses performed in the publication "A Cyclin dependent kinase-mediated phosphorylation switch of disordered protein condensation" (Valverde, Dubra et al., 2023). This current GitHub readme file was automaticaly generated, using R Studio, from a R markdown (README.Rmd) where all the code used in the paper is recompiled.

```{r setup and data loading,echo=FALSE,message=FALSE,warning=FALSE}

# Libraries required
library(jsonlite)
library(readr)
library(ggplot2)
library(GGally)
library(vcd)
library(kableExtra)
library(ggpubr)
# library(Rcpi)
#if fails => sudo R CMD javareconf; reinstall; reboot R
library(Biostrings)
library(eulerr)
library(flextable)
library(stringr)
library(DT)
library(dplyr)
library(tidyr)
library(RColorBrewer)
library(reshape2)
library(AnnotationDbi)
library(ggridges)
library(hrbrthemes)
library(ggrepel)
library(ggsci)
library(scales)
library(data.table)
library(frapplot)
library(viridis)
library(wesanderson)
library(pheatmap)
library(gplots)
library(ggplot2)
library(ComplexHeatmap)
library(useful)
library(missForest)
library(ggbreak)
library(heatmap3)
library(tibble)
library(forcats)
library(ggseqlogo)
library(Biostrings)
library(gghighlight)
library(Hmisc)
library(ggstance)



if (file.exists("~/CDK_mediated_phosphorylation_of_IDRs_WS.RData")) {
  load("~/CDK_mediated_phosphorylation_of_IDRs_WS.RData")
} else {
  




# load the function for obtaining disorder metrics from diferent predictors
source("auxiliary_functions/getDisorder_V2.R")
# load function to generate contingency tables from phosphorylation data
source("auxiliary_functions/contingencyTablesAnalysis_V2.R")
# Load the function to generate protein disorder plots from IUpred predictions
source("auxiliary_functions/IUpredScoresPlotGenerator.R")
# Load the functions to analyze sequence motifs
source("auxiliary_functions/motifs_functions.R")


#################################
######## Disorder data ##########
#################################


# ___________Disorder_____________

# YEAST

yeastDiso<-stream_in(con = file("disorder_data/MobiDB/yeast_mobiDB_full.mjson"),verbose = F)
yeastDiso$mobidb_data <- yeastDiso$mobidb_data$disorder
colnames(yeastDiso)[2] <- "disorder"

yeast_uniprot_table <- read_delim("uniprot_data/yeast_UniprotID_reviewed.tab", "\t", escape_double = FALSE, trim_ws = TRUE)

yeastDiso <- merge.data.frame(yeastDiso,yeast_uniprot_table,by.x = "acc",by.y = "Entry")

yeastDiso_full <- getMobiDB_PredDiso(yeastDiso,"acc",sequence_col = "Sequence")
## addd SPOT
yeastDiso_full <- getSpotPredDiso(yeastDiso_full,disoPath = "disorder_data/local_predictions/spot_run_yeast/",accession_col = "acc",length_col = "Length",sequence_col = "Sequence")

# HUMAN

humanDiso<-stream_in(con = file("disorder_data/MobiDB/HS_mobiDB_full.mjson"),verbose = F)
humanDiso$mobidb_data <- humanDiso$mobidb_data$disorder
colnames(humanDiso)[2] <- "disorder"

human_uniprot_table <- read_delim("uniprot_data/HS_UniprotID_reviewed.tab", "\t", escape_double = FALSE, trim_ws = TRUE)

humanDiso <- merge.data.frame(humanDiso,human_uniprot_table,by.x = "acc",by.y = "Entry")
# the entry number 3972 in mobidb is broken. I remove it now
humanDiso <- humanDiso[-3971,]
humanDiso_full <- getMobiDB_PredDiso(humanDiso,"acc",sequence_col = "Sequence")
humanDiso_full <- getSpotPredDiso(humanDiso_full,disoPath = "disorder_data/local_predictions/spot_run_human/",accession_col = "acc",length_col = "Length",sequence_col = "Sequence")


# Load Xenopus non-redundant sequence database used for proteomic approaches 
xenopus_MPI<-readAAStringSet("MPI_xenopus_database/Xenopus_Database_MPI.FASTA",format = "fasta")
xenopusDiso<-as.data.frame(xenopus_MPI)
colnames(xenopusDiso)<-"Sequence"
xenopusDiso$ID <- str_extract(names(xenopus_MPI), regex( "(?<=\\|)(.*)"))
row.names(xenopusDiso) <- NULL
xenopusDiso$Length <-nchar(xenopusDiso$Sequence)
# The ID mapping between Uniprot and MPI databases can be complex. That rules out the utilization of MobiDB as a disorder information source. For xenopus we calculate manually the values of disorder for its entire proteome using 3 predictors 
xenopusDiso <- getIUpredPredDiso(df = xenopusDiso,disoPath = "disorder_data/local_predictions/IUpred_run_xenopus/",sequence_col = "Sequence",accession_col = "ID",length_col = "Length")
xenopusDiso <- getVSLPredDiso(df = xenopusDiso,disoPath = "disorder_data/local_predictions/VSL2b_run_xenopus//",sequence_col = "Sequence",accession_col = "ID",length_col = "Length")
xenopusDiso <- getSpotPredDiso(df = xenopusDiso,disoPath = "disorder_data/local_predictions/spot_run_xenopus/",sequence_col = "Sequence",accession_col = "ID",length_col = "Length")


###### CDK1 targets data ########
#################################

# Yeast
## ___________Ubersax data_____________
ubersax_data <- read_delim("phosphorylation_data/Ubersax_CDK1targetSCereviseae.csv", 
                          "\t", quote = "\\\"", escape_double = FALSE, 
                          col_types = cols(X1 = col_skip(), input = col_skip(), 
                          `yourlist:M201907046746803381A1F0E0DB47453E0216320D4B0BD1N` = col_skip()), 
                          trim_ws = TRUE)

# Now the comumn "disorder" is a unnamed list with a complex structure for each row of the data frame

#The row 294 cannot be mapped with the uniprot, I will remove it from CDK1targetsSacCer and yeastDisoCDK1target. Slect only the rows wuth P-score > 2 most likely targets based on the paper

ubersax_data<-ubersax_data[-294,]
# ubersax_data <- merge.data.frame(ubersax_data,yeastDiso_full,by.x = "Entry",by.y = "acc")
# After the merging 7 uncharacterized protein are left out the dataset. n=680

# SPOT
# Not for now
# ubersax_data$dbDisoPerc <- yeastDiso_ubersax_subset$dbDisoPerc
# ubersax_data$predSpotDisoPerc <- yeastDiso_ubersax_subset$predSpotDisoPerc

# ubersax_data$target<-cut(ubersax_data$`P-Score`,breaks = c(2),labels = c("Non Cdk1 target","Cdk1 target"))
ubersax_data$target<-cut(ubersax_data$`P-Score`,breaks = c(-Inf,2,Inf),labels = c("Non Cdk1 target","Cdk1 target"),right = F)
ubersax_data$target <- relevel(factor(ubersax_data$target),"Cdk1 target")
ubersax_data$targetStrict<-cut(ubersax_data$`P-Score`,breaks = c(-Inf,3,Inf),labels = c("Non Cdk1 target","Cdk1 target"),right = F)
ubersax_data$targetStrict <- relevel(factor(ubersax_data$targetStrict),"Cdk1 target")

## ___________Holt Data_____________
uniprot_yeast <- read_delim("uniprot_data/uniprot_yeast.tab",
"\t", escape_double = FALSE, trim_ws = TRUE)
holt_MS <- read_delim("phosphorylation_data/Holt_CDK1target_SPombe.tsv",
"\t", escape_double = FALSE, trim_ws = TRUE)

#Remove phosphoTyrosines
holt_MS<-subset(holt_MS,!substr(site,start = 1,stop = 1)=="Y")
#Add Uniprot accesion. 58 entries lost (out of 9067) due to ID conversion
holt_MS <- holt_MS[which(holt_MS$ORF %in% uniprot_yeast$OLN),]

holt_MS$Uniprot <- uniprot_yeast$`Swiss-Prot-AC`[base::match(holt_MS$ORF,uniprot_yeast$OLN)]

# Create a new column indicating which sites are considered to be phosphorylated 
holt_MS <- holt_MS %>% mutate(psite=case_when(
    `log2H/L_asynchronous`<(-1)|`log2H/L_nocodazole`<(-1)|`log2H/L_clb2`<(-1) ~ " phosphorylated",
    TRUE ~ " non phosphorylated"
))
holt_MS$psite <- factor(holt_MS$psite)
holt_MS$psite <- relevel(holt_MS$psite,ref = " phosphorylated")

# Create a new column indicating which sites are considered to be phosphorylated in an CDK consensus motif 
holt_MS <- holt_MS %>% mutate(consensus_psite=case_when(
    (`log2H/L_asynchronous`<(-1)|`log2H/L_nocodazole`<(-1)|`log2H/L_clb2`<(-1)) & grepl("[st]P",centered_15mer) ~ " phosphorylated" ,
    TRUE ~ " non phosphorylated"
))
holt_MS$consensus_psite <- factor(holt_MS$consensus_psite)
holt_MS$consensus_psite <- relevel(holt_MS$consensus_psite,ref = " phosphorylated")

# The summarise function substrings the sites for removing the first letter, indicating the residue. Since I have to apply the same function for all the columns to be summarized, I added an space at the beginning of the psite and psites values and like this I don't loose the first letter
# the columns site, psite, and psite consensus are merged in a string separated by commas and grouped by uniprot ID
holt_data <- holt_MS %>% group_by(Uniprot) %>% summarise_at(c("site","psite","consensus_psite"),function(x){paste(substring(x,2,1000), collapse=",")})

# Convert the psites and consensus_psite columns in target_all and target_consensus columns respectively. If one of the psites is phosphorylated by CDK thats consider a CDK target

holt_data <- holt_data %>% mutate(target_all=case_when(
    grepl(psite,pattern = "(,|^)phosphorylated") ~ "Cdk1 target",
    TRUE ~ "Non Cdk1 target"
))

holt_data <- holt_data %>% mutate(target_consensus=case_when(
    grepl(consensus_psite,pattern = "(,|^)phosphorylated") ~ "Cdk1 target",
    TRUE ~ "Non Cdk1 target"
))

# Tibble doesn't like list and vectors as values, it's easier to reconvert to DF
holt_data <- as.data.frame(holt_data)
# Select which psites are cdk psites with and withouth consensus
## site is a vector now
holt_data$site <- lapply(holt_data$site, function(x){as.numeric(strsplit(x,',')[[1]])})
## psite and consensus_psite are logical vectors with TRUE="phosphorylated" and FALSE="non phosphorylated"
holt_data$psite <- lapply(holt_data$psite, function(x){strsplit(x,',')[[1]]=="phosphorylated"})
holt_data$consensus_psite <- lapply(holt_data$consensus_psite, function(x){strsplit(x,',')[[1]]=="phosphorylated"})

psites <- list()
consensus_psites <- list()
for (i in 1:nrow(holt_data)) {
  psites[[i]] <- holt_data$site[[i]][holt_data$psite[[i]]]
  consensus_psites[[i]] <- holt_data$site[[i]][holt_data$consensus_psite[[i]]]
}
holt_data$psite <- psites
holt_data$consensus_psite <- consensus_psites
rm(psites,consensus_psites)

## ___________Intersection yeast_____________

# The universe is the union between Holt and Ubersax
yeastIntersect_All_data <- merge.data.frame(ubersax_data[,c("Entry","target","targetStrict")],
                                            holt_data[,c("Uniprot","site","psite","consensus_psite","target_all","target_consensus")],
                                            by.x = "Entry", by.y = "Uniprot",all = T)

 yeastIntersect_All_data <- yeastIntersect_All_data %>% mutate(target_intersect=case_when(
    ((target=="Cdk1 target") & (target_all=="Cdk1 target")) ~ "Cdk1 target",
    TRUE ~ "Non Cdk1 target"
)) %>% as.data.frame()

#####Add disorder to yeast DATA
# Ubersax
ubersax_data <- merge.data.frame(ubersax_data,yeastDiso_full,by.x = "Entry",by.y = "acc")
ubersax_data$MW_kDa<-(ubersax_data$Length*120)/1000
ubersax_data$nmol<- (ubersax_data$`Protein Level`/ubersax_data$MW_kDa)/1000
ubersax_data$nEq_STP<- ubersax_data$nmol*ubersax_data$`[ST]-P`
ubersax_data$nEq_STPXKR<- ubersax_data$nmol*ubersax_data$`[ST]-P-X-[KR]`

# 4 Entries are lost in the conversion, 2 of them target_all="CDK1 target"
# Holt
holt_data<-merge.data.frame(holt_data,yeastDiso_full,by.x = "Uniprot",by.y = "acc")
# adding number of consensus
holtMinCons <- sapply(holt_data$Sequence,function(x){index<-unlist(gregexpr("[ST]P",x),use.names = F)
                                                      if (index[1]==-1) {return(0)
                                                      } else {return(length(index))}})
names(holtMinCons) <- NULL 
holtFullCons <- sapply(holt_data$Sequence,function(x){index<-unlist(gregexpr("[ST]P.[KR]",x),use.names = F)
                                                      if (index[1]==-1) {return(0)
                                                      } else {return(length(index))}})
names(holtFullCons) <- NULL 

holt_data$`[ST]-P-X-[KR]`<-holtFullCons
holt_data$`[ST]-P`<-holtMinCons

# Intersection
## add disorder MobiDB
yeastIntersect_All_data <- merge.data.frame(yeastIntersect_All_data,yeastDiso_full,by.x = "Entry",by.y = "acc")

yeastIntersect_CDK1_data <- subset(yeastIntersect_All_data, (target_all == "Cdk1 target") & (target=="Cdk1 target"))

# Human
## ___________Phosphosite Human Data_____________

human_data <- read_delim("phosphorylation_data/phosphositePlus_data/human_data_curated_V2_cleaned.tab", 
    "\t", escape_double = FALSE, col_types = cols(MOD_RSD = col_character()), 
    trim_ws = TRUE)

human_data$psite_CDK1 <- lapply(human_data$MOD_RSD, function(x){ return(as.numeric(strsplit(x,",")[[1]]))})
human_data$target <- rep("Cdk1 target",nrow(human_data))
human_data$target <- as.factor(human_data$target)

# human_data <- human_data[,c("GENE","PROTEIN","ACC#","MOD_RSD","target","psite_CDK1","KINASE")]

## ___________Phosphosite Human Universe_____________
# Load all the phosphosites in the phosphosite plus database 
human_universe_data <- read_delim("phosphorylation_data/phosphositePlus_data/Phosphorylation_site_dataset", 
    "\t", escape_double = FALSE, col_types = cols(HU_CHR_LOC = col_skip(), 
        SITE_GRP_ID = col_skip(), MW_kD = col_skip(), 
        DOMAIN = col_skip(), `SITE_+/-7_AA` = col_skip(), 
        LT_LIT = col_skip(), MS_LIT = col_skip(), 
        MS_CST = col_skip(), `CST_CAT#` = col_skip()), 
    trim_ws = TRUE)
human_universe_data<-reshape::rename(human_universe_data,c(ACC_ID="ACC#"))
# Select human proteins
human_universe_data <- subset(human_universe_data, ORGANISM == "human")
# Remove all information related to isoforms
human_universe_data <- subset(human_universe_data, !grepl("-",`ACC#`))
human_universe_data <- subset(human_universe_data, !grepl(" iso[0-9]",`PROTEIN`))
# Select targets with pS or pT
human_universe_data<-subset(human_universe_data,(substr(MOD_RSD,1,1)=="S"|substr(MOD_RSD,1,1)=="T"))
# format the MOD_RSD column and group by gene/protein/uniprot
human_universe_data <- human_universe_data %>% mutate(MOD_RSD=substr(MOD_RSD,1,nchar(MOD_RSD)-2)) 
human_universe_data <- human_universe_data %>% group_by(`ACC#`,GENE,PROTEIN) %>% summarise_at("MOD_RSD",function(x){paste(substr(x,2,2000), collapse=",")})
# Generate the psite column, with the vectors containing psites (for the contingency table analysis)
human_universe_data <- as.data.frame(human_universe_data)
human_universe_data$psite <- lapply(human_universe_data$MOD_RSD, function(x){ return(as.numeric(strsplit(x,",")[[1]]))})

# merge the tables and mark CDK1 targets and non CDK1 targets. Only by ACC, protein names in human data are still with  the isoform nomenclature
human_data <- merge.data.frame(x = human_data,y = human_universe_data,by = c("ACC#"),all = T,suffixes = c("_CDK1","_ALL"))
#remove PROTEIN and GENE comumns from x, and rename the y columns
human_data$GENE_CDK1<-NULL
human_data$PROTEIN_CDK1<-NULL
human_data <- reshape::rename(human_data,c(PROTEIN_ALL="PROTEIN",GENE_ALL="GENE"))
# Adding the target category "Non Cdk1 target"
levels(human_data$target) <- c("Cdk1 target","Non Cdk1 target")
human_data$target[is.na(human_data$target)]<-"Non Cdk1 target"

human_data<-merge.data.frame(human_data,humanDiso_full,by.x = "ACC#",by.y = "acc")


#__________________________________________
# addition of human MAPK,AURK,PLK,NEK,DYRK

mapk_data <- read_delim("phosphorylation_data/phosphositePlus_data/MAPK/PSP_MAPK_Target_HS.tab", 
    "\t", escape_double = FALSE, col_types = cols(MAPK_MOD_RSD = col_character()), 
    trim_ws = TRUE)

mapk_data$psite_MAPK <- lapply(mapk_data$MAPK_MOD_RSD, function(x){ return(as.numeric(strsplit(x,",")[[1]]))})

human_data<-merge.data.frame(human_data, mapk_data, by = c("GENE","PROTEIN","ACC#"),all.x = T)

human_data <- human_data %>% mutate(target_mapk=case_when(
    !is.na(MAPK_MOD_RSD) ~ "mapk target",
    TRUE ~ "Non mapk target"
))

aurk_data <- read_delim("phosphorylation_data/phosphositePlus_data/AURK/PSP-Gerber_data_curated_cleaned.tab", 
    "\t", escape_double = FALSE, col_types = cols(AURK_MOD_RSD = col_character()), 
    trim_ws = TRUE)

aurk_data$psite_aurk <- lapply(aurk_data$AURK_MOD_RSD, function(x){ return(as.numeric(strsplit(x,",")[[1]]))})

human_data<-merge.data.frame(human_data, aurk_data, by = c("ACC#"),all.x = T)

human_data <- human_data %>% mutate(target_aurk=case_when(
    !is.na(AURK_MOD_RSD) ~ "aurk target",
    TRUE ~ "Non aurk target"
))

plk_data <- read_delim("phosphorylation_data/phosphositePlus_data/PLK//PSP-Gerber_data_curated_cleaned.tab", 
    "\t", escape_double = FALSE, col_types = cols(PLK_MOD_RSD = col_character()), 
    trim_ws = TRUE)

plk_data$psite_plk <- lapply(plk_data$PLK_MOD_RSD, function(x){ return(as.numeric(strsplit(x,",")[[1]]))})

human_data<-merge.data.frame(human_data, plk_data, by = c("ACC#"),all.x = T)

human_data <- human_data %>% mutate(target_plk=case_when(
    !is.na(PLK_MOD_RSD) ~ "plk target",
    TRUE ~ "Non plk target"
))

nek_data <- read_delim("phosphorylation_data/phosphositePlus_data/NEK/PSP_NEK_Target_HS.tab", 
    "\t", escape_double = FALSE, col_types = cols(NEK_MOD_RSD = col_character()), 
    trim_ws = TRUE)

nek_data$psite_nek <- lapply(nek_data$NEK_MOD_RSD, function(x){ return(as.numeric(strsplit(x,",")[[1]]))})

human_data<-merge.data.frame(human_data, nek_data, by = c("ACC#"),all.x = T)

human_data <- human_data %>% mutate(target_nek=case_when(
    !is.na(NEK_MOD_RSD) ~ "nek target",
    TRUE ~ "Non nek target"
))

dyrk_data <- read_delim("phosphorylation_data/phosphositePlus_data/DYRK/PSP_DYRK_Target_HS.tab", 
    "\t", escape_double = FALSE, col_types = cols(DYRK_MOD_RSD = col_character()), 
    trim_ws = TRUE)

dyrk_data$psite_dyrk <- lapply(dyrk_data$DYRK_MOD_RSD, function(x){ return(as.numeric(strsplit(x,",")[[1]]))})

human_data<-merge.data.frame(human_data, dyrk_data, by = c("ACC#"),all.x = T)

human_data <- human_data %>% mutate(target_dyrk=case_when(
    !is.na(DYRK_MOD_RSD) ~ "dyrk target",
    TRUE ~ "Non dyrk target"
))
#__________________________________________




#Xenopus
## ___________Phosphosite Xenopus Data_____________

xenopus_data <- read_delim("phosphorylation_data/Valverde_data/all_phosphosites_xenopus_nr.tab", "\t", escape_double = FALSE, col_types = cols(`Leading proteins` = col_skip(), Protein = col_skip()), trim_ws = TRUE)

xenopus_data <- xenopus_data %>% dplyr::rename(ID=Proteins,psites=`Positions within proteins`,seqWindow=`Sequence window`,UID=`Unique identifier`) %>% group_by(ID) %>% summarise_at(c("psites","seqWindow","UID"),function(x){paste(x, collapse=",")})

xenopus_ANOVA_data <- read_delim("phosphorylation_data/Valverde_data/Xen_phospho_AnovaPhosphosites_curated.txt","\t", escape_double = FALSE, trim_ws = TRUE)

xenopus_data <- merge.data.frame(xenopus_data,xenopusDiso,by = "ID")

clusterA <- read_lines(file = "phosphorylation_data/Valverde_data/ids/xenopusclusterA_mpi.txt")
clusterB <- read_lines(file = "phosphorylation_data/Valverde_data/ids/xenopusclusterB_mpi.txt")
clusterC <- read_lines(file = "phosphorylation_data/Valverde_data/ids/xenopusclusterC_mpi.txt")
clusterD <- read_lines(file = "phosphorylation_data/Valverde_data/ids/xenopusclusterD_mpi.txt")
humanCDKtargets <- read_lines(file = "phosphorylation_data/Valverde_data/ids/humanCDKtargets_mpi.txt")

xenopus_data <- xenopus_data %>% mutate(cluster_A=case_when(
    ID %in% clusterA ~ TRUE,
    TRUE ~ FALSE
))
xenopus_data <- xenopus_data %>% mutate(cluster_B=case_when(
    ID %in% clusterB ~ TRUE,
    TRUE ~ FALSE
))
xenopus_data <- xenopus_data %>% mutate(cluster_C=case_when(
    ID %in% clusterC ~ TRUE,
    TRUE ~ FALSE
))
xenopus_data <- xenopus_data %>% mutate(cluster_D=case_when(
    ID %in% clusterD ~ TRUE,
    TRUE ~ FALSE
))

xenopus_data <- xenopus_data %>% mutate(ANOVA=case_when(
    ID %in% clusterD ~ TRUE,
    TRUE ~ FALSE
))

xenopus_data <- xenopus_data %>% mutate(ANOVA=case_when(
    (cluster_A | cluster_B | cluster_C | cluster_D) ~ "Dynamic",
    TRUE ~ "Non Dynamic"
))

xenopus_data <- xenopus_data %>% mutate(hCDKtarget=case_when(
    ID %in% humanCDKtargets ~ TRUE,
    TRUE ~ FALSE
))

xenopus_data$psites <- lapply(xenopus_data$psites, function(x){ return(as.numeric(strsplit(x,",")[[1]]))})
xenopus_data$UID <- lapply(xenopus_data$UID, function(x){ return(as.character(strsplit(x,",")[[1]]))})
xenopus_data$anova_psites <- apply(xenopus_data, 1, function(x){x$psites[x$UID %in% xenopus_ANOVA_data$UID]})


save.image("~/CDK_mediated_phosphorylation_of_IDRs_WS.RData")}
```

# Main figures

## Fig. 1

### c

```{r embryo anova heatmap,echo=FALSE,message=FALSE,warning=FALSE,fig.height=7,fig.width=6}

#Load data 

pSites_Median_Norm_Full_Matrix <- read.csv('phosphorylation_data/Valverde_data/All_pSites_ANOVA_Statistics.csv', stringsAsFactors = F)


#Creating unique useful identifier

pSites_Median_Norm_Full_Matrix_Unique_ID <- unite(pSites_Median_Norm_Full_Matrix, 
                                                  'Protein_Amino.acid_Site_NumberPhospho', 
                                                  c('Proteins', 'Amino.acid', 'Positions.within.proteins', 'Multiplicity'),
                                                  sep ='|', remove = FALSE)

#filtering for ANOVA +

pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig <- 
  pSites_Median_Norm_Full_Matrix_Unique_ID[pSites_Median_Norm_Full_Matrix_Unique_ID$ANOVA.Significant == '+', ] 

#Df with only mean intensities

pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean <- 
  pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig[, c(86:103)]


#Preparing data
#Transforming to matrix

pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix <-
  as.matrix(sapply(pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean, as.numeric))

#Changing column names

colnames(pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix) <- c('0h00', '0h15','0h30','0h45','1h00','1h15','1h25',
                                                                              '1h45','1h50','1h55','2h05','2h20','2h30','2h50',
                                                                              '3h00','3h10','3h20','3h35')

#Changing rownames

rownames(pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix) <- pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig[, 68]

library(circlize)
col_fun = colorRamp2(c(-3, 0, 3), c("cyan", "black", "yellow"))

#Plotting with Complex Heatmap
#Scaling is needed, not done automatically by Complex HeatMap

pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix_Scaled <-
  t(scale(t(pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix)))


#Heatmap using k-means for clustering

HM <- Heatmap(pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix_Scaled, 
             column_order = colnames(pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix),
             show_row_names = FALSE, row_km = 4, row_km_repeats = 100, clustering_distance_rows = 'spearman', 
             clustering_method_rows = 'complete', name = 'Z-Score', col = col_fun,
             cluster_row_slices = TRUE, border = FALSE, row_dend_width = unit(3, "cm"),
             show_parent_dend_line = FALSE, column_names_rot = 45)

HM

#Extracting row clusters
#From https://github.com/jokergoo/ComplexHeatmap/issues/136

r.dend <- row_dend(HM)  #Extract row dendrogram
rcl.list <- row_order(HM)  #Extract clusters (output is a list)

# loop to extract genes for each cluster.
for (i in 1:length(row_order(HM))){
  if (i == 1) {
    clu <- t(t(row.names(pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix_Scaled[row_order(HM)[[i]],])))
    out <- cbind(clu, paste("cluster", i, sep=""))
    colnames(out) <- c("ID", "Cluster")
  } else {
    clu <- t(t(row.names(pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix_Scaled[row_order(HM)[[i]],])))
    clu <- cbind(clu, paste("cluster", i, sep=""))
    out <- rbind(out, clu)
  }
}


#Matching cluster to identifier in scaled matrix
#transforming matrices to data frames in order to use merge

pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix_Scaled_df <- as.data.frame(
  pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix_Scaled)

out_df <- as.data.frame(out)

#rownames to column to use merge

pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix_Scaled_df_ready <- 
  pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix_Scaled_df %>% rownames_to_column('ID')

#merging

pSites_with_clusters_fromHM_Scaled <- merge(x = pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig_Mean_Matrix_Scaled_df_ready,
                                            y = out_df, by.x = 'ID', by.y = 'ID')

#writing table to use scaled intensities in single line plots of phosphoproteome

#Filtering based on cluster to get data frame for each cluster

Cluster1 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster1', ]

Cluster2 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster2', ]

Cluster3 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster3', ]

Cluster4 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster4', ]

#transforming to matrix

Cluster1_matrix <- data.matrix(Cluster1)
rownames(Cluster1_matrix) <- Cluster1[, 1]
Cluster1_matrix <- Cluster1_matrix[, -c(1, 20)]


Cluster2_matrix <- data.matrix(Cluster2)
rownames(Cluster2_matrix) <- Cluster2[, 1]
Cluster2_matrix <- Cluster2_matrix[, -c(1, 20)]

Cluster3_matrix <- data.matrix(Cluster3)
rownames(Cluster3_matrix) <- Cluster3[, 1]
Cluster3_matrix <- Cluster3_matrix[, -c(1, 20)]

Cluster4_matrix <- data.matrix(Cluster4)
rownames(Cluster4_matrix) <- Cluster4_matrix[, 1]
Cluster4_matrix <- Cluster4_matrix[, -c(1, 20)]

#Matching cluster to identifier in full matrix, in order to transform data to visualize with ggplot

pSites_with_clusters_fromHM <- merge(x = pSites_Median_Norm_Full_Matrix_Unique_ID_ANOVA_sig, 
                                     y = out, by.x = 'Protein_Amino.acid_Site_NumberPhospho', by.y = 'ID')

#Tidying up the data
#Leaving only per replicate normalized intensities with valuable information for plotting
#ANOVA
pSites_with_clusters_fromHM_tidy <- pSites_with_clusters_fromHM[, c(1:53, 69, 279:281, 285)]

#All
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy <- pSites_Median_Norm_Full_Matrix_Unique_ID[, c(1:52, 68, 69, 74, 279:281)]

#Changing colnames
#ANOVA

pSites_with_clusters_fromHM_tidy_v1 <- plyr::rename(pSites_with_clusters_fromHM_tidy, 
                                                    c('X00A' = '0h00_RepA', 'X00B' = '0h00_RepB', 'X00C' = '0h00_RepC',
                                                      'X01A' = '0h15_RepA', 'X01B' = '0h15_RepB',
                                                      'X02A' = '0h30_RepA', 'X02B' = '0h30_RepB', 'X02C' = '0h30_RepC',
                                                      'X03B' = '0h45_RepA', 'X03C' = '0h45_RepC',
                                                      'X04A' = '1h00_RepA', 'X04B' = '1h00_RepB', 'X04C' = '1h00_RepC',
                                                      'X04.5A' = '1h15_RepA', 'X04.5B' = '1h15_RepB', 'X04.5C' = '1h15_RepC',
                                                      'X05A' = '1h25_RepA', 'X05B' = '1h25_RepB', 'X05C' = '1h25_RepC',
                                                      'X06A' = '1h45_RepA', 'X06B' = '1h45_RepB', 'X06C' = '1h45_RepC',
                                                      'X07A'= '1h50_RepA', 'X07B' = '1h50_RepB', 'X07C' = '1h50_RepC',
                                                      'X08A' = '1h55_RepA', 'X08B' = '1h55_RepB', 'X08C' = '1h55_RepC',
                                                      'X09A' = '2h05_RepA', 'X09B' = '2h05_RepB', 'X09C' = '2h05_RepC',
                                                      'X10A' = '2h20_RepA', 'X10B' = '2h20_RepB', 'X10C' = '2h20_RepC',
                                                      'X11A' = '2h30_RepA', 'X11B' = '2h30_RepB', 'X11C' = '2h30_RepC',
                                                      'X12A' = '2h50_RepA', 'X12B' = '2h50_RepB', 'X12C' = '2h50_RepC',
                                                      'X13A' = '3h00_RepA', 'X13B' = '3h00_RepB', 'X13C' = '3h00_RepC',
                                                      'X14A' = '3h10_RepA', 'X14B' = '3h10_RepB', 'X14C' = '3h10_RepC',
                                                      'X15A' = '3h20_RepA', 'X15B' = '3h20_RepB', 'X15C' = '3h20_RepC',
                                                      'X16A' = '3h35_RepA', 'X16B' = '3h35_RepB', 'X16C' = '3h35_RepC',
                                                      'Protein_Amino.acid_Site_NumberPhospho' = 'ID'))

#All

pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1 <- plyr::rename(pSites_Median_Norm_Full_Matrix_Unique_ID_tidy, 
                                                                 c('X00A' = '0h00_RepA', 'X00B' = '0h00_RepB', 'X00C' = '0h00_RepC',
                                                                   'X01A' = '0h15_RepA', 'X01B' = '0h15_RepB',
                                                                   'X02A' = '0h30_RepA', 'X02B' = '0h30_RepB', 'X02C' = '0h30_RepC',
                                                                   'X03B' = '0h45_RepA', 'X03C' = '0h45_RepC',
                                                                   'X04A' = '1h00_RepA', 'X04B' = '1h00_RepB', 'X04C' = '1h00_RepC',
                                                                   'X04.5A' = '1h15_RepA', 'X04.5B' = '1h15_RepB', 'X04.5C' = '1h15_RepC',
                                                                   'X05A' = '1h25_RepA', 'X05B' = '1h25_RepB', 'X05C' = '1h25_RepC',
                                                                   'X06A' = '1h45_RepA', 'X06B' = '1h45_RepB', 'X06C' = '1h45_RepC',
                                                                   'X07A'= '1h50_RepA', 'X07B' = '1h50_RepB', 'X07C' = '1h50_RepC',
                                                                   'X08A' = '1h55_RepA', 'X08B' = '1h55_RepB', 'X08C' = '1h55_RepC',
                                                                   'X09A' = '2h05_RepA', 'X09B' = '2h05_RepB', 'X09C' = '2h05_RepC',
                                                                   'X10A' = '2h20_RepA', 'X10B' = '2h20_RepB', 'X10C' = '2h20_RepC',
                                                                   'X11A' = '2h30_RepA', 'X11B' = '2h30_RepB', 'X11C' = '2h30_RepC',
                                                                   'X12A' = '2h50_RepA', 'X12B' = '2h50_RepB', 'X12C' = '2h50_RepC',
                                                                   'X13A' = '3h00_RepA', 'X13B' = '3h00_RepB', 'X13C' = '3h00_RepC',
                                                                   'X14A' = '3h10_RepA', 'X14B' = '3h10_RepB', 'X14C' = '3h10_RepC',
                                                                   'X15A' = '3h20_RepA', 'X15B' = '3h20_RepB', 'X15C' = '3h20_RepC',
                                                                   'X16A' = '3h35_RepA', 'X16B' = '3h35_RepB', 'X16C' = '3h35_RepC',
                                                                   'Protein_Amino.acid_Site_NumberPhospho' = 'ID'))


#changing several columns to factors

#ANOVA
pSites_with_clusters_fromHM_tidy_v1$Amino.acid <- as.factor(pSites_with_clusters_fromHM_tidy_v1[, 'Amino.acid'])
pSites_with_clusters_fromHM_tidy_v1$Leading.proteins <- as.factor(pSites_with_clusters_fromHM_tidy_v1[, 'Leading.proteins'])
pSites_with_clusters_fromHM_tidy_v1$Protein <- as.factor(pSites_with_clusters_fromHM_tidy_v1[, 'Protein'])
pSites_with_clusters_fromHM_tidy_v1$Cluster <- as.factor(pSites_with_clusters_fromHM_tidy_v1[, 'Cluster'])

#All
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1$Amino.acid <- as.factor(pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1[, 'Amino.acid'])
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1$Leading.proteins <- as.factor(pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1[, 'Leading.proteins'])
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1$Protein <- as.factor(pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1[, 'Protein'])
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1$ANOVA.Significant <- as.factor(pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1[, 'ANOVA.Significant'])

#From wide format to long format

#ANOVA
pSites_with_clusters_fromHM_tidy_v2 <- melt(pSites_with_clusters_fromHM_tidy_v1)

#All
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v2 <- melt(pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v1)

#Separate Time and replicate

#ANOVA
pSites_with_clusters_fromHM_tidy_v3 <- pSites_with_clusters_fromHM_tidy_v2 %>%
  separate(variable, c('Time', 'Replicate'), '_')

#All
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v3 <- pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v2 %>%
  separate(variable, c('Time', 'Replicate'), '_')

#Separate Time into hours and minutes

#ANOVA
pSites_with_clusters_fromHM_tidy_v4 <- pSites_with_clusters_fromHM_tidy_v3 %>%
  separate(Time, c('Hours', 'Minutes'), 'h')

#All
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4 <- pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v3 %>%
  separate(Time, c('Hours', 'Minutes'), 'h')

#Converting hours and minutes to numeric

#ANOVA
pSites_with_clusters_fromHM_tidy_v4$Hours <- as.numeric(pSites_with_clusters_fromHM_tidy_v4$Hours)
pSites_with_clusters_fromHM_tidy_v4$Minutes <- as.numeric(pSites_with_clusters_fromHM_tidy_v4$Minutes)

#All
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4$Hours <- as.numeric(pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4$Hours)
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4$Minutes <- as.numeric(pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4$Minutes)

#multiplying hours time 60 to get minutes

#ANOVA
pSites_with_clusters_fromHM_tidy_v4$HoursX60 <- pSites_with_clusters_fromHM_tidy_v4$Hours*60

#All
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4$HoursX60 <- pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4$Hours*60

#Adding all minutes to get final time

#ANOVA
pSites_with_clusters_fromHM_tidy_v4$Time <- pSites_with_clusters_fromHM_tidy_v4$HoursX6 + 
  pSites_with_clusters_fromHM_tidy_v4$Minutes

#All
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4$Time <- pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4$HoursX60 + 
  pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4$Minutes

#Tidying up the data frame: removing unwanted columns

#ANOVA
pSites_with_clusters_fromHM_tidy_v5 <- subset(pSites_with_clusters_fromHM_tidy_v4, select = -c(HoursX60, Hours, Minutes))

#All
pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v5 <- subset(pSites_Median_Norm_Full_Matrix_Unique_ID_tidy_v4, select = -c(HoursX60, Hours, Minutes))
#checking properties of the data frame






```


### e

```{r extracts anova heatmap,echo=FALSE,message=FALSE,warning=FALSE,fig.height=7,fig.width=6}

pSites_ANOVASig_Zscored <- read.csv('phosphorylation_data/Valverde_data/extracts/Phosphosites_XL_Extract_ANOVA0.05_MedianNorm_Avgs_Zsocred_CorrectGroups.csv', stringsAsFactors = F)


#Creating unique useful identifier

pSites_ANOVASig_Zscored_Unique_ID <- unite(pSites_ANOVASig_Zscored, 
                                           'Protein_Amino.acid_Site_NumberPhospho', 
                                           c('Protein', 'Amino.acid', 'Position', 'Multiplicity'),
                                           sep ='|', remove = FALSE)


#Df with only intensities of time point of interest
#Individual (per bio replicate)

pSites_ANOVASig_Zscored_Unique_ID_BioRep <-
  pSites_ANOVASig_Zscored_Unique_ID[, c(1:21)]

#Means

pSites_ANOVASig_Zscored_Unique_ID_Means <-
  pSites_ANOVASig_Zscored_Unique_ID[, c(21, 47:53)]

#Medians

pSites_ANOVASig_Zscored_Unique_ID_Medians <-
  pSites_ANOVASig_Zscored_Unique_ID[, c(21, 55:61)]

#Preparing data
#reordering columns

col_order_BioRep <- c('Interphase_000_A', 'Interphase_000_B', 'Interphase_000_C', 'Interphase_030_A', 'Interphase_030_B_R', 
                      'Interphase_030_C', 'Interphase_060_A', 'Interphase_060_B', 'Interphase_060_C', 'Interphase_090_A', 
                      'Interphase_090_B', 'Interphase_090_C', 'Interphase_120_A_R', 'Interphase_120_B_R', 'Interphase_120_C', 
                      'Mitosis_CyB_A', 'Mitosis_CyB_C_R', 'Mitosis_CyB_D', 'Mitosis_CSF', 'Mitosis_CSF_TR_R','Protein_Amino.acid_Site_NumberPhospho')
col_order_Means <- c('Mean.Interphase_000', "Mean.Interphase_030", 'Mean.Interphase_060', 'Mean.Interphase_090',
                     'Mean.Interphase_120', 'Mean.Mitosis_CyB','Mean.Mitosis_CSF',
                     'Protein_Amino.acid_Site_NumberPhospho')
col_order_Medians <- c('Median.Interphase_000', "Median.Interphase_030", 'Median.Interphase_060', 'Median.Interphase_090',
                       'Median.Interphase_120','Median.Mitosis_CyB', 'Median.Mitosis_CSF',
                       'Protein_Amino.acid_Site_NumberPhospho')

pSites_ANOVASig_Zscored_Unique_ID_BioRep_Ord <- pSites_ANOVASig_Zscored_Unique_ID_BioRep[, col_order_BioRep]

pSites_ANOVASig_Zscored_Unique_ID_Means_Ord <- pSites_ANOVASig_Zscored_Unique_ID_Means[, col_order_Means]

pSites_ANOVASig_Zscored_Unique_ID_Medians_Ord <- pSites_ANOVASig_Zscored_Unique_ID_Medians[, col_order_Medians]

#Transforming to numeric matrix

pSites_ANOVASig_Zscored_Unique_ID_BioRep_Ord_Matrix <-
  as.matrix(sapply(pSites_ANOVASig_Zscored_Unique_ID_BioRep_Ord, as.numeric))

pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix <-
  as.matrix(sapply(pSites_ANOVASig_Zscored_Unique_ID_Means_Ord, as.numeric))

pSites_ANOVASig_Zscored_Unique_ID_Medians_Ord_Matrix <-
  as.matrix(sapply(pSites_ANOVASig_Zscored_Unique_ID_Medians_Ord, as.numeric))

#Adding rownames

rownames(pSites_ANOVASig_Zscored_Unique_ID_BioRep_Ord_Matrix) <- pSites_ANOVASig_Zscored_Unique_ID_BioRep_Ord[, 21]
rownames(pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix) <- pSites_ANOVASig_Zscored_Unique_ID_Means_Ord[, 8]
rownames(pSites_ANOVASig_Zscored_Unique_ID_Medians_Ord_Matrix) <- pSites_ANOVASig_Zscored_Unique_ID_Medians_Ord[, 8]

#removing useless column

pSites_ANOVASig_Zscored_Unique_ID_BioRep_Ord_Matrix_ready <- pSites_ANOVASig_Zscored_Unique_ID_BioRep_Ord_Matrix[, -c(21)]
pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready <- pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix[, -c(8)]
pSites_ANOVASig_Zscored_Unique_ID_Medians_Ord_Matrix_ready <- pSites_ANOVASig_Zscored_Unique_ID_Medians_Ord_Matrix[, -c(8)]

#Plotting with Complex Heatmap

col_fun = colorRamp2(c(-2.5, 0, 2.5), c("cyan", "black", "yellow"))

set.seed(5346)
HM_extracts = Heatmap(pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready, 
             column_order = colnames(pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready),
             show_row_names = FALSE, row_km = 6, row_km_repeats = 100, clustering_distance_rows = 'spearman', 
             clustering_method_rows = 'average', name = 'Z-Score', col = col_fun, row_dend_width = unit(3, 'cm'))


#Extracting row clusters
#From https://github.com/jokergoo/ComplexHeatmap/issues/136

HM_extracts = draw(HM_extracts)           #assign Heat Map to object and draw it
r.dend <- row_dend(HM_extracts)  #Extract row dendrogram
rcl.list <- row_order(HM_extracts)  #Extract clusters (output is a list)


# loop to extract genes for each cluster.
for (i in 1:length(row_order(HM_extracts))){
  if (i == 1) {
    clu <- t(t(row.names(pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready[row_order(HM_extracts)[[i]],])))
    out <- cbind(clu, paste("cluster", i, sep=""))
    colnames(out) <- c("ID", "Cluster")
  } else {
    clu <- t(t(row.names(pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready[row_order(HM_extracts)[[i]],])))
    clu <- cbind(clu, paste("cluster", i, sep=""))
    out <- rbind(out, clu)
  }
}


#Matching cluster to identifier in scaled matrix, in order to plot heatmap for different clusters
#number of clusters corresponds to the number from top to bottom of the heatmap (logical order) not the onw shown on the heatmap
#transforming matrices to data frames in order to use merge

pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready_df <-
  as.data.frame(pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready)
out_df <- as.data.frame(out)


#rownames to column to use merge

pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready_df_NoRowNames <- 
  pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready_df %>% rownames_to_column('ID')

#merging

pSites_with_clusters_fromHM_Scaled <- merge(x = pSites_ANOVASig_Zscored_Unique_ID_Means_Ord_Matrix_ready_df_NoRowNames,
                                            y = out_df, by.x = 'ID', by.y = 'ID')


#Filtering based on cluster to get data frame for each cluster

Cluster1 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster1', ]

Cluster2 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster2', ]

Cluster3 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster3', ]

Cluster4 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster4', ]

Cluster5 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster5', ]

Cluster6 <- pSites_with_clusters_fromHM_Scaled[pSites_with_clusters_fromHM_Scaled$Cluster == 'cluster6', ]

#transforming to matrix

Cluster1_matrix <- data.matrix(Cluster1)
rownames(Cluster1_matrix) <- Cluster1[, 1]
Cluster1_matrix <- Cluster1_matrix[, -c(1, 9)]

Cluster2_matrix <- data.matrix(Cluster2)
rownames(Cluster2_matrix) <- Cluster2[, 1]
Cluster2_matrix <- Cluster2_matrix[, -c(1, 9)]

Cluster3_matrix <- data.matrix(Cluster3)
rownames(Cluster3_matrix) <- Cluster3[, 1]
Cluster3_matrix <- Cluster3_matrix[, -c(1, 9)]

Cluster4_matrix <- data.matrix(Cluster4)
rownames(Cluster4_matrix) <- Cluster4_matrix[, 1]
Cluster4_matrix <- Cluster4_matrix[, -c(1, 9)]

Cluster5_matrix <- data.matrix(Cluster5)
rownames(Cluster5_matrix) <- Cluster5_matrix[, 1]
Cluster5_matrix <- Cluster5_matrix[, -c(1, 9)]

Cluster6_matrix <- data.matrix(Cluster6)
rownames(Cluster6_matrix) <- Cluster6_matrix[, 1]
Cluster6_matrix <- Cluster6_matrix[, -c(1, 9)]



#Loading data before Z-scoring in order to plot trends of normalized intensities per bio replicate with ggplot

pSites_ANOVASig_Normalized <- read.csv('phosphorylation_data/Valverde_data/extracts/Phosphosites_XL_Extract_ANOVASig_Norm_NoOA.csv', stringsAsFactors = F)

#Creating unique useful identifier

pSites_ANOVASig_Normalized_Unique_ID <- unite(pSites_ANOVASig_Normalized, 
                                              'Protein_Amino.acid_Site_NumberPhospho', 
                                              c('Protein', 'Amino.acid', 'Position', 'Multiplicity'),
                                              sep ='|', remove = FALSE)


#Df with only columns of interest


pSites_ANOVASig_Normalized_Unique_ID_BioRep <-
  pSites_ANOVASig_Normalized_Unique_ID[, c(1:22, 26, 38, 40:42)]

#Creating object to organize columnes

col_order_BioRep_Normalized <- c('Interphase_000_A', 'Interphase_000_B', 'Interphase_000_C', 'Interphase_030_A', 'Interphase_030_B_R', 
                                 'Interphase_030_C', 'Interphase_060_A', 'Interphase_060_B', 'Interphase_060_C', 'Interphase_090_A', 
                                 'Interphase_090_B', 'Interphase_090_C', 'Interphase_120_A_R', 'Interphase_120_B_R', 'Interphase_120_C', 
                                 'Mitosis_CyB_A', 'Mitosis_CyB_C_R', 'Mitosis_CyB_D', 'Mitosis_CSF', 'Mitosis_CSF_TR_R','Protein_Amino.acid_Site_NumberPhospho',
                                 'Amino.acid', 'Multiplicity', 'Proteins', 'Leading.proteins', 'Protein', 'Sequence.window')

#reordering columns 

pSites_ANOVASig_Normalized_Unique_ID_BioRep_Ord <- pSites_ANOVASig_Normalized_Unique_ID_BioRep[, col_order_BioRep_Normalized]

#merging

pSites_with_clusters_fromHM_Normalized <- merge(x = pSites_ANOVASig_Normalized_Unique_ID_BioRep_Ord,
                                                y = out_df, by.x = 'Protein_Amino.acid_Site_NumberPhospho', by.y = 'ID')

#Tidying up the data

#Changing colnames 

pSites_with_clusters_fromHM_Normalized_tidy <- plyr::rename(pSites_with_clusters_fromHM_Normalized, 
                                                            c('Interphase_120_A_R' = 'Interphase_120_A', 'Interphase_120_B_R' = 'Interphase_120_B',
                                                              'Interphase_030_B_R' = 'Interphase_030_B', 'Mitosis_CyB_C_R' = 'Mitosis_CyB_C',
                                                              'Mitosis_CSF_TR_R' = 'Mitosis_CSF_B', 'Mitosis_CSF' = 'Mitosis_CSF_A', 'Protein_Amino.acid_Site_NumberPhospho' = 'ID'))

#changing several columns to factors

pSites_with_clusters_fromHM_Normalized_tidy$Amino.acid <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy[, 'Amino.acid'])
pSites_with_clusters_fromHM_Normalized_tidy$Multiplicity <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy[, 'Multiplicity'])
pSites_with_clusters_fromHM_Normalized_tidy$Proteins <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy[, 'Proteins'])
pSites_with_clusters_fromHM_Normalized_tidy$Leading.proteins <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy[, 'Leading.proteins'])
pSites_with_clusters_fromHM_Normalized_tidy$Protein <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy[, 'Protein'])
pSites_with_clusters_fromHM_Normalized_tidy$Cluster <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy[, 'Cluster'])


#From wide format to long format

pSites_with_clusters_fromHM_Normalized_tidy_v1 <- melt(pSites_with_clusters_fromHM_Normalized_tidy)

#Separate the variable into 'Condition', 'Time/Treatment', 'Replicate'

pSites_with_clusters_fromHM_Normalized_tidy_v2 <- pSites_with_clusters_fromHM_Normalized_tidy_v1 %>%
  separate(variable, c('Condition', 'Time_Or_Treatment', 'Replicate'), '_')

#changing several columns to factors

pSites_with_clusters_fromHM_Normalized_tidy_v2$Condition <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_v2[, 'Condition'])
pSites_with_clusters_fromHM_Normalized_tidy_v2$Time_Or_Treatment <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_v2[, 'Time_Or_Treatment'])
pSites_with_clusters_fromHM_Normalized_tidy_v2$Replicate <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_v2[, 'Replicate'])

#Transforming all treatments to numbers to use geom_smooth to plot overall trends
#Naming mitotic condition as 150

pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric <- plyr::rename(pSites_with_clusters_fromHM_Normalized, 
                                                                        c('Interphase_120_A_R' = 'Interphase_120_A', 'Interphase_120_B_R' = 'Interphase_120_B',
                                                                          'Interphase_030_B_R' = 'Interphase_030_B', 'Mitosis_CyB_C_R' = 'Mitosis_150_CyBC',
                                                                          'Mitosis_CyB_A' = 'Mitosis_150_CyBA', 'Mitosis_CSF' = 'Mitosis_150_CSFA', 'Mitosis_CSF_TR_R' = 'Mitosis_150_CSFB',
                                                                          'Mitosis_CyB_D' = 'Mitosis_150_CyBD', 
                                                                          'Protein_Amino.acid_Site_NumberPhospho' = 'ID'))

#changing several columns to factors

pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric$Amino.acid <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric[, 'Amino.acid'])
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric$Multiplicity <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric[, 'Multiplicity'])
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric$Proteins <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric[, 'Proteins'])
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric$Leading.proteins <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric[, 'Leading.proteins'])
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric$Protein <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric[, 'Protein'])
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric$Cluster <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric[, 'Cluster'])

#From wide format to long format

pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v1 <- melt(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric)

#Separate the variable into 'Condition', 'Time/Treatment', 'Replicate'

pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2 <- pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v1 %>%
  separate(variable, c('Condition', 'Time_Or_Treatment', 'Replicate'), '_')


#Chaging time column to numeric

pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2$Time_Or_Treatment <- as.numeric(as.character(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2$Time_Or_Treatment))
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2$Condition <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2[, 'Condition'])
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2$Replicate <- as.factor(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2[, 'Replicate'])


#loading hits per motif

load('phosphorylation_data/Valverde_data/extracts/Hits_Aurora_All.RData')
load('phosphorylation_data/Valverde_data/extracts/Hits_Cdc7_All.RData')
load('phosphorylation_data/Valverde_data/extracts/Hits_CdkMinimal_All.RData')
load('phosphorylation_data/Valverde_data/extracts/Hits_CdkFull_All.RData')
# Hits_Aurora_All_List
# Hits_Cdc7_All_List
# Hits_Cdk_Minimal_All_List
# Hits_Cdk_Full_All_List

#Filtering for only IDs with Aurora motif
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Aurora_Only <- 
  subset(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2, ID %in% Hits_Aurora_All_List)

#Filtering for only IDs with Cdc7 motif
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Cdc7_Only <- 
  subset(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2, ID %in% Hits_Cdc7_All_List)

#Filtering for only IDs with cdk minimal motif
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Cdk_Minimal_Only <- 
  subset(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2, ID %in% Hits_Cdk_Minimal_All_List)

#Filtering for only IDs with cdk full motif
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Cdk_Full_Only <- 
  subset(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2, ID %in% Hits_Cdk_Full_All_List)

#Trying to generate a column with motif identifier

pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Aurora_Only$Motif <- 'aurora'
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Cdc7_Only$Motif <- 'cdc7'
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Cdk_Minimal_Only$Motif <- 'cdk_min'
pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Cdk_Full_Only$Motif <- 'cdk_full'

pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_WithMotif <- rbind(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Aurora_Only, 
                                                                              pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Cdc7_Only, 
                                                                              pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Cdk_Minimal_Only, 
                                                                              pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_Cdk_Full_Only)


```

### g

```{r motifs count, echo=FALSE,message=FALSE,warning=FALSE,fig.height=7,fig.width=6}

phosphoSiteBackground <- readr::read_delim("phosphorylation_data/Valverde_data/allPhosphosites_background.tab", "\t",
                                    escape_double = FALSE,
                                    col_types = cols(
                                      ANOVA = readr::col_factor(levels = c("+","-"))),
                                    trim_ws = TRUE)
phosphoSiteBackground <- phosphoSiteBackground %>% mutate(type=case_when(
  substr(sequence,start = 17,stop = 17) == "P" ~ "Proline directed",
  TRUE ~ "Non proline directed"
))

phosphoSiteBackground <- phosphoSiteBackground %>% mutate(sequence_red=  substr(sequence,start = 9,stop = 23))

#______________________________________________________________________________________________________________________________________#

phosphoSiteClusters <- readr::read_delim("phosphorylation_data/Valverde_data/clusterGero.tsb", "\t",
                                  escape_double = FALSE, 
                                  col_types = cols(
                                    Cluster = readr::col_factor(levels = c("cluster1","cluster2", "cluster3", "cluster4")),
                                    sequence = col_character()), 
                                  trim_ws = TRUE)
phosphoSiteClusters <- phosphoSiteClusters %>% mutate(type=case_when(
  substr(sequence,start = 17,stop = 17) == "P" ~ "Proline directed",
  TRUE ~ "Non proline directed"
))

phosphoSiteClusters <- phosphoSiteClusters %>% mutate(sequence_red=  substr(sequence,start = 9,stop = 23))

#Calculate the background estimated probabilities for each aminoacid. All the phosphosites detected have been used for stablishing the background calculations
backgroundSeqs <- Biostrings::AAStringSet(unlist(phosphoSiteBackground$sequence_red))
names(backgroundSeqs)<-phosphoSiteBackground$ID

bkgEstProbs <- colSums(alphabetFrequency(backgroundSeqs))[1:20]/sum(colSums(alphabetFrequency(backgroundSeqs))[1:20])


IC_matrix_list <- list()

# Anova +
IC_matrix_list[["ANOVA positive - All phosphosites"]] <- informationContent_calculator(phosphoSiteClusters$sequence_red,bkgEstProbs)
IC_matrix_list[["ANOVA positive - Proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,type=="Proline directed")$sequence_red,bkgEstProbs)
IC_matrix_list[["ANOVA positive - Non proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,type=="Non proline directed")$sequence_red,bkgEstProbs)

# Cluster A (cluster3)

IC_matrix_list[["Cluster A - All phosphosites"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster3")$sequence_red,bkgEstProbs)
IC_matrix_list[["Cluster A - Proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster3" & type=="Proline directed")$sequence_red,bkgEstProbs)
IC_matrix_list[["Cluster A - Non proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster3" & type=="Non proline directed")$sequence_red,bkgEstProbs)

# Cluster B (cluster4)

IC_matrix_list[["Cluster B - All phosphosites"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster4")$sequence_red,bkgEstProbs)
IC_matrix_list[["Cluster B - Proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster4" & type=="Proline directed")$sequence_red,bkgEstProbs)
IC_matrix_list[["Cluster B - Non proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster4" & type=="Non proline directed")$sequence_red,bkgEstProbs)

# Cluster C (cluster1)

IC_matrix_list[["Cluster C - All phosphosites"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster1")$sequence_red,bkgEstProbs)
IC_matrix_list[["Cluster C - Proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster1" & type=="Proline directed")$sequence_red,bkgEstProbs)
IC_matrix_list[["Cluster C - Non proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster1" & type=="Non proline directed")$sequence_red,bkgEstProbs)

# Cluster D (cluster2)

IC_matrix_list[["Cluster D - All phosphosites"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster2")$sequence_red,bkgEstProbs)
IC_matrix_list[["Cluster D - Proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster2" & type=="Proline directed")$sequence_red,bkgEstProbs)
IC_matrix_list[["Cluster D - Non proline directed"]] <- informationContent_calculator(subset(phosphoSiteClusters,Cluster=="cluster2" & type=="Non proline directed")$sequence_red,bkgEstProbs)

# all phosphosites
IC_matrix_list[["Whole set - All phosphosites"]] <- informationContent_calculator(phosphoSiteBackground$sequence_red,bkgEstProbs)
IC_matrix_list[["Whole set - Proline directed"]] <- informationContent_calculator(subset(phosphoSiteBackground, type=="Proline directed")$sequence_red,bkgEstProbs)
IC_matrix_list[["Whole set - Non proline directed"]] <- informationContent_calculator(subset(phosphoSiteBackground,type=="Non proline directed")$sequence_red,bkgEstProbs)


kinaseMotifList <- list()

# Plk: [D/N/E/Y]-X-[S/T]-[ ?? /F; no P]-[ ?? /X] 
kinaseMotifList[["Plk"]] <- "^.{5}[D|N|E|Y].[S|T][A|V|I||L|F|W|Y|M]"

# # Plk1: [D/E]-X-[S/T]-[??]-X-[D/E] (https://www.jbc.org/content/278/28/25277.long)
# kinaseMotifList[["Plk1"]] <- "^.{5}[D|E].[S|T][A|V|I||L|F|W|Y|M].[D|E]"

# Aurora: R/K-X-S/T-[ ?? /F; no P] -> [KR].[ST][^P] (https:///pwww.ncbi.nlm.nih.govmc/articles/PMC2999863/#b29)
kinaseMotifList[["Aurora A/B"]] <- "^.{5}[R|K].[S|T][^P]" 

# Nek:  (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6570481/)
# Nek general: [L|M|F|W].[S|T][A|V|I||L|F|W|Y|M][K|R]
kinaseMotifList[["Nek1/2/3/4/5/8/10"]] <- "^.{5}[L|M|F|W].[S|T][A|V|I||L|F|W|Y|M][K|R]"
# Nek6/7/9":[F/L/M/W]-X-R-T
kinaseMotifList[["Nek6/7/9"]] <- "^.{4}[L|M|F][D|E|N][D|E|Y]S[A|V|I||L|F|W|Y|M]"

# Casein kinase 1: D/E-D/E-D/E-X-X-S/T or # pS/pT-X-X-S/T 
kinaseMotifList[["Ck1"]] <- "^((.{2}[E|D][E|D][E|D])|(.{4}[S|T])).{2}[S|T]"

# Casein kinase 2: [pS/pT]-{P}-x-[E/D] or [pS/pT]-{P}-x-pS, (https://www.sciencedirect.com/science/article/pii/S2405580815000680)
kinaseMotifList[["Ck2"]] <- "^.{6}[S|T][S|T].[E|D|S]"

# CDC7: [E/D]-[pS/pT]-[E/D]-x-[E/D] 
kinaseMotifList[["Cdc7"]] <- "^((.{7}[S|T][E|D].[E|D])|(.{7}[S|T][S|T][P]))"

# PKA: R-R/K-X-S-??
kinaseMotifList[["Pka"]] <- "^.{4}R[R|K].[S|T][A|V|I||L|F|W|Y|M]"

# Cdk1 full
kinaseMotifList[["Cdk full"]] <- "^.{7}[S|T]P.[K|R]"

# Cddk minimal
kinaseMotifList[["Cdk minimal"]] <- "^.{7}[S|T]P"

# TOTAL
kinaseMotifList[["Total"]] <- "^.{7}[S|T|Y]"



motifCounts_clusterA <- sapply(kinaseMotifList, function(x){
  length(grep(x,subset(phosphoSiteClusters, Cluster=="cluster3")$sequence_red))
})
motifCounts_clusterB <- sapply(kinaseMotifList, function(x){
  length(grep(x,subset(phosphoSiteClusters, Cluster=="cluster4")$sequence_red))
})
motifCounts_clusterC <- sapply(kinaseMotifList, function(x){
  length(grep(x,subset(phosphoSiteClusters, Cluster=="cluster1")$sequence_red))
})
motifCounts_clusterD <- sapply(kinaseMotifList, function(x){
  length(grep(x,subset(phosphoSiteClusters, Cluster=="cluster2")$sequence_red))
})
motifCounts_all <- sapply(kinaseMotifList, function(x){
  length(grep(x,phosphoSiteClusters$sequence_red))
})

motifCounts_table <- rbind(motifCounts_clusterA,motifCounts_clusterB,motifCounts_clusterC,motifCounts_clusterD,motifCounts_all)

row.names(motifCounts_table) <- c("Cluster A","Cluster B","Cluster C","Cluster D","Total")
# colnames(NonProDirected_motifCounts_table) <- c("Plk","Aurora","Nek","Ck1","Unknown","Ck2","Pka")



```


## Fig. 2

### b

```{r embryo heatmap - High-time resolutionr,echo=FALSE,message=FALSE,warning=FALSE,fig.height=6,fig.width=5}
#Load data#

PeptideRatioResults <- read.csv('phosphorylation_data/Valverde_data/Peptide_Ratio_Results.csv', stringsAsFactors = FALSE)

#changing class of variable of interest#

PeptideRatioResults$RatioLightToHeavy <- as.numeric(as.character(PeptideRatioResults$RatioLightToHeavy))

#Remove iRT peptides#

PeptideRatioResults <-  filter(PeptideRatioResults, Protein != '>sp|P00001|Biognosys|iRT Kit Fusion')

#Transforming data to log2#

PeptideRatioResults <- PeptideRatioResults %>%
  mutate(log2values = log2(RatioLightToHeavy))

#Converting time values to numeric#

#Change of time values so that they are sorted chronologically#

PeptideRatioResults$Condition[PeptideRatioResults$Condition == "0'"] <- "0h00'"

PeptideRatioResults$Condition[PeptideRatioResults$Condition == "30'"] <- "0h30'"

PeptideRatioResults$Condition[PeptideRatioResults$Condition == "15'"] <- "0h15'"

PeptideRatioResults$Condition[PeptideRatioResults$Condition == "60'"] <- "1h00'"

#Separating hours and minutes in different columns#

PeptideRatioResults <- PeptideRatioResults %>%
  separate(Condition, c("hours", "minutes"), "h")

#Gather only numerical from minutes column#

PeptideRatioResults$minutes <-  parse_number(PeptideRatioResults$minutes)

#Converting hours to numeric values#

PeptideRatioResults$hours <- as.numeric(PeptideRatioResults$hours)

#Multiplying hours value per 60 to get minutes#

PeptideRatioResults$hoursX60 <- PeptideRatioResults$hours * 60

#Adding minutes from hours to minutes to get final time#

PeptideRatioResults$TimeMinutes <- PeptideRatioResults$minutes + PeptideRatioResults$hoursX60

#Tidying up#

PeptideRatioResults <- subset(PeptideRatioResults, select = -c(hours, minutes, hoursX60))

#Mean intensities and SD per time point (averaging of bio-replicates)#

PeptideRatioResults <- PeptideRatioResults %>%
  group_by(Protein, Peptide, Peptide.Modified.Sequence, TimeMinutes) %>%
  summarise(MeanIntensity = mean(RatioLightToHeavy), SDIntensity = sd(RatioLightToHeavy), MeanLog2 = mean(log2values), SDLog2 = sd(log2values))

#Data from Long format to Wide format#

PeptideRatioResults_Wide <- dcast(PeptideRatioResults, Protein + Peptide + Peptide.Modified.Sequence ~ TimeMinutes, value.var = c('MeanLog2'))

#Uniting columns#

PeptideRatioResults_Wide <- unite(PeptideRatioResults_Wide, 'Protein_Peptide', c('Protein', 'Peptide.Modified.Sequence'), sep = '|', remove = FALSE )

#Removing columns and adding row names#
rownames(PeptideRatioResults_Wide) <- PeptideRatioResults_Wide[, 1]
PeptideRatioResults_Wide <- PeptideRatioResults_Wide[, -c(1, 2, 3, 4)] 

#Converting to a numeric matrix#

PeptideRatioResults_Wide_AsMatrix <- as.matrix(PeptideRatioResults_Wide)

#Removing temporarily those with inf values#

PeptideRatioResults_Wide_AsMatrix <- PeptideRatioResults_Wide_AsMatrix[!is.infinite(rowSums(PeptideRatioResults_Wide_AsMatrix)),]

#plotting#

# heatmap(PeptideRatioResults_Wide_AsMatrix, Colv = NA, scale = "row")

#Using Complex Heatmap

#transposing in order to apply scale#

PeptideRatioResults_Wide_AsMatrix <- t(PeptideRatioResults_Wide_AsMatrix)

#scaling

PeptideRatioResults_Wide_AsMatrix_Zscored <- scale(PeptideRatioResults_Wide_AsMatrix)

#transposing again#

PeptideRatioResults_Wide_AsMatrix_Zscored <- t(PeptideRatioResults_Wide_AsMatrix_Zscored)

#plotting with complex heatmap
# Heatmap(PeptideRatioResults_Wide_AsMatrix_Zscored, column_order = colnames(PeptideRatioResults_Wide_AsMatrix_Zscored),
#         clustering_distance_rows = 'spearman', clustering_method_rows = 'complete',
#         show_row_names =TRUE, row_dend_side = "left", width = unit(7.5, "cm"),height = unit(15, 'cm'),
#         cluster_row_slices = TRUE, border = TRUE, row_dend_width = unit(2, 'cm'),
#         show_parent_dend_line = FALSE, name = 'Z-score', row_names_gp = gpar(fontsize = 5), column_names_gp = gpar(fontsize = 5),
#         column_names_rot = 45)

#to replace missing values 
#since data is mostly complete, I will replace a few missing values found in some of the p-sites, in order to not miss on their trends

PeptideRatioResults_Wide_Complete <- PeptideRatioResults_Wide

#Turning -inf into NA
is.na(PeptideRatioResults_Wide_Complete) <- do.call(cbind,lapply(PeptideRatioResults_Wide_Complete, is.infinite))

#Replacing missing values
PeptideRatioResults_Wide_Complete <- simple.impute.data.frame(PeptideRatioResults_Wide_Complete, min)

#Assigning row names again
row.names(PeptideRatioResults_Wide_Complete) <- row.names(PeptideRatioResults_Wide_Complete)

#Plotting again to assess results

#Converting to a numeric matrix#

PeptideRatioResults_Wide_Complete_AsMatrix <- as.matrix(PeptideRatioResults_Wide_Complete)

#Getting z scores 

#transposing in order to apply scale#

PeptideRatioResults_Wide_Complete_AsMatrix <- t(PeptideRatioResults_Wide_Complete_AsMatrix)

#scaling

PeptideRatioResults_Wide_Complete_AsMatrix_Zscored <- scale(PeptideRatioResults_Wide_Complete_AsMatrix)

#transposing again#

PeptideRatioResults_Wide_Complete_AsMatrix_Zscored <- t(PeptideRatioResults_Wide_Complete_AsMatrix_Zscored)

#plotting with complex heatmap

#All (scale has to include from -4 to 3)
# Heatmap(PeptideRatioResults_Wide_Complete_AsMatrix_Zscored, column_order = colnames(PeptideRatioResults_Wide_Complete_AsMatrix_Zscored),
#         clustering_distance_rows = 'spearman', clustering_method_rows = 'complete',
#         show_row_names =TRUE, row_dend_side = "left", width = unit(7.5, "cm"),height = unit(15, 'cm'),
#         cluster_row_slices = TRUE, border = TRUE, row_dend_width = unit(2, 'cm'),
#         show_parent_dend_line = FALSE, name = 'Z-score', row_names_gp = gpar(fontsize = 5), column_names_gp = gpar(fontsize = 5),
#         column_names_rot = 45)


library(circlize)
col_fun = colorRamp2(c(-2.5, 0, 2.5), c("cyan", "black", "yellow"))

Heatmap(PeptideRatioResults_Wide_Complete_AsMatrix_Zscored, column_order = colnames(PeptideRatioResults_Wide_Complete_AsMatrix_Zscored),
        clustering_distance_rows = 'spearman', clustering_method_rows = 'complete',
        show_row_names =FALSE, row_dend_side = "left",
        cluster_row_slices = TRUE, border = TRUE, row_dend_width = unit(1, 'cm'),
        show_parent_dend_line = FALSE, name = 'Z-score', row_names_gp = gpar(fontsize = 5), column_names_gp = gpar(fontsize = 6),
        column_names_rot = 45, col = col_fun)



```


### c


```{r single psites - biological processes ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=6,fig.width=5}
PeptideRatioResults <- read.csv('phosphorylation_data/Valverde_data/Peptide_Ratio_Results.csv', stringsAsFactors = FALSE)

#Remove iRT peptides#

PeptideRatioResults <-  filter(PeptideRatioResults, Protein != '>sp|P00001|Biognosys|iRT Kit Fusion')

#changing class of variable of interest

PeptideRatioResults$RatioLightToHeavy <- as.numeric(as.character(PeptideRatioResults$RatioLightToHeavy))

#Change of time values so that they are sorted chronologically#

PeptideRatioResults$Condition[PeptideRatioResults$Condition == "0'"] <- "0h00'"

PeptideRatioResults$Condition[PeptideRatioResults$Condition == "30'"] <- "0h30'"

PeptideRatioResults$Condition[PeptideRatioResults$Condition == "15'"] <- "0h15'"

PeptideRatioResults$Condition[PeptideRatioResults$Condition == "60'"] <- "1h00'"

#Converting time values to numeric#

#Separating hours and minutes in different columns#

PeptideRatioResults <- PeptideRatioResults %>%
  separate(Condition, c("hours", "minutes"), "h")

#Gather only numerical from minutes column#

PeptideRatioResults$minutes <-  parse_number(PeptideRatioResults$minutes)

#Converting hours to numeric values#

PeptideRatioResults$hours <- as.numeric(PeptideRatioResults$hours)

#Multiplying hours value per 60 to get minutes#

PeptideRatioResults$hoursX60 <- PeptideRatioResults$hours * 60

#Adding minutes from hours to minutes to get final time#

PeptideRatioResults$TimeMinutes <- PeptideRatioResults$minutes + PeptideRatioResults$hoursX60

#Tidying up the matrix#

PeptideRatioResults <- subset(PeptideRatioResults, select = -c(hours, minutes, hoursX60))

#Log2 transforming data ---- REMEMBER: comment out for line plots!!!!

PeptideRatioResults$RatioLightToHeavy <- log2(PeptideRatioResults$RatioLightToHeavy)

#Removing -infs

PeptideRatioResults_NoMVS <- PeptideRatioResults[!PeptideRatioResults$RatioLightToHeavy=='-Inf',]

PeptideRatioResults_NoMVS_mean <- PeptideRatioResults_NoMVS %>% 
  group_by(Peptide.Modified.Sequence, TimeMinutes) %>% 
  dplyr::summarize(average = mean(RatioLightToHeavy))

#Plotting 

#Chromatin remodeling
x <- ggplot(subset(PeptideRatioResults_NoMVS, Peptide.Modified.Sequence %in% c("SLFLSVES[+80]PK")), aes(x = TimeMinutes, y = RatioLightToHeavy, color = Peptide.Modified.Sequence))+
  geom_point(size = 6)+
  scale_color_manual(values = c("darkorchid1", "darkmagenta", "darkorchid4"))+
  geom_point(colour = 'white', size = 2)+
  geom_line(subset(PeptideRatioResults_NoMVS_mean, Peptide.Modified.Sequence %in% c("SLFLSVES[+80]PK")), 
            mapping = aes(x = TimeMinutes, y = average), size = 1) +
  ggtitle("ID_step3_010113|kdm3b")+
  geom_vline(xintercept = 85, linetype = "dashed", color = "black", size = 0.5)+
  geom_vline(xintercept = 118, linetype = "dashed", color = "black", size = 0.5)+
  theme_bw()+
  scale_y_continuous(breaks = c(0, 0.05, 0.10, 0.15, 0.20))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.5), 
        axis.text.x = element_text(face="bold", size = 12), axis.text.y = element_text(face="bold", size = 12), legend.position = 'none',
        panel.border = element_rect(colour = "black", fill=NA, size=0.5))

x + scale_x_break(c(35, 57), scales = 4, space = 0.5)

#DNA replication

q <- ggplot(subset(PeptideRatioResults_NoMVS, Peptide.Modified.Sequence %in% c("SSTLTSSAC[+57]SVFQS[+80]PK")), aes(x = TimeMinutes, y = RatioLightToHeavy, color = Peptide.Modified.Sequence))+
  geom_point(size = 6)+
  scale_color_manual(values = c("darkturquoise", "cyan", "darkslategray3", "darkslategray1"))+
  geom_point(colour = 'white', size = 2)+
  geom_line(subset(PeptideRatioResults_NoMVS_mean, Peptide.Modified.Sequence %in% c("SSTLTSSAC[+57]SVFQS[+80]PK")), 
            mapping = aes(x = TimeMinutes, y = average), size = 1) +
  ggtitle("ID_step3_001277|mcm10")+
  geom_vline(xintercept = 85, linetype = "dashed", color = "black", size = 0.5)+
  geom_vline(xintercept = 118, linetype = "dashed", color = "black", size = 0.5)+
  theme_bw()+
  scale_y_continuous(breaks = c(-0.4, -0.2, 0, 0.2, 0.4, 0.6, 0.8))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.5), 
        axis.text.x = element_text(face="bold", size = 12), axis.text.y = element_text(face="bold", size = 12), legend.position = 'none',
        panel.border = element_rect(colour = "black", fill=NA, size=0.5))
q + scale_x_break(c(35, 57), scales = 4, space = 0.5)

#RNP processing

z <- ggplot(subset(PeptideRatioResults_NoMVS, Peptide.Modified.Sequence %in% c("VDVKPEAPVHS[+80]PR")), aes(x = TimeMinutes, y = RatioLightToHeavy, color = Peptide.Modified.Sequence))+
  geom_point(size = 6)+
  scale_color_manual(values = c("darkorange"))+
  geom_point(colour = 'white', size = 2)+
  geom_line(subset(PeptideRatioResults_NoMVS_mean, Peptide.Modified.Sequence %in% c("VDVKPEAPVHS[+80]PR")), 
            mapping = aes(x = TimeMinutes, y = average), size = 1) +
  ggtitle("ID_step3_008661|g3bp2")+
  geom_vline(xintercept = 85, linetype = "dashed", color = "black", size = 0.5)+
  geom_vline(xintercept = 118, linetype = "dashed", color = "black", size = 0.5)+
  theme_bw()+
  scale_y_continuous(breaks = c(0, 0.1, 0.2, 0.3, 0.4, 0.5))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.5), 
        axis.text.x = element_text(face="bold", size = 12), axis.text.y = element_text(face="bold", size = 12), legend.position = 'none',
        panel.border = element_rect(colour = "black", fill=NA, size=0.5))
z + scale_x_break(c(35, 57), scales = 4, space = 0.5)
#NPC

w <- ggplot(subset(PeptideRatioResults_NoMVS, Peptide.Modified.Sequence %in% c("FSNFPSSLDPYPDVRS[+80]PR")), aes(x = TimeMinutes, y = RatioLightToHeavy, color = Peptide.Modified.Sequence))+
  geom_point(size = 6)+
  scale_color_manual(values = c("deepskyblue4"))+
  geom_point(colour = 'white', size = 2)+
  geom_line(subset(PeptideRatioResults_NoMVS_mean, Peptide.Modified.Sequence %in% c("FSNFPSSLDPYPDVRS[+80]PR")), 
            mapping = aes(x = TimeMinutes, y = average), size = 1) +
  ggtitle("D_step3_007660|nup98")+
  geom_vline(xintercept = 85, linetype = "dashed", color = "black", size = 0.5)+
  geom_vline(xintercept = 118, linetype = "dashed", color = "black", size = 0.5)+
  theme_bw()+
  scale_y_continuous(breaks = c(-0.25, 0, 0.25, 0.5, 0.75, 1.0, 1.25, 1.5))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.5), 
        axis.text.x = element_text(face="bold", size = 12), axis.text.y = element_text(face="bold", size = 12), legend.position = 'none',
        panel.border = element_rect(colour = "black", fill=NA, size=0.5))

w + scale_x_break(c(35, 57), scales = 4, space = 0.5)

```

### d

```{r single psites - CDK1 Y15,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=6}

ggplot(subset(PeptideRatioResults_NoMVS, Protein %in% c("ID_step3_000299|cdk1a")), aes(x = TimeMinutes, y = 2^RatioLightToHeavy, color = Peptide.Modified.Sequence))+
  geom_point(size = 4)+
  scale_color_manual(values = c("dodgerblue2"))+
  geom_point(colour = 'white', size = 1)+
  geom_line(subset(PeptideRatioResults_NoMVS_mean, Peptide.Modified.Sequence %in% c("IGEGTY[+80]GVVYK")), 
            mapping = aes(x = TimeMinutes, y = 2^average), size = 1) +
  ggtitle("Cdk1 - Y15")+
  geom_vline(xintercept = 85, linetype = "dashed", color = "black", size = 0.5)+
  geom_vline(xintercept = 118, linetype = "dashed", color = "black", size = 0.5)+
  theme_bw() + 
  # scale_y_continuous(breaks = c(0.00025,0.00050,0.00075,0.01), limits = c(0, 0.001))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.5),
       axis.text.x = element_text(face="bold", size = 12), axis.text.y = element_text(face="bold", size = 12), legend.position = 'none',
       panel.border = element_rect(colour = "black", fill=NA, size=0.5)) + scale_x_break(c(35, 57), scales = 4, space = 0.5)


```

### f

```{r single psites - CDK1 network oscillating,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=5}

#Plotting 


ggplot(subset(PeptideRatioResults_NoMVS, Peptide.Modified.Sequence %in% c("DAEQS[+80]PAFAYGR")), aes(x = TimeMinutes, y = 2^RatioLightToHeavy, color = Peptide.Modified.Sequence))+
  geom_point(size = 4)+
  scale_color_manual(values = c("red3"))+
  geom_point(colour = 'white', size = 1)+
  geom_line(subset(PeptideRatioResults_NoMVS_mean, Peptide.Modified.Sequence %in% c("DAEQS[+80]PAFAYGR")), 
            mapping = aes(x = TimeMinutes, y = 2^average), size = 1) +
  ggtitle("Cdk1 - Y15")+
  geom_vline(xintercept = 85, linetype = "dashed", color = "black", size = 0.5)+
  geom_vline(xintercept = 118, linetype = "dashed", color = "black", size = 0.5)+
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.5),
       axis.text.x = element_text(face="bold", size = 12), axis.text.y = element_text(face="bold", size = 12), legend.position = 'none',
       panel.border = element_rect(colour = "black", fill=NA, size=0.5)) + scale_x_break(c(35, 57), scales = 4, space = 0.5) + ggtitle("NIPA - S324")


ggplot(subset(PeptideRatioResults_NoMVS, Peptide.Modified.Sequence %in% c("SYNS[+80]PINVSNVSEPSK")), aes(x = TimeMinutes, y = 2^RatioLightToHeavy, color = Peptide.Modified.Sequence))+
  geom_point(size = 4)+
  scale_color_manual(values = c("red3"))+
  geom_point(colour = 'white', size = 1)+
  geom_line(subset(PeptideRatioResults_NoMVS_mean, Peptide.Modified.Sequence %in% c("SYNS[+80]PINVSNVSEPSK")), 
            mapping = aes(x = TimeMinutes, y = 2^average), size = 1) +
  ggtitle("Cdk1 - Y15")+
  geom_vline(xintercept = 85, linetype = "dashed", color = "black", size = 0.5)+
  geom_vline(xintercept = 118, linetype = "dashed", color = "black", size = 0.5)+
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.5),
       axis.text.x = element_text(face="bold", size = 12), axis.text.y = element_text(face="bold", size = 12), legend.position = 'none',
       panel.border = element_rect(colour = "black", fill=NA, size=0.5)) + scale_x_break(c(35, 57), scales = 4, space = 0.5) + ggtitle("GWL - S677")


ggplot(subset(PeptideRatioResults_NoMVS, Peptide.Modified.Sequence %in% c("HLAPMDEAGMLS[+80]PVAELR")), aes(x = TimeMinutes, y = 2^RatioLightToHeavy, color = Peptide.Modified.Sequence))+
  geom_point(size = 4)+
  scale_color_manual(values = c("red3"))+
  geom_point(colour = 'white', size = 1)+
  geom_line(subset(PeptideRatioResults_NoMVS_mean, Peptide.Modified.Sequence %in% c("HLAPMDEAGMLS[+80]PVAELR")), 
            mapping = aes(x = TimeMinutes, y = 2^average), size = 1) +
  ggtitle("Cdk1 - Y15")+
  geom_vline(xintercept = 85, linetype = "dashed", color = "black", size = 0.5)+
  geom_vline(xintercept = 118, linetype = "dashed", color = "black", size = 0.5)+
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.5),
       axis.text.x = element_text(face="bold", size = 12), axis.text.y = element_text(face="bold", size = 12), legend.position = 'none',
       panel.border = element_rect(colour = "black", fill=NA, size=0.5)) + scale_x_break(c(35, 57), scales = 4, space = 0.5) + ggtitle("APC1 - S558")



```

## Fig. 3

### b

#### Xenopus

```{r Expected Vs observed Phospho S/T in IDRs - xenopus,echo=FALSE,message=FALSE,warning=FALSE}



all_protein_id_phospho <- read_lines("phosphorylation_data/Valverde_data/ids/xenopus_allphosphosites_mpi.txt")
cluster_A <- read_lines("phosphorylation_data/Valverde_data/ids/xenopusclusterA_mpi.txt")
cluster_B <- read_lines("phosphorylation_data/Valverde_data/ids/xenopusclusterB_mpi.txt")
cluster_C <- read_lines("phosphorylation_data/Valverde_data/ids/xenopusclusterC_mpi.txt")
cluster_D <- read_lines("phosphorylation_data/Valverde_data/ids/xenopusclusterD_mpi.txt")
significant_anova <- Reduce(union, list(cluster_A,cluster_B,cluster_C,cluster_D))
human_CDK1targets <- read_lines("phosphorylation_data/Valverde_data/ids/humanCDKtargets_mpi.txt")
xenopus_ANOVA_data <- read_delim("phosphorylation_data/Valverde_data/Xen_phospho_AnovaPhosphosites_curated.txt","\t", escape_double = FALSE, trim_ws = TRUE)
xenopus_extract_ANOVA_data <- read_delim("phosphorylation_data/Valverde_data/Xen_Extracts_phospho_AnovaPhosphosites_curated.txt","\t", escape_double = FALSE, trim_ws = TRUE)


disoPath <- "disorder_data/local_predictions/IUpred_run_xenopus/"
file.names <- dir(disoPath, pattern =".iupred")
ids <-unlist(strsplit(x = file.names,split = ".iupred",fixed = T))
all_predictions <- data.frame(ids)
colnames(all_predictions) <- "ID"
all_predictions$ID <- as.character(all_predictions$ID)
positions <- list()
scores <- list()
disordered <- list()

for (n in 1:length(file.names)) {
  aux_table <- read_delim(paste(disoPath,file.names[n],sep = ""),"\t", escape_double = FALSE, col_names = FALSE,col_types = cols(X1 = col_integer(),X2 = col_character(),X3 = col_double()),comment = "#", trim_ws = TRUE)
  aux_table <- as.data.frame(aux_table)
  colnames(aux_table) <- c("POS","RES","IUPRED_SCORE")
  aux_table$IUPRED_DISO <- aux_table$IUPRED_SCORE>=0.5
  all_predictions[n,"sequence"] <- paste(aux_table$RES,collapse = "")
  positions[[n]] <- as.numeric(aux_table$POS)
  scores[[n]] <- as.numeric(aux_table$IUPRED_SCORE)
  # all_predictions[n,"phospho"] <- NULL
  all_predictions[n,"threshold"] <- 0.5
  disordered[[n]] <- which(aux_table$IUPRED_DISO)
}
all_predictions$positions <-positions
all_predictions$IUPredScores <-scores
all_predictions$disordered <-disordered

# remove unannotated proteins
# all_protein_id_phospho <- all_protein_id_phospho[all_protein_id_phospho %in% all_predictions$ID]
all_predictions_phospho <- subset(all_predictions, ID %in% all_protein_id_phospho)
all_predictions_phospho$length <- nchar(all_predictions_phospho$sequence)

# Introduce phosphosites (from all_proteins variable) using the apply function
phosphosites <- read_delim("phosphorylation_data/Valverde_data/all_phosphosites_xenopus_nr.tab", "\t", escape_double = FALSE, col_types = cols(`Leading proteins` = col_skip(), Protein = col_skip()), trim_ws = TRUE)

# Add phosphosites from the extracts!!! uncomment if wanted. 
phosphosites_extracts <- read_delim("phosphorylation_data/Valverde_data/all_phosphosites_xenopus_extracts_nr.tab", "\t", escape_double = FALSE, trim_ws = TRUE)
phosphosites <- rbind(phosphosites,phosphosites_extracts)


phosphosites <- phosphosites %>% dplyr::rename(ID=Proteins,psites=`Positions within proteins`,seqWindow=`Sequence window`,UID=`Unique identifier`) %>% group_by(ID) %>% summarise_at(c("psites","seqWindow","UID"),function(x){paste(x, collapse=",")})
# The data has been opened by excel and some proteins have their names modified to be dates.


phosphosites$psites <- lapply(phosphosites$psites, function(x){ return(as.numeric(strsplit(x,",")[[1]]))})
phosphosites$psites_count <- sapply(phosphosites$psites, length)
phosphosites$UID <- lapply(phosphosites$UID, function(x){ return(as.character(strsplit(x,",")[[1]]))})

xenopus_dynamic <- xenopus_ANOVA_data$UID
# Add phosphosites from the extracts!!! uncomment if wanted.
xenopus_dynamic <- c(xenopus_dynamic,xenopus_extract_ANOVA_data$UID)
#

phosphosites$anova_psites <- apply(phosphosites, 1, function(x){x$psites[x$UID %in% xenopus_dynamic]})
all_predictions_phospho <- merge.data.frame(all_predictions_phospho,phosphosites,by = "ID")


all_predictions_phospho <- all_predictions_phospho %>% mutate(anova_sig=case_when(
  ID %in% significant_anova ~ "Dynamic",
  TRUE ~ "Non dynamic"
))


all_predictions_phospho <- all_predictions_phospho %>% mutate(hCDK1target=case_when(
  ID %in% human_CDK1targets ~ "Human CDK1 target",
  TRUE ~ "other"
))


phosphoDiso_ST <- list()
phosphoDiso_obs <- numeric()
phosphoDiso_expct <- numeric()
phosphoDiso_expct_prob <- numeric()
for (i in 1:nrow(all_predictions_phospho)) {
  TStotalIndexes <- as.numeric(gregexpr("S|T", all_predictions_phospho[i,"sequence"])[[1]]) 
  TSinDiso_count <- sum(TStotalIndexes %in% all_predictions_phospho[i,"disordered"][[1]])
  TSinDiso_fraction <- TSinDiso_count/length(TStotalIndexes)
  phosphoDiso_ST[[i]] <- TStotalIndexes
  phosphoDiso_expct_prob[i] <- TSinDiso_fraction
  phosphoDiso_expct[i] <- all_predictions_phospho[i,"psites_count"][[1]]*TSinDiso_fraction
  phosphoDiso_obs[i] <- sum(all_predictions_phospho[i,"psites"][[1]] %in% all_predictions_phospho[i,"disordered"][[1]])
}

all_predictions_phospho$ST_residues <- phosphoDiso_ST
all_predictions_phospho$psites_obsv_diso <- phosphoDiso_obs
all_predictions_phospho$psites_expct_diso <- phosphoDiso_expct
all_predictions_phospho$psites_expct_diso_prob <- phosphoDiso_expct_prob


# Binomial test:
# Calculate percentage of disordered region from all proteins
all_predictions_phospho <- as.data.table(all_predictions_phospho)
all_predictions_phospho[,binom := purrr::pmap(.(psites_obsv_diso, psites_count, psites_expct_diso_prob), binom.test, alternative="greater")]
all_predictions_phospho[,binom_p := mapply("[[", binom, "p.value", SIMPLIFY = T)]
all_predictions_phospho[,binom_q := mapply(p.adjust, binom_p)]
all_predictions_phospho[,binom_sig := factor(ifelse(binom_q < 0.05, ifelse(binom_q < 0.01, "1% FDR", "5% FDR"), "n.s."), levels=c("n.s.","5% FDR", "1% FDR")) ]



ggplot(all_predictions_phospho) + 
  geom_point(aes(x=psites_obsv_diso,y=psites_expct_diso, colour = binom_sig,shape=anova_sig),size=2,alpha=0.80)+
  geom_abline(color="darkslategrey",slope = 1,size=0.5,linetype = "dashed")+
  ggpubr::theme_classic2() + 
  theme(text = element_text(size=15),legend.position = c(0.32,0.60),legend.box.just = "left",legend.box.margin = margin(2, 2, 2, 2),legend.box.background = element_rect(color="darkslategrey"),legend.title = element_text(size = 13)) +
  guides(color=guide_legend(title="Statistical significance")) +
  scale_x_continuous(limits = c(0, 30),breaks = c(seq(0, 30, by = 5)))+ xlab("Observed phospho S/T in IDR") +
  scale_y_continuous(limits = c(0, 30),breaks = c(seq(0, 30, by = 5)))+ ylab("Expected phospho S/T in IDR") + 
  scale_colour_manual(values = c(pal_jco()(10)[3],"#ffdd15ff",pal_jco()(10)[4]))+
  scale_shape_manual(values = c(16,4,4))

```

#### Human

```{r Expected Vs observed Phospho S/T in IDRs - human,echo=FALSE,message=FALSE,warning=FALSE}
disoPath <- "disorder_data/local_predictions/IUpred_run_human/"
file.names <- dir(disoPath, pattern =".iupred")
ids <-unlist(strsplit(x = file.names,split = ".iupred",fixed = T))
all_predictions <- data.frame(ids)
colnames(all_predictions) <- "ID"
all_predictions$ID <- as.character(all_predictions$ID)
positions <- list()
scores <- list()
disordered <- list()

for (n in 1:length(file.names)) {
  aux_table <- read_delim(paste(disoPath,file.names[n],sep = ""),"\t", escape_double = FALSE, col_names = FALSE,col_types = cols(X1 = col_integer(),X2 = col_character(),X3 = col_double()),comment = "#", trim_ws = TRUE)
  aux_table <- as.data.frame(aux_table)
  colnames(aux_table) <- c("POS","RES","IUPRED_SCORE")
  aux_table$IUPRED_DISO <- aux_table$IUPRED_SCORE>=0.5
  all_predictions[n,"sequence"] <- paste(aux_table$RES,collapse = "")
  positions[[n]] <- as.numeric(aux_table$POS)
  scores[[n]] <- as.numeric(aux_table$IUPRED_SCORE)
  disordered[[n]] <- which(aux_table$IUPRED_DISO)
}
all_predictions$positions <-positions
all_predictions$IUPredScores <-scores
all_predictions$disordered <-disordered

all_predictions$length <- nchar(all_predictions$sequence)

human_data<-merge.data.frame(human_data,all_predictions,by.x = "ACC#",by.y = "ID")
human_data$psites_count <- sapply(human_data$psite, length)
human_data$psites_CDK1_count <- sapply(human_data$psite_CDK1, length)


phosphoDiso_ST <- list()
phosphoDiso_obs <- numeric()
phosphoDiso_expct <- numeric()
phosphoDiso_expct_prob <- numeric()
for (i in 1:nrow(human_data)) {
  TStotalIndexes <- as.numeric(gregexpr("S|T", human_data[i,"Sequence"])[[1]]) 
  TSinDiso_count <- sum(TStotalIndexes %in% human_data[i,"disordered"][[1]])
  TSinDiso_fraction <- TSinDiso_count/length(TStotalIndexes)
  phosphoDiso_ST[[i]] <- TStotalIndexes
  phosphoDiso_expct_prob[i] <- TSinDiso_fraction
  phosphoDiso_expct[i] <- human_data[i,"psites_count"][[1]]*TSinDiso_fraction
  phosphoDiso_obs[i] <- sum(human_data[i,"psite"][[1]] %in% human_data[i,"disordered"][[1]])
}

human_data$ST_residues <- phosphoDiso_ST
human_data$psites_obsv_diso <- phosphoDiso_obs
human_data$psites_expct_diso <- phosphoDiso_expct
human_data$psites_expct_diso_prob <- phosphoDiso_expct_prob


human_data_dt <- as.data.table(human_data)
human_data_dt[,binom := purrr::pmap(.(psites_obsv_diso, psites_count, psites_expct_diso_prob), binom.test, alternative="greater")]
human_data_dt[,binom_p := mapply("[[", binom, "p.value", SIMPLIFY = T)]
human_data_dt[,binom_q := mapply(p.adjust, binom_p)]
human_data_dt[,binom_sig := factor(ifelse(binom_q < 0.05, ifelse(binom_q < 0.01, "1% FDR", "5% FDR"), "n.s."), levels=c("n.s.","5% FDR", "1% FDR")) ]



ggplot(human_data_dt) + 
  geom_point(aes(x=psites_obsv_diso,y=psites_expct_diso, colour = binom_sig,shape=target),size=2,alpha=0.80)+
  geom_abline(color="darkslategrey",slope = 1,size=0.5,linetype = "dashed")+
  ggpubr::theme_classic2() + 
  theme(text = element_text(size=15),legend.position = c(0.32,0.60),legend.box.just = "left",legend.box.margin = margin(2, 2, 2, 2),legend.box.background = element_rect(color="darkslategrey"),legend.title = element_text(size = 13)) +
  guides(color=guide_legend(title="Statistical significance")) +
  scale_x_continuous(limits = c(0, 60),breaks = c(seq(0, 60, by = 10)))+ xlab("Observed phospho S/T in IDR") +
  scale_y_continuous(limits = c(0, 60),breaks = c(seq(0, 60, by = 10)))+ ylab("Expected phospho S/T in IDR") + 
  scale_colour_manual(values = c(pal_jco()(10)[3],"#ffdd15ff",pal_jco()(10)[4]))+
  scale_shape_manual(values = c(16,4,4))

rm(human_data_dt)

```

### c

```{r Expected vs Observed phospho S/T distribution in xenopus,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=5}
melted_all_predictions_phospho <- all_predictions_phospho %>% melt(id.vars=c("ID","anova_sig","hCDK1target","length"),value.name = "psites_diso",measure.vars=c("psites_expct_diso","psites_obsv_diso"))



# ALL
wilcox.test(psites_diso~variable,data=melted_all_predictions_phospho,paired=T,estimate =T,conf.int=T)
ggplot(melted_all_predictions_phospho) + 
  geom_boxplot(aes(y=psites_diso,x=variable, fill = variable,color = variable),outlier.shape = NA)+
  ggpubr::theme_classic2()  + 
  theme(text = element_text(size=20),legend.position = "none",axis.ticks.x = element_blank()) +
  geom_segment(aes(x = 1, y = 16.1, xend = 2, yend = 16.1)) + annotate(geom="text", x=1.5, y=16.3, label="***",size=10) +
  guides(color=guide_legend(title="Statistical significance")) +
  scale_x_discrete(labels = c("Expected","Observed")) + xlab(element_blank()) +
  scale_y_continuous(limits = c(0, 31),breaks = c(seq(0, 30, by = 2)),expand = c(0.05,0.05))+ ylab("Phospho S/T in IDR") + 
  scale_colour_manual(values = pal_jco()(10)[c(7,10)]) +
  scale_fill_manual(values = pal_jco()(10)[c(2,5)]) + ggtitle("All \n phosphoproteins")

# ANOVA +
wilcox.test(psites_diso~variable,data=subset(melted_all_predictions_phospho,anova_sig == "Dynamic"),paired=T,estimate =T,conf.int=T)
ggplot(subset(melted_all_predictions_phospho,anova_sig == "Dynamic")) + 
  geom_boxplot(aes(y=psites_diso,x=variable, fill = variable,color = variable),outlier.shape = NA)+
  ggpubr::theme_classic2()  + 
  theme(text = element_text(size=20),legend.position = "none",axis.ticks.x = element_blank()) +
  geom_segment(aes(x = 1, y = 16.1, xend = 2, yend = 16.1)) + annotate(geom="text", x=1.5, y=16.3, label="***",size=10) +
  guides(color=guide_legend(title="Statistical significance")) +
  scale_x_discrete(labels = c("Expected","Observed")) + xlab(element_blank()) +
  scale_y_continuous(limits = c(0, 31),breaks = c(seq(0, 30, by = 2)),expand = c(0.05,0.05))+ ylab("Phospho S/T in IDR") + 
  scale_colour_manual(values = pal_jco()(10)[c(7,10)]) +
  scale_fill_manual(values = pal_jco()(10)[c(2,5)]) + ggtitle("Dynamic")

# HumanCDK targets in Dynamic
wilcox.test(psites_diso~variable,data=subset(melted_all_predictions_phospho,anova_sig == "Dynamic" & hCDK1target == "Human CDK1 target"),paired=T,estimate =T,conf.int=T)
ggplot(subset(melted_all_predictions_phospho,hCDK1target == "Human CDK1 target" & anova_sig == "Dynamic")) + 
  geom_boxplot(aes(y=psites_diso,x=variable, fill = variable,color = variable),outlier.shape = NA)+
  ggpubr::theme_classic2()  + 
  theme(text = element_text(size=20),legend.position = "none",axis.ticks.x = element_blank()) +
  geom_segment(aes(x = 1, y = 16.1, xend = 2, yend = 16.1)) + annotate(geom="text", x=1.5, y=16.3, label="***",size=10) +
  guides(color=guide_legend(title="Statistical significance")) +
  scale_x_discrete(labels = c("Expected","Observed")) + xlab(element_blank()) +
  scale_y_continuous(limits = c(0, 31),breaks = c(seq(0, 30, by = 2)),expand = c(0.05,0.05))+ ylab("Phospho S/T in IDR") + 
  scale_colour_manual(values = pal_jco()(10)[c(7,10)]) +
  scale_fill_manual(values = pal_jco()(10)[c(2,5)])+ ggtitle("Dynamic and \n Human CDK target") 
```

### d

```{r Enrichment of Phospho Ser and Ther in IDRs ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=3,fig.width=20}


predictors <- c("IUPred","SPOT","VSL2b")



yeastIntersect_idx_cols <- c("iupl_disoIndexes","spot_disoIndexes","vsl_disoIndexes" )
yeastIntersect_organism <- (rep("Yeast CDK targets",length(yeastIntersect_idx_cols)))
yeastIntersect_proteome_length <- sum(yeastDiso_full$Length)
yeastIntersect_diso_fraction <- numeric()
yeastIntersect_CMHpval <- numeric()
yeastIntersect_CMHcommonOR <- numeric()
yeastIntersect_FisherPval <- numeric()
yeastIntersect_FisherOR <- numeric()
yeastIntersect_FisherCI <- character()
for (i in 1:length(yeastIntersect_idx_cols)) {
  yeastStratContingencyArray <- getStratContingencyArray(df = subset(yeastIntersect_CDK1_data,target_intersect=="Cdk1 target"),
                                                    sequence_col = "Sequence",
                                                    diso_index_col = yeastIntersect_idx_cols[i],
                                                    psites_col = "psite")
  yeastIntersect_CMHtest <- mantelhaen.test(yeastStratContingencyArray)
  yeastIntersect_CMHpval[i]<-yeastIntersect_CMHtest$p.value
  yeastIntersect_CMHcommonOR[i]<-yeastIntersect_CMHtest$estimate

  yeastContingency <- matrix(c(sum(yeastStratContingencyArray[1,1,]),
                               sum(yeastStratContingencyArray[1,2,]),
                               sum(yeastStratContingencyArray[2,1,]),
                               sum(yeastStratContingencyArray[2,2,])),
                             nrow = 2,byrow = T)
  yeastIntersect_FisherTest <- fisher.test(yeastContingency)
  yeastIntersect_FisherPval[i]<-yeastIntersect_FisherTest$p.value
  yeastIntersect_FisherOR[i]<-yeastIntersect_FisherTest$estimate
  yeastIntersect_FisherCI[[i]]<-paste(round(yeastIntersect_FisherTest$conf.int,4), collapse =  " - ")
  yeastIntersect_diso_fraction[i] <-
    (sum(lengths(strsplit(subset(yeastDiso_full,!is.na(yeastDiso_full[, yeastIntersect_idx_cols[i]]))[,yeastIntersect_idx_cols[i]],","))))/yeastIntersect_proteome_length
}


human_idx_cols <- c("iupl_disoIndexes","spot_disoIndexes","vsl_disoIndexes" )
human_organism <- (rep("Human CDK targets",length(human_idx_cols)))
human_proteome_length <- sum(humanDiso_full$Length)
human_diso_fraction <- numeric()
human_CMHpval <- numeric()
human_CMHcommonOR <- numeric()
human_FisherPval <- numeric()
human_FisherOR <- numeric()
human_FisherCI <- character()
for (i in 1:length(human_idx_cols)) {
  humanStratContingencyArray <- getStratContingencyArray(df = subset(human_data,target=="Cdk1 target"),
                                                    sequence_col = "Sequence",
                                                    diso_index_col = human_idx_cols[i],
                                                    psites_col = "psite_CDK1"
                                                    ) 
  human_CMHtest <- mantelhaen.test(humanStratContingencyArray)
  human_CMHpval[i]<-human_CMHtest$p.value
  human_CMHcommonOR[i]<-human_CMHtest$estimate
  
  humanContingency <- matrix(c(sum(humanStratContingencyArray[1,1,]),
                               sum(humanStratContingencyArray[1,2,]),
                               sum(humanStratContingencyArray[2,1,]),
                               sum(humanStratContingencyArray[2,2,])),
                             nrow = 2,byrow = T)
  human_FisherTest <- fisher.test(humanContingency)
  human_FisherPval[i]<-human_FisherTest$p.value
  human_FisherOR[i]<-human_FisherTest$estimate
  human_FisherCI[[i]]<-paste(round(human_FisherTest$conf.int,4), collapse =  " - ")
  human_diso_fraction[i]<-(sum(lengths(strsplit(subset(humanDiso_full,!is.na(humanDiso_full[,human_idx_cols[i]]))[,human_idx_cols[i]],","))))/human_proteome_length

}

#__________________________________________
#TEST addition of human MAPK and AURK
human_mapk_organism <- (rep("Human MAPK targets",length(human_idx_cols)))
human_mapk_CMHpval <- numeric()
human_mapk_CMHcommonOR <- numeric()
human_mapk_FisherPval <- numeric()
human_mapk_FisherOR <- numeric()
human_mapk_FisherCI <- character()
for (i in 1:length(human_idx_cols)) {
  humanStratContingencyArray <- getStratContingencyArray(df = subset(human_data,target_mapk=="mapk target"),
                                                    sequence_col = "Sequence",
                                                    diso_index_col = human_idx_cols[i],
                                                    psites_col = "psite_MAPK"
                                                    ) 
  human_mapk_CMHtest <- mantelhaen.test(humanStratContingencyArray)
  human_mapk_CMHpval[i]<-human_mapk_CMHtest$p.value
  human_mapk_CMHcommonOR[i]<-human_mapk_CMHtest$estimate
  
  humanContingency <- matrix(c(sum(humanStratContingencyArray[1,1,]),
                               sum(humanStratContingencyArray[1,2,]),
                               sum(humanStratContingencyArray[2,1,]),
                               sum(humanStratContingencyArray[2,2,])),
                             nrow = 2,byrow = T)
  human_mapk_FisherTest <- fisher.test(humanContingency)
  human_mapk_FisherPval[i]<-human_mapk_FisherTest$p.value
  human_mapk_FisherOR[i]<-human_mapk_FisherTest$estimate
  human_mapk_FisherCI[[i]]<-paste(round(human_mapk_FisherTest$conf.int,4), collapse =  " - ")
  # Here I removed the proteins with zero prediction of disorder for calculating the fraction of disorder residues in the set. I should keep those and remove the NAs
  # human_diso_fraction[i]<-(sum(lengths(strsplit(subset(humanDiso_full,humanDiso_full[,human_idx_cols[i]]!="0")[,human_idx_cols[i]],","))))/human_proteome_length

}

human_aurk_organism <- (rep("Human AURK targets",length(human_idx_cols)))
human_aurk_CMHpval <- numeric()
human_aurk_CMHcommonOR <- numeric()
human_aurk_FisherPval <- numeric()
human_aurk_FisherOR <- numeric()
human_aurk_FisherCI <- character()
for (i in 1:length(human_idx_cols)) {
  humanStratContingencyArray <- getStratContingencyArray(df = subset(human_data,target_aurk=="aurk target"),
                                                    sequence_col = "Sequence",
                                                    diso_index_col = human_idx_cols[i],
                                                    psites_col = "psite_aurk"
                                                    ) 
  human_aurk_CMHtest <- mantelhaen.test(humanStratContingencyArray)
  human_aurk_CMHpval[i]<-human_aurk_CMHtest$p.value
  human_aurk_CMHcommonOR[i]<-human_aurk_CMHtest$estimate
  
  humanContingency <- matrix(c(sum(humanStratContingencyArray[1,1,]),
                               sum(humanStratContingencyArray[1,2,]),
                               sum(humanStratContingencyArray[2,1,]),
                               sum(humanStratContingencyArray[2,2,])),
                             nrow = 2,byrow = T)
  human_aurk_FisherTest <- fisher.test(humanContingency)
  human_aurk_FisherPval[i]<-human_aurk_FisherTest$p.value
  human_aurk_FisherOR[i]<-human_aurk_FisherTest$estimate
  human_aurk_FisherCI[[i]]<-paste(round(human_aurk_FisherTest$conf.int,4), collapse =  " - ")
  
  
  
  # Here I removed the proteins with zero prediction of disorder for calculating the fraction of disorder residues in the set. I should keep those and remove the NAs
  # human_diso_fraction[i]<-(sum(lengths(strsplit(subset(humanDiso_full,humanDiso_full[,human_idx_cols[i]]!="0")[,human_idx_cols[i]],","))))/human_proteome_length

}

human_plk_organism <- (rep("Human PLK targets",length(human_idx_cols)))
human_plk_CMHpval <- numeric()
human_plk_CMHcommonOR <- numeric()
human_plk_FisherPval <- numeric()
human_plk_FisherOR <- numeric()
human_plk_FisherCI <- character()
for (i in 1:length(human_idx_cols)) {
  humanStratContingencyArray <- getStratContingencyArray(df = subset(human_data,target_plk=="plk target"),
                                                    sequence_col = "Sequence",
                                                    diso_index_col = human_idx_cols[i],
                                                    psites_col = "psite_plk"
                                                    ) 
  human_plk_CMHtest <- mantelhaen.test(humanStratContingencyArray)
  human_plk_CMHpval[i]<-human_plk_CMHtest$p.value
  human_plk_CMHcommonOR[i]<-human_plk_CMHtest$estimate
  
  humanContingency <- matrix(c(sum(humanStratContingencyArray[1,1,]),
                               sum(humanStratContingencyArray[1,2,]),
                               sum(humanStratContingencyArray[2,1,]),
                               sum(humanStratContingencyArray[2,2,])),
                             nrow = 2,byrow = T)
  human_plk_FisherTest <- fisher.test(humanContingency)
  human_plk_FisherPval[i]<-human_plk_FisherTest$p.value
  human_plk_FisherOR[i]<-human_plk_FisherTest$estimate
  human_plk_FisherCI[[i]]<-paste(round(human_plk_FisherTest$conf.int,4), collapse =  " - ")
}

human_dyrk_organism <- (rep("Human DYRK targets",length(human_idx_cols)))
human_dyrk_CMHpval <- numeric()
human_dyrk_CMHcommonOR <- numeric()
human_dyrk_FisherPval <- numeric()
human_dyrk_FisherOR <- numeric()
human_dyrk_FisherCI <- character()
for (i in 1:length(human_idx_cols)) {
  humanStratContingencyArray <- getStratContingencyArray(df = subset(human_data,target_dyrk=="dyrk target"),
                                                    sequence_col = "Sequence",
                                                    diso_index_col = human_idx_cols[i],
                                                    psites_col = "psite_dyrk"
                                                    ) 
  human_dyrk_CMHtest <- mantelhaen.test(humanStratContingencyArray)
  human_dyrk_CMHpval[i]<-human_dyrk_CMHtest$p.value
  human_dyrk_CMHcommonOR[i]<-human_dyrk_CMHtest$estimate
  
  humanContingency <- matrix(c(sum(humanStratContingencyArray[1,1,]),
                               sum(humanStratContingencyArray[1,2,]),
                               sum(humanStratContingencyArray[2,1,]),
                               sum(humanStratContingencyArray[2,2,])),
                             nrow = 2,byrow = T)
  human_dyrk_FisherTest <- fisher.test(humanContingency)
  human_dyrk_FisherPval[i]<-human_dyrk_FisherTest$p.value
  human_dyrk_FisherOR[i]<-human_dyrk_FisherTest$estimate
  human_dyrk_FisherCI[[i]]<-paste(round(human_dyrk_FisherTest$conf.int,4), collapse =  " - ")
}

human_nek_organism <- (rep("Human NEK targets",length(human_idx_cols)))
human_nek_CMHpval <- numeric()
human_nek_CMHcommonOR <- numeric()
human_nek_FisherPval <- numeric()
human_nek_FisherOR <- numeric()
human_nek_FisherCI <- character()
for (i in 1:length(human_idx_cols)) {
  humanStratContingencyArray <- getStratContingencyArray(df = subset(human_data,target_nek=="nek target"),
                                                    sequence_col = "Sequence",
                                                    diso_index_col = human_idx_cols[i],
                                                    psites_col = "psite_nek"
                                                    ) 
  human_nek_CMHtest <- mantelhaen.test(humanStratContingencyArray)
  human_nek_CMHpval[i]<-human_nek_CMHtest$p.value
  human_nek_CMHcommonOR[i]<-human_nek_CMHtest$estimate
  
  humanContingency <- matrix(c(sum(humanStratContingencyArray[1,1,]),
                               sum(humanStratContingencyArray[1,2,]),
                               sum(humanStratContingencyArray[2,1,]),
                               sum(humanStratContingencyArray[2,2,])),
                             nrow = 2,byrow = T)
  human_nek_FisherTest <- fisher.test(humanContingency)
  human_nek_FisherPval[i]<-human_nek_FisherTest$p.value
  human_nek_FisherOR[i]<-human_nek_FisherTest$estimate
  human_nek_FisherCI[[i]]<-paste(round(human_nek_FisherTest$conf.int,4), collapse =  " - ")
}
#__________________________________________


xenopus_idx_cols <- c("iupl_disoIndexes","spot_disoIndexes","vsl_disoIndexes" )
xenopus_organism <- (rep("Xenopus dynamic",length(xenopus_idx_cols)))
xenopus_proteome_length <- sum(xenopusDiso$Length)
xenopus_diso_fraction <- numeric()
xenopus_CMHpval <- numeric()
xenopus_CMHcommonOR <- numeric()
xenopus_FisherPval <- numeric()
xenopus_FisherOR <- numeric()
xenopus_FisherCI <- character()
for (i in 1:length(xenopus_idx_cols)) {
  xenopusStratContingencyArray <- getStratContingencyArray(df = subset(xenopus_data,ANOVA=="Dynamic"),
                                                    sequence_col = "Sequence",
                                                    diso_index_col = xenopus_idx_cols[i],
                                                    psites_col = "anova_psites"
                                                    ) 
  xenopus_CMHtest <- mantelhaen.test(xenopusStratContingencyArray)
  xenopus_CMHpval[i]<-xenopus_CMHtest$p.value
  xenopus_CMHcommonOR[i]<-xenopus_CMHtest$estimate
  
  
  xenopusContingency <- matrix(c(sum(xenopusStratContingencyArray[1,1,]),
                               sum(xenopusStratContingencyArray[1,2,]),
                               sum(xenopusStratContingencyArray[2,1,]),
                               sum(xenopusStratContingencyArray[2,2,])),
                             nrow = 2,byrow = T)
  xenopus_FisherTest <- fisher.test(xenopusContingency)
  xenopus_FisherPval[i]<-xenopus_FisherTest$p.value
  xenopus_FisherOR[i]<-xenopus_FisherTest$estimate
  xenopus_FisherCI[[i]]<-paste(round(xenopus_FisherTest$conf.int,4), collapse =  " - ")
  xenopus_diso_fraction[i]<-(sum(lengths(strsplit(subset(xenopusDiso,!is.na(xenopusDiso[,xenopus_idx_cols[i]]))[,xenopus_idx_cols[i]],","))))/xenopus_proteome_length
}


#FISHER
yeastIntersect_Fisher_data <- cbind(predictors,yeastIntersect_diso_fraction,yeastIntersect_FisherPval,yeastIntersect_FisherOR,yeastIntersect_FisherCI,yeastIntersect_organism)
human_Fisher_data <- cbind(predictors,human_diso_fraction,human_FisherPval,human_FisherOR,human_FisherCI,human_organism)
human_MAPK_Fisher_data <- cbind(predictors,human_diso_fraction,human_mapk_FisherPval,human_mapk_FisherOR,human_mapk_FisherCI,human_mapk_organism)
human_AURK_Fisher_data <- cbind(predictors,human_diso_fraction,human_aurk_FisherPval,human_aurk_FisherOR,human_aurk_FisherCI,human_aurk_organism)
human_PLK_Fisher_data <- cbind(predictors,human_diso_fraction,human_plk_FisherPval,human_plk_FisherOR,human_plk_FisherCI,human_plk_organism)
human_NEK_Fisher_data <- cbind(predictors,human_diso_fraction,human_nek_FisherPval,human_nek_FisherOR,human_nek_FisherCI,human_nek_organism)
human_DYRK_Fisher_data <- cbind(predictors,human_diso_fraction,human_dyrk_FisherPval,human_dyrk_FisherOR,human_dyrk_FisherCI,human_dyrk_organism)
xenopus_Fisher_data <- cbind(predictors,xenopus_diso_fraction,xenopus_FisherPval,xenopus_FisherOR,xenopus_FisherCI,xenopus_organism)

predictorsFisher_data<-as.data.frame(rbind(yeastIntersect_Fisher_data,
                                           human_Fisher_data,
                                           human_MAPK_Fisher_data,
                                           human_AURK_Fisher_data,
                                           human_PLK_Fisher_data,
                                           human_NEK_Fisher_data,
                                           human_DYRK_Fisher_data,
                                           xenopus_Fisher_data))
colnames(predictorsFisher_data) <- c("predictors","diso_fraction","pval","OR","conf.int","organism")
predictorsFisher_data$diso_fraction <- as.numeric(as.character(predictorsFisher_data$diso_fraction))
predictorsFisher_data$pval <- as.numeric(as.character(predictorsFisher_data$pval))
predictorsFisher_data$OR <- as.numeric(as.character(predictorsFisher_data$OR))
predictorsFisher_data$organism <- as.factor(predictorsFisher_data$organism)
predictorsFisher_data$adjPval <- p.adjust(predictorsFisher_data$pval,method = "BH")




#________________________________
# Lollipop plots for CDK targets
subset(predictorsFisher_data,grepl(pattern = "CDK|dynamic",organism)) %>%  
    mutate(organism=factor(organism,c("Yeast CDK targets","Human CDK targets","Xenopus dynamic"))) %>% 
    ggplot(aes(x=organism, y=OR)) +
    geom_segment( aes(xend=organism, yend=0)) +
    geom_point(aes(color=diso_fraction),size=12) +
    theme_linedraw(base_size = 22)+
    theme(panel.grid.major.y = element_blank())+
    scale_y_continuous(limits = c(0,57),minor_breaks = c(0:57),breaks = seq(0,57,5),expand = c(0,0)) +
    coord_flip() +
    facet_wrap(~predictors,ncol=3)+
    scale_color_gradient(low = pal_jco()(10)[1],high = pal_jco()(10)[4],limits=c(0.19,0.51),breaks=c(0.2,0.3,0.4,0.5)) +
    xlab("")


```

### e

#### Yeast

```{r Holt disorder percentage,echo=FALSE,message=FALSE,warning=FALSE,fig.height=4,fig.width=7}

##### Violion Plot version###########
yeastIntersect_disoPerc_cols_idx <-grep("_perc",colnames(yeastIntersect_All_data))
yeastIntersect_reshaped_disoPerc <- melt(yeastIntersect_All_data[,yeastIntersect_disoPerc_cols_idx])
colnames(yeastIntersect_reshaped_disoPerc) <- c("Predictor","Percentage of disorder")
yeastIntersect_reshaped_disoPerc$target <- rep(yeastIntersect_All_data$target_intersect,length(yeastIntersect_disoPerc_cols_idx))
yeastIntersect_reshaped_disoPerc$target<- relevel(as.factor(yeastIntersect_reshaped_disoPerc$target),ref="Non Cdk1 target")
yeastIntersect_reshaped_disoPerc$Predictor <- factor(as.character(yeastIntersect_reshaped_disoPerc$Predictor),levels =  c("pfilt_perc","seg_perc","dis465_perc","disHL_perc","espD_perc","espN_perc","espX_perc","glo_perc","iupl_perc","iups_perc","jronn_perc","vsl_perc","spot_perc"))

# Main Figure 4E
yeastIntersect_reshaped_disoPerc_figure <-  subset(yeastIntersect_reshaped_disoPerc,(Predictor=="iupl_perc")|(Predictor=="vsl_perc")|(Predictor=="spot_perc"))

yeastIntersect_reshaped_disoPerc_figure$Predictor <- 
  factor(yeastIntersect_reshaped_disoPerc_figure$Predictor, c("iupl_perc","spot_perc","vsl_perc"))

ggplot(yeastIntersect_reshaped_disoPerc_figure,aes(x=Predictor, y=`Percentage of disorder`)) +
geom_violin(aes(fill=target,color=target),position=position_dodge(0.9),bw=10,alpha=1,trim = F) +
geom_boxplot(aes(color=target),position=position_dodge(0.9),width=0.4,outlier.size = 0.0001) +
ggpubr::theme_classic2() +
theme(text = element_text(size=20),legend.position = "bottom",legend.title = element_blank(),axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed")) +
scale_y_continuous(limits = c(-20, 135), breaks = c(seq(0, 100, by = 20))) +
scale_x_discrete(labels = c("IUPred", "SPOT","VSL2b")) + xlab(NULL) +
geom_segment(aes(x = 0.75, y = 133, xend = 1.25, yend = 133)) + annotate(geom="text", x=1, y=135, label="***") +
geom_segment(aes(x = 1.75, y = 133, xend = 2.25, yend = 133)) + annotate(geom="text", x=2, y=135, label="***") +
geom_segment(aes(x = 2.75, y = 133, xend = 3.25, yend = 133)) + annotate(geom="text", x=3, y=135, label="***") +
scale_fill_manual(values = pal_jco()(10)[c(5,2)]) +
scale_color_manual(values = pal_jco()(10)[c(10,7)],guide=FALSE)


yeastIntersect_reshaped_disoPerc_figure$target <- relevel(yeastIntersect_reshaped_disoPerc_figure$target,ref="Cdk1 target")
wilcox.test(`Percentage of disorder`~target,data=yeastIntersect_reshaped_disoPerc_figure,subset = Predictor=="iupl_perc",conf.int=TRUE)
wilcox.test(`Percentage of disorder`~target,data=yeastIntersect_reshaped_disoPerc_figure,subset = Predictor=="spot_perc",conf.int=TRUE)
wilcox.test(`Percentage of disorder`~target,data=yeastIntersect_reshaped_disoPerc_figure,subset = Predictor=="vsl_perc",conf.int=TRUE)


####################################
```

#### Human

```{r human disorder percentage,echo=FALSE,message=FALSE,warning=FALSE,fig.height=4,fig.width=7}

human_disoPerc_cols_idx <-grep("_perc",colnames(human_data))
human_reshaped_disoPerc <- melt(human_data[,human_disoPerc_cols_idx])
colnames(human_reshaped_disoPerc) <- c("Predictor","Percentage of disorder")
human_reshaped_disoPerc$target <- rep(human_data$target,length(human_disoPerc_cols_idx))
human_reshaped_disoPerc$target<- relevel(as.factor(human_reshaped_disoPerc$target),ref="Non Cdk1 target")
human_reshaped_disoPerc$Predictor <- factor(as.character(human_reshaped_disoPerc$Predictor),levels = c("pfilt_perc","seg_perc","dis465_perc","disHL_perc","espD_perc","espN_perc","espX_perc","glo_perc","iupl_perc","iups_perc","jronn_perc","vsl_perc","spot_perc"))

human_reshaped_disoPerc_figure <-  subset(human_reshaped_disoPerc,(Predictor=="iupl_perc")|(Predictor=="vsl_perc")|(Predictor=="spot_perc"))

human_reshaped_disoPerc_figure$Predictor <- 
  factor(human_reshaped_disoPerc_figure$Predictor, c("iupl_perc","spot_perc","vsl_perc"))

ggplot(human_reshaped_disoPerc_figure,aes(x=Predictor, y=`Percentage of disorder`)) +
geom_violin(aes(fill=target),position=position_dodge(0.9),bw=10,alpha=1,trim = F) +
geom_boxplot(aes(color=target),position=position_dodge(0.9),width=0.4,outlier.shape = NA) +
ggpubr::theme_classic2() +
theme(text = element_text(size=20),legend.position = "bottom",legend.title = element_blank(),axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed")) +
scale_y_continuous(limits = c(-20, 135), breaks = c(seq(0, 100, by = 20))) +
scale_x_discrete(labels = c("IUPred", "SPOT","VSL2b")) + xlab(NULL) +
geom_segment(aes(x = 0.75, y = 133, xend = 1.25, yend = 133)) + annotate(geom="text", x=1, y=135, label="***") +
geom_segment(aes(x = 1.75, y = 133, xend = 2.25, yend = 133)) + annotate(geom="text", x=2, y=135, label="***") +
geom_segment(aes(x = 2.75, y = 133, xend = 3.25, yend = 133)) + annotate(geom="text", x=3, y=135, label="***") +
scale_fill_manual(values = pal_jco()(10)[c(5,2)]) +
scale_color_manual(values = pal_jco()(10)[c(10,7)],guide=FALSE)

human_reshaped_disoPerc_figure$target <- relevel(human_reshaped_disoPerc_figure$target,ref="Cdk1 target")
wilcox.test(`Percentage of disorder`~target,data=human_reshaped_disoPerc_figure,subset = Predictor=="iupl_perc",conf.int=TRUE)
wilcox.test(`Percentage of disorder`~target,data=human_reshaped_disoPerc_figure,subset = Predictor=="spot_perc",conf.int=TRUE)
wilcox.test(`Percentage of disorder`~target,data=human_reshaped_disoPerc_figure,subset = Predictor=="vsl_perc",conf.int=TRUE)
```

#### Xenopus

```{r xenopus disorder percentage,echo=FALSE,message=FALSE,warning=FALSE,fig.height=4,fig.width=7}

xenopus_disoPerc_cols_idx <-grep("_perc",colnames(xenopus_data))
xenopus_reshaped_disoPerc <- melt(xenopus_data[,xenopus_disoPerc_cols_idx])
colnames(xenopus_reshaped_disoPerc) <- c("Predictor","Percentage of disorder")
xenopus_reshaped_disoPerc$ANOVA <- rep(xenopus_data$ANOVA,length(xenopus_disoPerc_cols_idx))
xenopus_reshaped_disoPerc$ANOVA<- relevel(as.factor(xenopus_reshaped_disoPerc$ANOVA),ref="Non Dynamic")
xenopus_reshaped_disoPerc$hCDKtarget <- rep(xenopus_data$hCDKtarget,length(xenopus_disoPerc_cols_idx))
xenopus_reshaped_disoPerc <- xenopus_reshaped_disoPerc %>% mutate(hCDKtarget=case_when(
    hCDKtarget  ~ "Human CDK target",
    TRUE ~ "Other"
))
xenopus_reshaped_disoPerc$Predictor <- factor(xenopus_reshaped_disoPerc$Predictor,levels = ordered(levels(xenopus_reshaped_disoPerc$Predictor)))

# ggplot(xenopus_reshaped_disoPerc,aes(x=Predictor, y=`Percentage of disorder`)) +
#     geom_violin(aes(fill=cluster),position=position_dodge(0.9),bw=5,alpha=0.5,trim = F) +
#     geom_boxplot(aes(fill=cluster),position=position_dodge(0.9),width=0.05,outlier.size = 0.0001) +
#     ggpubr::theme_pubclean() + 
#     theme(text = element_text(size=20),axis.text.x = element_text(angle=45),legend.position = "none") + 
#     scale_y_continuous(limits = c(-20, 120), breaks = c(seq(0, 100, by = 10)))

# Main Figure 4E
xenopus_reshaped_disoPerc_figure <-  subset(xenopus_reshaped_disoPerc,(Predictor=="iupl_perc")|(Predictor=="vsl_perc")|(Predictor=="spot_perc"))

xenopus_reshaped_disoPerc_figure$Predictor <- 
  factor(xenopus_reshaped_disoPerc_figure$Predictor, c("iupl_perc","spot_perc","vsl_perc"))

ggplot(xenopus_reshaped_disoPerc_figure,aes(x=Predictor, y=`Percentage of disorder`)) +
geom_violin(aes(fill=ANOVA,color=ANOVA),position=position_dodge(0.9),bw=10,alpha=1,trim = F) +
geom_boxplot(aes(color=ANOVA),position=position_dodge(0.9),width=0.4,outlier.size = 0.0001) +
ggpubr::theme_classic2() +
theme(text = element_text(size=20),legend.position = "bottom",legend.title = element_blank(),axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed")) +
scale_y_continuous(limits = c(-20, 135), breaks = c(seq(0, 100, by = 20))) +
scale_x_discrete(labels = c("IUPred", "SPOT","VSL2b")) + xlab(NULL) +
geom_segment(aes(x = 0.75, y = 133, xend = 1.25, yend = 133)) + annotate(geom="text", x=1, y=135, label="***") +
geom_segment(aes(x = 1.75, y = 133, xend = 2.25, yend = 133)) + annotate(geom="text", x=2, y=135, label="***") +
geom_segment(aes(x = 2.75, y = 133, xend = 3.25, yend = 133)) + annotate(geom="text", x=3, y=135, label="***") +
scale_fill_manual(values = pal_jco()(10)[c(5,2)]) +
scale_color_manual(values = pal_jco()(10)[c(10,7)],guide=FALSE)

xenopus_reshaped_disoPerc$ANOVA<- relevel(xenopus_reshaped_disoPerc$ANOVA,ref="Dynamic")
wilcox.test(`Percentage of disorder`~ANOVA,data=xenopus_reshaped_disoPerc_figure,subset = Predictor=="iupl_perc",conf.int=TRUE)
wilcox.test(`Percentage of disorder`~ANOVA,data=xenopus_reshaped_disoPerc_figure,subset = Predictor=="spot_perc",conf.int=TRUE)
wilcox.test(`Percentage of disorder`~ANOVA,data=xenopus_reshaped_disoPerc_figure,subset = Predictor=="vsl_perc",conf.int=TRUE)

```

### f

```{r disorder percentages for other kinases,echo=FALSE,message=FALSE,warning=FALSE,fig.height=4,fig.width=15}

#This data is available only for Human

phosphoProteome_human_disoPerc <- subset(human_data,target=="Non Cdk1 target" &
                                         target_mapk=="Non mapk target" &
                                         target_aurk=="Non aurk target" &
                                         target_plk=="Non plk target" &
                                         target_nek=="Non nek target" &
                                         target_dyrk=="Non dyrk target")[,c("iupl_perc","spot_perc","vsl_perc")]
human_disoPerc_data <- cbind(phosphoProteome_human_disoPerc,subset=rep("Phosphoproteome",nrow(phosphoProteome_human_disoPerc)))

cdk_human_disoPerc <- subset(human_data,target=="Cdk1 target")[,c("iupl_perc","spot_perc","vsl_perc")]
human_disoPerc_data <- rbind(human_disoPerc_data,cbind(cdk_human_disoPerc,subset=rep("CDK",nrow(cdk_human_disoPerc))))

mapk_human_disoPerc <- subset(human_data,target_mapk=="mapk target")[,c("iupl_perc","spot_perc","vsl_perc")]
human_disoPerc_data <- rbind(human_disoPerc_data,cbind(mapk_human_disoPerc,subset=rep("MAPK",nrow(mapk_human_disoPerc))))

aurk_human_disoPerc <- subset(human_data,target_aurk=="aurk target")[,c("iupl_perc","spot_perc","vsl_perc")]
human_disoPerc_data <- rbind(human_disoPerc_data,cbind(aurk_human_disoPerc,subset=rep("AURK",nrow(aurk_human_disoPerc))))

plk_human_disoPerc <- subset(human_data,target_plk=="plk target")[,c("iupl_perc","spot_perc","vsl_perc")]
human_disoPerc_data <- rbind(human_disoPerc_data,cbind(plk_human_disoPerc,subset=rep("PLK",nrow(plk_human_disoPerc))))

nek_human_disoPerc <- subset(human_data,target_nek=="nek target")[,c("iupl_perc","spot_perc","vsl_perc")]
human_disoPerc_data <- rbind(human_disoPerc_data,cbind(nek_human_disoPerc,subset=rep("NEK",nrow(nek_human_disoPerc))))

dyrk_human_disoPerc <- subset(human_data,target_dyrk=="dyrk target")[,c("iupl_perc","spot_perc","vsl_perc")]
human_disoPerc_data <- rbind(human_disoPerc_data,cbind(dyrk_human_disoPerc,subset=rep("DYRK",nrow(dyrk_human_disoPerc))))

human_disoPerc_data$subset <- forcats::as_factor(human_disoPerc_data$subset)

human_disoPerc_data <- melt(human_disoPerc_data,id.vars = "subset",variable.name = "Predictor",value.name = "Percentage of disorder")



ggplot(human_disoPerc_data,aes(x=Predictor, y=`Percentage of disorder`)) +
geom_violin(aes(fill=subset),position=position_dodge(0.8),trim = F,adjust = 1) +
geom_boxplot(aes(color=subset),position=position_dodge(0.8),width=0.15,outlier.shape = NA) +
ggpubr::theme_classic2() +
theme(text = element_text(size=20),legend.position = "bottom",legend.title = element_blank(),axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed")) +
scale_y_continuous(limits = c(-20, 135), breaks = c(seq(0, 100, by = 20))) +
scale_x_discrete(labels = c("IUPred", "SPOT","VSL2b")) + xlab(NULL) +
scale_fill_manual(values = c("#ffdd15ff",pal_tron()(7)[c(3,4,1,2,5,6)])) + guides(fill = guide_legend(nrow = 1)) +
scale_color_manual(values = c("#ffdd15ff",pal_tron()(7)[c(3,4,1,2,5,6)]),guide=FALSE)
```

```{r post-hoc test for all kinases disorder percentage,echo=FALSE,message=FALSE,warning=FALSE,fig.height=6,fig.width=22}

#__________________________
# Tests Kruskal-willis dunn post-hoc
library(FSA)

dunnnMat <- matrix(ncol = 7,nrow = 7)
colnames(dunnnMat) <- c("Phosphoproteome","CDK","MAPK","AURK","PLK","NEK","DYRK")
rownames(dunnnMat) <- c("Phosphoproteome","CDK","MAPK","AURK","PLK","NEK","DYRK")
#---------------------Helper functions---------------------------
# Get lower triangle of the correlation matrix
  get_lower_tri<-function(cormat){
    cormat[upper.tri(cormat)] <- NA
    return(cormat)
  }
  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
  }

reorder_mat <- function(cormat){
# Use correlation between variables as distance
dd <- as.dist((1-cormat)/2)
hc <- hclust(dd)
cormat <-cormat[hc$order, hc$order]
}

#--------------------------------------------------------------

#IUPred
Summarize(`Percentage of disorder` ~ subset,
data=subset(human_disoPerc_data,Predictor=="iupl_perc"),
digits=3)
iupl_model=lm(`Percentage of disorder` ~ subset,
data=subset(human_disoPerc_data,Predictor=="iupl_perc"))
kruskal.test(`Percentage of disorder` ~ subset, data = subset(human_disoPerc_data,Predictor=="iupl_perc"))
iupl_PT = dunnTest(`Percentage of disorder` ~ subset, data = subset(human_disoPerc_data,Predictor=="iupl_perc"),method="bh")
iupl_PT_res <- iupl_PT$res
iupl_dunnnMat <- dunnnMat
for (n in 1:nrow(iupl_PT_res)) {
  comp_vec <- str_split(iupl_PT_res[n,"Comparison"],pattern = " - ")[[1]]
  iupl_dunnnMat[comp_vec[1],comp_vec[2]] <- iupl_PT_res[n,"P.adj"]
  iupl_dunnnMat[comp_vec[2],comp_vec[1]] <- iupl_PT_res[n,"P.adj"]
}
iupl_dunnnMat[is.na(iupl_dunnnMat)] <- 1


# iupl_dunnnMat_ord <- reorder_mat(iupl_dunnnMat)
iupl_upper_tri <- get_upper_tri(iupl_dunnnMat)
# Melt the correlation matrix
melted_iupl_dunnnMat <- melt(iupl_upper_tri, na.rm = TRUE)
melted_iupl_dunnnMat$value_log <- -log10(melted_iupl_dunnnMat$value)
melted_iupl_dunnnMat <- melted_iupl_dunnnMat %>% mutate(value_log=replace(value_log, value_log>10, 10)) 
melted_iupl_dunnnMat$value<- prettyNum(melted_iupl_dunnnMat$value, format = "e", digits = 1)
# Create a ggheatmap
ggplot(melted_iupl_dunnnMat, aes(Var2, Var1, fill = value_log))+
    geom_tile(color = "white")+
    geom_text(aes(Var2, Var1, label = value), color = "white", size = 4,fontface = "bold") +
    scale_fill_gradient(low = pal_jco()(10)[4],high = pal_jco()(10)[1], 
                        limit = c(-log10(0.05),10), space = "Lab", 
                        name="P-value",
                        breaks=c(-log10(0.05),5,10),labels=c("0.05","10e-5","<10e-10")) +
    theme_classic(base_size = 15,)+ # minimal theme
    theme(axis.text.x = element_text(angle = 90, vjust = 1,hjust = 1),
          axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.line = element_blank(),
          panel.border = element_rect(colour = "black", fill=NA, size=1))+
    coord_fixed()+
    guides(fill = guide_colorbar(title.position = "top", title.hjust = 0.5))

#SPOT
Summarize(`Percentage of disorder` ~ subset,
data=subset(human_disoPerc_data,Predictor=="spot_perc"),
digits=3)
spot_model=lm(`Percentage of disorder` ~ subset,
data=subset(human_disoPerc_data,Predictor=="spot_perc"))
kruskal.test(`Percentage of disorder` ~ subset, data = subset(human_disoPerc_data,Predictor=="spot_perc"))
spot_PT = dunnTest(`Percentage of disorder` ~ subset, data = subset(human_disoPerc_data,Predictor=="spot_perc"),method="bh")
spot_PT_res <- spot_PT$res
spot_dunnnMat <- dunnnMat
for (n in 1:nrow(spot_PT_res)) {
  comp_vec <- str_split(spot_PT_res[n,"Comparison"],pattern = " - ")[[1]]
  spot_dunnnMat[comp_vec[1],comp_vec[2]] <- spot_PT_res[n,"P.adj"]
  spot_dunnnMat[comp_vec[2],comp_vec[1]] <- spot_PT_res[n,"P.adj"]
}
spot_dunnnMat[is.na(spot_dunnnMat)] <- 1


# spot_dunnnMat_ord <- reorder_mat(spot_dunnnMat)
spot_upper_tri <- get_upper_tri(spot_dunnnMat)
# Melt the correlation matrix
melted_spot_dunnnMat <- melt(spot_upper_tri, na.rm = TRUE)
melted_spot_dunnnMat$value_log <- -log10(melted_spot_dunnnMat$value)
melted_spot_dunnnMat <- melted_spot_dunnnMat %>% mutate(value_log=replace(value_log, value_log>10, 10)) 
melted_spot_dunnnMat$value<- prettyNum(melted_spot_dunnnMat$value, format = "e", digits = 1)
# Create a ggheatmap
ggplot(melted_spot_dunnnMat, aes(Var2, Var1, fill = value_log))+
    geom_tile(color = "white")+
    geom_text(aes(Var2, Var1, label = value), color = "white", size = 4,fontface = "bold") +
    scale_fill_gradient(low = pal_jco()(10)[4],high = pal_jco()(10)[1], 
                        limit = c(-log10(0.05),10), space = "Lab", 
                        name="P-value",
                        breaks=c(-log10(0.05),5,10),labels=c("0.05","10e-5","<10e-10")) +
    theme_classic(base_size = 15,)+ # minimal theme
    theme(axis.text.x = element_text(angle = 90, vjust = 1,hjust = 1),
          axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.line = element_blank(),
          panel.border = element_rect(colour = "black", fill=NA, size=1))+
    coord_fixed()+
    guides(fill = guide_colorbar(title.position = "top", title.hjust = 0.5))

#VSL2B
Summarize(`Percentage of disorder` ~ subset,
data=subset(human_disoPerc_data,Predictor=="vsl_perc"),
digits=3)
vsl_model=lm(`Percentage of disorder` ~ subset,
data=subset(human_disoPerc_data,Predictor=="vsl_perc"))
kruskal.test(`Percentage of disorder` ~ subset, data = subset(human_disoPerc_data,Predictor=="vsl_perc"))
vsl_PT = dunnTest(`Percentage of disorder` ~ subset, data = subset(human_disoPerc_data,Predictor=="vsl_perc"),method="bh")
vsl_PT_res <- vsl_PT$res
vsl_dunnnMat <- dunnnMat
for (n in 1:nrow(vsl_PT_res)) {
  comp_vec <- str_split(vsl_PT_res[n,"Comparison"],pattern = " - ")[[1]]
  vsl_dunnnMat[comp_vec[1],comp_vec[2]] <- vsl_PT_res[n,"P.adj"]
  vsl_dunnnMat[comp_vec[2],comp_vec[1]] <- vsl_PT_res[n,"P.adj"]
}
vsl_dunnnMat[is.na(vsl_dunnnMat)] <- 1


# vsl_dunnnMat_ord <- reorder_mat(vsl_dunnnMat)
vsl_upper_tri <- get_upper_tri(vsl_dunnnMat)
# Melt the correlation matrix
melted_vsl_dunnnMat <- melt(vsl_upper_tri, na.rm = TRUE)
melted_vsl_dunnnMat$value_log <- -log10(melted_vsl_dunnnMat$value)
melted_vsl_dunnnMat <- melted_vsl_dunnnMat %>% mutate(value_log=replace(value_log, value_log>10, 10)) 
melted_vsl_dunnnMat$value<- prettyNum(melted_vsl_dunnnMat$value, format = "e", digits = 1)
# Create a ggheatmap
ggplot(melted_vsl_dunnnMat, aes(Var2, Var1, fill = value_log))+
    geom_tile(color = "white")+
    geom_text(aes(Var2, Var1, label = value), color = "white", size = 4,fontface = "bold") +
    scale_fill_gradient(low = pal_jco()(10)[4],high = pal_jco()(10)[1], 
                        limit = c(-log10(0.05),10), space = "Lab", 
                        name="P-value",
                        breaks=c(-log10(0.05),5,10),labels=c("0.05","10e-5","<10e-10")) +
    theme_classic(base_size = 15,)+ # minimal theme
    theme(axis.text.x = element_text(angle = 90, vjust = 1,hjust = 1),
          axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.line = element_blank(),
          panel.border = element_rect(colour = "black", fill=NA, size=1))+
    coord_fixed()+
    guides(fill = guide_colorbar(title.position = "top", title.hjust = 0.5))


```

## Fig. 4

### a

```{r CDK targets and Dynamic phosphoproteins in MLOs,echo=FALSE,message=FALSE,warning=FALSE,fig.height=4,fig.width=10}
# 
redundant_MLO_lilaDB <-  read_delim("MLOs_data/MLO_lilaDB_redundant_20052021.tab", "\t", escape_double = FALSE, trim_ws = TRUE)
# 
MLO_lilaDB_nr <- redundant_MLO_lilaDB %>% group_by(`ACC#`) %>% summarise_at("body",function(x){paste(unique(x),collapse = ",")})
# 
# 
# 
dynamic_symbol_to_HsUniprot <- read_delim("phosphorylation_data/Valverde_data/ids/dynamic_symbol_to_HsUniprot.tab", "\t", escape_double = FALSE, trim_ws = TRUE)
# 
MLO_lilaDB_nr <- MLO_lilaDB_nr %>% mutate(xenopus=case_when(
    `ACC#` %in% dynamic_symbol_to_HsUniprot$`Human Uniprot` ~ "Dynamic",
    TRUE ~ "Non Dynamic"
))

cdkTargets_uniprot <- read_lines("phosphorylation_data/Valverde_data/ids/humanCDKtargets_uniprot.txt",skip = 1)
MLO_lilaDB_nr <- MLO_lilaDB_nr %>% mutate(human=case_when(
  `ACC#` %in% cdkTargets_uniprot ~ "Cdk target",
  TRUE ~ "Non Cdk target"
))

MLO_lilaDB_nr_curated <- read_delim("MLOs_data/MLO_lilaDB_nr_260521_curated.tab", "\t", escape_double = FALSE, trim_ws = TRUE)

humanCDKtarget_MLOS <- subset(MLO_lilaDB_nr_curated, human=="Cdk target")
humanCDKtarget_MLOS$dominant_body <- as.factor(humanCDKtarget_MLOS$dominant_body)

xenopusDynamic_MLOS <- subset(MLO_lilaDB_nr_curated, xenopus=="Dynamic" )
xenopusDynamic_MLOS$dominant_body <- as.factor(xenopusDynamic_MLOS$dominant_body)

#_____________ barplots for MLOS _______________


dynamic_symbol_to_HsUniprot <- read_delim("phosphorylation_data/Valverde_data/ids/dynamic_symbol_to_HsUniprot.tab",delim = "\t", escape_double = FALSE, trim_ws = TRUE)

xenopusDynamic_MLOS <- merge(dynamic_symbol_to_HsUniprot,xenopusDynamic_MLOS,by.x = "Human Uniprot",by.y = "ACC#",all.x = T)
xenopusDynamic_MLOS$dominant_body <- as.character(xenopusDynamic_MLOS$dominant_body)
xenopusDynamic_MLOS$dominant_body[is.na(xenopusDynamic_MLOS$dominant_body)] <- "Not assigned"

xenopus_mlo_count <- as.data.frame(table(xenopusDynamic_MLOS$dominant_body))
colnames(xenopus_mlo_count)[1] <- "MLO"

humanCdkTargets <- read_delim("phosphorylation_data/phosphositePlus_data/human_data_curated_V2_cleaned.tab","\t", escape_double = FALSE, col_types = cols(MOD_RSD = col_character()), trim_ws = TRUE)

humanCDKtarget_MLOS <- merge(humanCdkTargets,humanCDKtarget_MLOS,by.x = "ACC#",by.y = "ACC#",all.x = T)

humanCDKtarget_MLOS$dominant_body <- as.character(humanCDKtarget_MLOS$dominant_body)
humanCDKtarget_MLOS$dominant_body[is.na(humanCDKtarget_MLOS$dominant_body)] <- "Not assigned"

human_mlo_count <- as.data.frame(table(humanCDKtarget_MLOS$dominant_body))
colnames(human_mlo_count)[1] <- "MLO"


intersection_MLOS <- merge(xenopusDynamic_MLOS,humanCDKtarget_MLOS,by.x ="Human Uniprot" ,by.y = "ACC#")

intersection_mlo_count <- as.data.frame(table(intersection_MLOS$dominant_body.x))
colnames(intersection_mlo_count)[1] <- "MLO"



plot_table <- as.data.frame(rbind(xenopus_mlo_count,human_mlo_count,intersection_mlo_count))

plot_table$MLO <- factor(plot_table$MLO,levels = c("Not assigned","P-bodies","PML bodies","Cajal bodies","53BP1 bodies","Superenhancers-Mediator","NPC","Speckles","Stress granules","Nucleoli"))
plot_table$dataset <- factor(c(rep("Xenopus",10),rep("Human",10),rep("Intersection",9)),levels=c("Human","Xenopus","Intersection"))


ggplot(plot_table, aes(fill=MLO, y=Freq, x=dataset)) + 
  geom_bar(position="fill", stat="identity",width = 0.5,color="black",size=0.1) + 
  theme_classic() +
    scale_fill_manual(values = c("white",pal_tron()(7)[c(4,1,3,2,5)],"aquamarine4","brown2","khaki2","dodgerblue3")) + coord_flip() 
    
```

### b

#### Human

```{r IUpred scores - human ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=4,fig.width=20}
IupredHuman <- IUpredScoresPlotGenerator(subset(human_data, PROTEIN %in% c("coilin","NPM1","KI-67","NUP35","NUP98","53BP1","PML")),id_col = "PROTEIN",sites_col = "psite",subset_sites_col = "psite_CDK1",sequence_col = "Sequence")

 for (plot in IupredHuman) {
   print(plot)
 }

```

#### Xenopus

```{r IUpred scores - xenopus ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=4,fig.width=20}
IupredXenopus <- IUpredScoresPlotGenerator(subset(all_predictions_phospho,ID %in% c("coil","npm","ki67","nup53","nup98","tp53b")),sites_col = "psites")

 for (plot in IupredXenopus) {
   print(plot)
 }

```

## Fig. 6

### b

```{r ki67 microscopy ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=3,fig.width=6}

#==========================================H3K9me3=========================================================#

# Low Ki-67 expression
DAPI_for_H3K9me3_low <- read_csv("ki67_microscopy_data/Krasinska_data/DAPI for H3K9me3 low.csv")
DAPI_for_H3K9me3_low$channel <- rep("DAPI",nrow(DAPI_for_H3K9me3_low))
DAPI_for_H3K9me3_low$scaled_intensity <- rescale(DAPI_for_H3K9me3_low$Y)
GFP_for_H3K9me3_low <- read_csv("ki67_microscopy_data/Krasinska_data/GFP for H3K9me3 low.csv")
GFP_for_H3K9me3_low$channel <- rep("GFP-Ki-67",nrow(GFP_for_H3K9me3_low))
GFP_for_H3K9me3_low$scaled_intensity <- rescale(GFP_for_H3K9me3_low$Y)
RED_for_H3K9me3_low <- read_csv("ki67_microscopy_data/Krasinska_data/red for H3K9me3 low.csv")
RED_for_H3K9me3_low$channel <- rep("H3K9me3",nrow(RED_for_H3K9me3_low))
RED_for_H3K9me3_low$scaled_intensity <- rescale(RED_for_H3K9me3_low$Y)

merge_for_H3K9me3_low <- rbind(DAPI_for_H3K9me3_low,GFP_for_H3K9me3_low,RED_for_H3K9me3_low)
colnames(merge_for_H3K9me3_low)[1:2] <- c("distance","intensity")


# High Ki-67 expression
DAPI_for_H3K9me3_high <- read_csv("ki67_microscopy_data/Krasinska_data/DAPI for H3K9me3 high.csv")
DAPI_for_H3K9me3_high$channel <- rep("DAPI",nrow(DAPI_for_H3K9me3_high))
DAPI_for_H3K9me3_high$scaled_intensity <- rescale(DAPI_for_H3K9me3_high$Y)
GFP_for_H3K9me3_high <- read_csv("ki67_microscopy_data/Krasinska_data/GFP for H3K9me3 high.csv")
GFP_for_H3K9me3_high$channel <- rep("GFP-Ki-67",nrow(GFP_for_H3K9me3_high))
GFP_for_H3K9me3_high$scaled_intensity <- rescale(GFP_for_H3K9me3_high$Y)
RED_for_H3K9me3_high <- read_csv("ki67_microscopy_data/Krasinska_data/red for H3K9me3 high.csv")
RED_for_H3K9me3_high$channel <- rep("H3K9me3",nrow(RED_for_H3K9me3_high))
RED_for_H3K9me3_high$scaled_intensity <- rescale(RED_for_H3K9me3_high$Y)

merge_for_H3K9me3_high <- rbind(DAPI_for_H3K9me3_high,GFP_for_H3K9me3_high,RED_for_H3K9me3_high)
colnames(merge_for_H3K9me3_high)[1:2] <- c("distance","intensity")

#==========================================MeCP2=========================================================#

# Low Ki-67 expression
DAPI_for_MeCP2_low <- read_csv("ki67_microscopy_data/Krasinska_data/DAPI for MeCP2 low.csv")
DAPI_for_MeCP2_low$channel <- rep("DAPI",nrow(DAPI_for_MeCP2_low))
DAPI_for_MeCP2_low$scaled_intensity <- rescale(DAPI_for_MeCP2_low$Y)
GFP_for_MeCP2_low <- read_csv("ki67_microscopy_data/Krasinska_data/GFP for MeCP2 low.csv")
GFP_for_MeCP2_low$channel <- rep("GFP-Ki-67",nrow(GFP_for_MeCP2_low))
GFP_for_MeCP2_low$scaled_intensity <- rescale(GFP_for_MeCP2_low$Y)
RED_for_MeCP2_low <- read_csv("ki67_microscopy_data/Krasinska_data/red for MeCP2 low.csv")
RED_for_MeCP2_low$channel <- rep("MeCP2",nrow(RED_for_MeCP2_low))
RED_for_MeCP2_low$scaled_intensity <- rescale(RED_for_MeCP2_low$Y)

merge_for_MeCP2_low <- rbind(DAPI_for_MeCP2_low,GFP_for_MeCP2_low,RED_for_MeCP2_low)
colnames(merge_for_MeCP2_low)[1:2] <- c("distance","intensity")

# High Ki-67 expression
DAPI_for_MeCP2_high <- read_csv("ki67_microscopy_data/Krasinska_data/DAPI for MeCP2 high.csv")
DAPI_for_MeCP2_high$channel <- rep("DAPI",nrow(DAPI_for_MeCP2_high))
DAPI_for_MeCP2_high$scaled_intensity <- rescale(DAPI_for_MeCP2_high$Y)
GFP_for_MeCP2_high <- read_csv("ki67_microscopy_data/Krasinska_data/GFP for MeCP2 high.csv")
GFP_for_MeCP2_high$channel <- rep("GFP-Ki-67",nrow(GFP_for_MeCP2_high))
GFP_for_MeCP2_high$scaled_intensity <- rescale(GFP_for_MeCP2_high$Y)
RED_for_MeCP2_high <- read_csv("ki67_microscopy_data/Krasinska_data/red for MeCP2 high.csv")
RED_for_MeCP2_high$channel <- rep("MeCP2",nrow(RED_for_MeCP2_high))
RED_for_MeCP2_high$scaled_intensity <- rescale(RED_for_MeCP2_high$Y)

merge_for_MeCP2_high <- rbind(DAPI_for_MeCP2_high,GFP_for_MeCP2_high,RED_for_MeCP2_high)
colnames(merge_for_MeCP2_high)[1:2] <- c("distance","intensity")

# Low Ki-67 expression

ggplot(merge_for_H3K9me3_low, aes(x=distance,y=scaled_intensity)) + 
    geom_line(aes(color=channel),size=1.5) +
    theme_classic() +
    theme(legend.title=element_blank(),
          text = element_text(size=20),
          axis.line.x.bottom=element_line(linewidth = 1.1),
          axis.line.y.left=element_line(linewidth = 1.1)
          ) +
    scale_color_manual(values=c("#3753a4ff","#3eb649ff","#ee2e2eff"),labels=c("DAPI","GFP-Ki67","H3K9me3"))


# High Ki-67 expression

ggplot(merge_for_H3K9me3_high, aes(x=distance,y=scaled_intensity)) + 
    geom_line(aes(color=channel),size=1.5) +
    theme_classic() +
    theme(legend.title=element_blank(),
          text = element_text(size=20),
          axis.line.x.bottom=element_line(linewidth = 1.1),
          axis.line.y.left=element_line(linewidth = 1.1)
          ) +
    scale_color_manual(values=c("#3753a4ff","#3eb649ff","#ee2e2eff"),labels=c("DAPI","GFP-Ki67","H3K9me3"))


#==========================================MeCP2=========================================================#

# Low Ki-67 expression
ggplot(merge_for_MeCP2_low, aes(x=distance,y=scaled_intensity)) + 
    geom_line(aes(color=channel),size=1.5) +
    theme_classic() +
    theme(legend.title=element_blank(),
          text = element_text(size=20),
          axis.line.x.bottom=element_line(linewidth = 1.1),
          axis.line.y.left=element_line(linewidth = 1.1)
          ) +
    scale_color_manual(values=c("#3753a4ff","#3eb649ff","#ee2e2eff"),labels=c("DAPI","GFP-Ki67","MeCP2"))

# High Ki-67 expression


ggplot(merge_for_MeCP2_high, aes(x=distance,y=scaled_intensity)) + 
    geom_line(aes(color=channel),size=1.5) +
    theme_classic() +
    theme(legend.title=element_blank(),
          text = element_text(size=20),
          axis.line.x.bottom=element_line(linewidth = 1.1),
          axis.line.y.left=element_line(linewidth = 1.1)
          ) +
    scale_color_manual(values=c("#3753a4ff","#3eb649ff","#ee2e2eff"),labels=c("DAPI","GFP-Ki67","MeCP2"))

```

### c

#### Interphase

```{r FRAP modelling - interphase ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=7}

#preprocessing the data
# https://github.com/GuanqiaoDing/frapplot/blob/master/data-raw/preprocess.R
# Generate data files from raw data
#

#load the data from bleached regions

Avg_Spectrum_ROI1 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase/Avg_Spectrum_ROI1.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

Avg_Spectrum_ROI2 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase/Avg_Spectrum_ROI2.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

#load the data from unbleached regions

Avg_Spectrum_UB1 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase/Avg_Spectrum_UB1.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB2 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase/Avg_Spectrum_UB2.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

#load the data from nucleus to normalise with it later on
Avg_Spectrum_nucleus <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase/Avg_Spectrum_nucleus.txt", 
     delim = "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

#load the data from all the UB regions (unbleached)

##for the final figure we removed the last 50s (because acquisition was made for 150s, we keep now from 1s to 100s) of the analysis because the cells were frequently moving in these last seconds, so we use the following code to normalise and remove the 50s last seconds:

bkg_normalizing_vector <- as.numeric(Avg_Spectrum_nucleus[2:99,2][[1]])

ROI_spectrum_matrix <- as.matrix(cbind(Avg_Spectrum_ROI1[2:99,2],Avg_Spectrum_ROI2[2:99,2]))
ROI_spectrum_matrix_norm <- ROI_spectrum_matrix/bkg_normalizing_vector

UB_spectrum_matrix <- as.matrix(cbind(Avg_Spectrum_UB1[2:99,2],Avg_Spectrum_UB2[2:99,2]))
UB_spectrum_matrix_norm <- UB_spectrum_matrix/bkg_normalizing_vector


#DATA MODELING
#______________Model FRAP data____________________________________________

frap_input <- list(ROI=ROI_spectrum_matrix_norm)#,UB=UB_spectrum_df_norm)
time_vector <- 0:(nrow(ROI_spectrum_matrix)-2)
frap_info<-frapprocess(frap_input,time_vector)

FRAP_plotData<-as.data.frame(cbind(frap_info$time_points,frap_info$sample_means,frap_info$sample_sd))


FRAP_plotData <- cbind(FRAP_plotData,rep("Bleached",nrow(FRAP_plotData)))

colnames(FRAP_plotData) <- c("time","normalized_intensity","sd","region")

#______________UnBleached data____________________________________________


# Emulate the frapprocess processing of the data in the unbleached data
#Normalized by the time before bleaching and remove that datapoint
UB_spectrum_matrix_norm <- t(t(UB_spectrum_matrix_norm)/UB_spectrum_matrix_norm[1,])
UB_spectrum_matrix_norm <- UB_spectrum_matrix_norm[-1,]

# calculate sample mean and standard deviation
UB_spectrum_means <- rowMeans(UB_spectrum_matrix_norm)
UB_spectrum_sd <- apply(UB_spectrum_matrix_norm, 1, sd)

UnBleached_data <- cbind(FRAP_plotData$time,UB_spectrum_means,UB_spectrum_sd,rep("Un-bleached",nrow(FRAP_plotData)))
colnames(UnBleached_data) <- c("time","normalized_intensity","sd","region")

plotData <- as.data.frame(rbind(FRAP_plotData,UnBleached_data))
plotData$time <- as.numeric(plotData$time)
plotData$normalized_intensity <- as.numeric(plotData$normalized_intensity)
plotData$sd <- as.numeric(plotData$sd)

#NON-LINEAR REGRESSION MODEL
#   model: FR ~ c - a * exp(b * time)
a <- frap_info$model[[1]]$m$getPars()[1]
b <- frap_info$model[[1]]$m$getPars()[2]
c <- frap_info$model[[1]]$m$getPars()[3]
t_half <- frap_info$summary$halftime
tau <- frap_info$summary$tau
recov <- frap_info$summary$total_recovery


#FINAL VERSION OF THE CODE FOR THE FIGURES OF RECOVERY CURVE


ggplot(data=plotData,aes(x = time,y = normalized_intensity,color=region))+
    geom_errorbar(aes(ymin=normalized_intensity-sd, ymax=normalized_intensity+sd),color="gray") +
    geom_point() + 
    geom_function(fun = function(time){c - a * exp(b * time)}, color="#f69220ff",linewidth=1.9,alpha=0.7) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"),
    axis.text=element_text(size=12),
    axis.title=element_text(size=18),
    legend.text=element_text(size=12),
    legend.title = element_text(size=19),
    axis.text.x = element_text(face="bold",size=18),
    axis.text.y = element_text(face="bold",size=18)) +
    ylab("Normalized intensity")+ xlab("Time (s)") +
    scale_color_manual(values = c("deepskyblue4","cyan4"))


```

#### Interphase (high expression)

```{r FRAP modelling - interphase (High expression) ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=7}

#preprocessing the data
# https://github.com/GuanqiaoDing/frapplot/blob/master/data-raw/preprocess.R
# Generate data files from raw data
#

#load the data from bleached regions

Avg_Spectrum_ROI1 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression/Avg_Spectrum_ROI1.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_ROI2 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression/Avg_Spectrum_ROI2.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_ROI3 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression/Avg_Spectrum_ROI3.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_ROI4 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression/Avg_Spectrum_ROI4.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_ROI5 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression/Avg_Spectrum_ROI5.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_ROI6 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression/Avg_Spectrum_ROI6.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_ROI7 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression/Avg_Spectrum_ROI7.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

#load the data from unbleached regions

Avg_Spectrum_UB1 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression//Avg_Spectrum_UB1.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB2 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression//Avg_Spectrum_UB2.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB3 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression//Avg_Spectrum_UB3.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB4 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression//Avg_Spectrum_UB5.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB5 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression//Avg_Spectrum_UB5.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB6 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression//Avg_Spectrum_UB6.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB7 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression//Avg_Spectrum_UB7.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

#load the data from nucleus to normalise with it later on
Avg_Spectrum_nucleus <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/interphase high expression//Avg_Spectrum_nucleus.txt", 
     delim = "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)

#load the data from all the UB regions (unbleached)

##for the final figure we removed the last 50s (because acquisition was made for 150s, we keep now from 1s to 100s) of the analysis because the cells were frequently moving in these last seconds, so we use the following code to normalise and remove the 50s last seconds:

bkg_normalizing_vector <- as.numeric(Avg_Spectrum_nucleus[2:99,2][[1]])

ROI_spectrum_matrix <- as.matrix(cbind(Avg_Spectrum_ROI1[2:99,2],Avg_Spectrum_ROI2[2:99,2],Avg_Spectrum_ROI3[2:99,2],Avg_Spectrum_ROI4[2:99,2],Avg_Spectrum_ROI5[2:99,2],Avg_Spectrum_ROI6[2:99,2],Avg_Spectrum_ROI7[2:99,2]))
ROI_spectrum_matrix_norm <- ROI_spectrum_matrix/bkg_normalizing_vector

UB_spectrum_matrix <- as.matrix(cbind(Avg_Spectrum_UB1[2:99,2],Avg_Spectrum_UB2[2:99,2],Avg_Spectrum_UB3[2:99,2],Avg_Spectrum_UB4[2:99,2],Avg_Spectrum_UB5[2:99,2],Avg_Spectrum_UB6[2:99,2],Avg_Spectrum_UB7[2:99,2]))
UB_spectrum_matrix_norm <- UB_spectrum_matrix/bkg_normalizing_vector


#DATA MODELING
#______________Model FRAP data____________________________________________

frap_input <- list(ROI=ROI_spectrum_matrix_norm)#,UB=UB_spectrum_df_norm)
time_vector <- 0:(nrow(ROI_spectrum_matrix)-2)
frap_info<-frapprocess(frap_input,time_vector)

FRAP_plotData<-as.data.frame(cbind(frap_info$time_points,frap_info$sample_means,frap_info$sample_sd))


FRAP_plotData <- cbind(FRAP_plotData,rep("Bleached",nrow(FRAP_plotData)))

colnames(FRAP_plotData) <- c("time","normalized_intensity","sd","region")

#______________UnBleached data____________________________________________


# Emulate the frapprocess processing of the data in the unbleached data
#Normalized by the time before bleaching and remove that datapoint
UB_spectrum_matrix_norm <- t(t(UB_spectrum_matrix_norm)/UB_spectrum_matrix_norm[1,])
UB_spectrum_matrix_norm <- UB_spectrum_matrix_norm[-1,]

# calculate sample mean and standard deviation
UB_spectrum_means <- rowMeans(UB_spectrum_matrix_norm)
UB_spectrum_sd <- apply(UB_spectrum_matrix_norm, 1, sd)

UnBleached_data <- cbind(FRAP_plotData$time,UB_spectrum_means,UB_spectrum_sd,rep("Un-bleached",nrow(FRAP_plotData)))
colnames(UnBleached_data) <- c("time","normalized_intensity","sd","region")

plotData <- as.data.frame(rbind(FRAP_plotData,UnBleached_data))
plotData$time <- as.numeric(plotData$time)
plotData$normalized_intensity <- as.numeric(plotData$normalized_intensity)
plotData$sd <- as.numeric(plotData$sd)

#NON-LINEAR REGRESSION MODEL
#   model: FR ~ c - a * exp(b * time)
a <- frap_info$model[[1]]$m$getPars()[1]
b <- frap_info$model[[1]]$m$getPars()[2]
c <- frap_info$model[[1]]$m$getPars()[3]
t_half <- frap_info$summary$halftime
tau <- frap_info$summary$tau
recov <- frap_info$summary$total_recovery


#FINAL VERSION OF THE CODE FOR THE FIGURES OF RECOVERY CURVE


ggplot(data=plotData,aes(x = time,y = normalized_intensity,color=region))+
    geom_errorbar(aes(ymin=normalized_intensity-sd, ymax=normalized_intensity+sd),color="gray") +
    geom_point() + 
    geom_function(fun = function(time){c - a * exp(b * time)}, color="#f69220ff",linewidth=1.9,alpha=0.7) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"),
    axis.text=element_text(size=12),
    axis.title=element_text(size=18),
    legend.text=element_text(size=12),
    legend.title = element_text(size=19),
    axis.text.x = element_text(face="bold",size=18),
    axis.text.y = element_text(face="bold",size=18)) +
    ylab("Normalized intensity")+ xlab("Time (s)") +
    scale_color_manual(values = c("deepskyblue4","cyan4"))


```

#### Mitosis

```{r FRAP modelling - mitosis,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=7}

#preprocessing the data
# https://github.com/GuanqiaoDing/frapplot/blob/master/data-raw/preprocess.R
# Generate data files from raw data
#

#load the data from bleached regions

Avg_Spectrum_ROI1 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/mitosis/Avg_Spectrum_ROI1.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_ROI2 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/mitosis/Avg_Spectrum_ROI2.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_ROI3 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/mitosis/Avg_Spectrum_ROI3.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_ROI4 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/mitosis/Avg_Spectrum_ROI4.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

#load the data from unbleached regions

Avg_Spectrum_UB1 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/mitosis/Avg_Spectrum_UB1.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB2 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/mitosis/Avg_Spectrum_UB2.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB3 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/mitosis/Avg_Spectrum_UB3.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
Avg_Spectrum_UB4 <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/mitosis/Avg_Spectrum_UB4.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

#load the data from nucleus to normalise with it later on
Avg_Spectrum_nucleus <- read_delim("FRAP_data/Andres-Sanchez_data/230829 raw data of representative cells/mitosis/Avg_Spectrum_mitoticchr.txt", 
     delim = "\t", escape_double = FALSE, col_names = FALSE, 
     trim_ws = TRUE)


#load the data from all the UB regions (unbleached)

##for the final figure we removed the last 50s (because acquisition was made for 150s, we keep now from 1s to 100s) of the analysis because the cells were frequently moving in these last seconds, so we use the following code to normalise and remove the 50s last seconds:

bkg_normalizing_vector <- as.numeric(Avg_Spectrum_nucleus[2:99,2][[1]])

ROI_spectrum_matrix <- as.matrix(cbind(Avg_Spectrum_ROI1[2:99,2],Avg_Spectrum_ROI2[2:99,2],Avg_Spectrum_ROI3[2:99,2],Avg_Spectrum_ROI4[2:99,2]))
ROI_spectrum_matrix_norm <- ROI_spectrum_matrix/bkg_normalizing_vector

UB_spectrum_matrix <- as.matrix(cbind(Avg_Spectrum_UB1[2:99,2],Avg_Spectrum_UB2[2:99,2],Avg_Spectrum_UB3[2:99,2],Avg_Spectrum_UB4[2:99,2]))
UB_spectrum_matrix_norm <- UB_spectrum_matrix/bkg_normalizing_vector


#DATA MODELING
#______________Model FRAP data____________________________________________

frap_input <- list(ROI=ROI_spectrum_matrix_norm)#,UB=UB_spectrum_df_norm)
time_vector <- 0:(nrow(ROI_spectrum_matrix)-2)
frap_info<-frapprocess(frap_input,time_vector)

FRAP_plotData<-as.data.frame(cbind(frap_info$time_points,frap_info$sample_means,frap_info$sample_sd))


FRAP_plotData <- cbind(FRAP_plotData,rep("Bleached",nrow(FRAP_plotData)))

colnames(FRAP_plotData) <- c("time","normalized_intensity","sd","region")

#______________UnBleached data____________________________________________


# Emulate the frapprocess processing of the data in the unbleached data
#Normalized by the time before bleaching and remove that datapoint
UB_spectrum_matrix_norm <- t(t(UB_spectrum_matrix_norm)/UB_spectrum_matrix_norm[1,])
UB_spectrum_matrix_norm <- UB_spectrum_matrix_norm[-1,]

# calculate sample mean and standard deviation
UB_spectrum_means <- rowMeans(UB_spectrum_matrix_norm)
UB_spectrum_sd <- apply(UB_spectrum_matrix_norm, 1, sd)

UnBleached_data <- cbind(FRAP_plotData$time,UB_spectrum_means,UB_spectrum_sd,rep("Un-bleached",nrow(FRAP_plotData)))
colnames(UnBleached_data) <- c("time","normalized_intensity","sd","region")

plotData <- as.data.frame(rbind(FRAP_plotData,UnBleached_data))
plotData$time <- as.numeric(plotData$time)
plotData$normalized_intensity <- as.numeric(plotData$normalized_intensity)
plotData$sd <- as.numeric(plotData$sd)

#NON-LINEAR REGRESSION MODEL
#   model: FR ~ c - a * exp(b * time)
a <- frap_info$model[[1]]$m$getPars()[1]
b <- frap_info$model[[1]]$m$getPars()[2]
c <- frap_info$model[[1]]$m$getPars()[3]
t_half <- frap_info$summary$halftime
tau <- frap_info$summary$tau
recov <- frap_info$summary$total_recovery


#FINAL VERSION OF THE CODE FOR THE FIGURES OF RECOVERY CURVE


ggplot(data=plotData,aes(x = time,y = normalized_intensity,color=region))+
    geom_errorbar(aes(ymin=normalized_intensity-sd, ymax=normalized_intensity+sd),color="gray") +
    geom_point() + 
    geom_function(fun = function(time){c - a * exp(b * time)}, color="#f69220ff",linewidth=1.9,alpha=0.7) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"),
    axis.text=element_text(size=12),
    axis.title=element_text(size=18),
    legend.text=element_text(size=12),
    legend.title = element_text(size=19),
    axis.text.x = element_text(face="bold",size=18),
    axis.text.y = element_text(face="bold",size=18)) +
    ylab("Normalized intensity")+ xlab("Time (s)") +
    scale_color_manual(values = c("deepskyblue4","cyan4"))


```

### d

```{r FRAP quantification ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=5}

thalf <- read_delim("FRAP_data/thalfboth.csv", 
    delim = ";", escape_double = FALSE, col_names = TRUE, 
    trim_ws = TRUE)


thalf$group <- factor(thalf$group, levels=c("mitosis", "interphase", "interphase high expression"))


ggplot(data=thalf, aes(x=group,y=thalf,fill=group))+
  geom_boxplot(outlier.shape=NA)+
  geom_dotplot(binaxis='y',fill="black",binwidth=1,alpha=0.8,stackdir = "center")+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  axis.text=element_text(size=16,face="bold"),
  axis.title=element_text(size=18,face="bold"),
  legend.position = "none")+
  ylab("t half (s)")+xlab("")+
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10))+
  scale_fill_manual(values = c("#2aabe2ff", "#fedd10ff", "#f69220ff"))

#statistical test
#wilcoxon test comparing mitosis vs interphase
wilcox.test(thalf ~ group, data = subset(thalf,group!="interphase high expression"), alternative = "two.sided",conf.int=T,exact=FALSE)
#wilcoxon test comparing mitosis vs interphase high expression
wilcox.test(thalf ~ group, data = subset(thalf,group!="interphase"), alternative = "two.sided",conf.int=T,exact=FALSE)
#wilcoxon test comparing interphase vs interphase high expression
wilcox.test(thalf ~ group, data = subset(thalf,group!="mitosis"), alternative = "two.sided",conf.int=T,exact=FALSE)




##to plot the graphs of the recovery data
recover <- read_delim("FRAP_data/percboth.csv",
    delim = ";", escape_double = FALSE, col_names = TRUE, 
    trim_ws = TRUE)

recover$group <- factor(recover$group, levels=c("mitosis", "interphase", "interphase high expression"))

ggplot(data=recover, aes(x=group,y=percent,fill=group))+
  geom_boxplot(outlier.shape=NA)+
  geom_dotplot(binaxis='y',fill="black",binwidth=3,alpha=0.8,stackdir = "center")+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  axis.text=element_text(size=16,face="bold"),
  axis.title=element_text(size=18,face="bold"),
  legend.position = "none")+
  ylab("% recovery")+xlab("")+
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10))+
  scale_fill_manual(values = c("#2aabe2ff", "#fedd10ff", "#f69220ff"))

#wilcoxon test comparing mitosis vs interphase
wilcox.test(percent ~ group, data = subset(recover,group!="interphase high expression"), alternative = "two.sided",conf.int=T,exact=FALSE)
#wilcoxon test comparing mitosis vs interphase high expression
wilcox.test(percent ~ group, data = subset(recover,group!="interphase"), alternative = "two.sided",conf.int=T,exact=FALSE)
#wilcoxon test comparing interphase vs interphase high expression
wilcox.test(percent ~ group, data = subset(recover,group!="mitosis"), alternative = "two.sided",conf.int=T,exact=FALSE)

```

## Fig. 7

### c

```{r optodroplet quantification - FL ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=6}

Full_length_quant <- read_delim("Optodroplet_data/AlGhoul_data/Fig7_FL.tab", 
                      delim = "\t", escape_double = FALSE, 
                      trim_ws = TRUE)

### Full Lenght ###
kruskal.test(Foci_Nucleus ~ Treat_collapsed, data = Full_length_quant)

# Kruskal-Wallis rank sum test
# 
# data:  Foci_Nucleus by Treat_collapsed
# Kruskal-Wallis chi-squared = 205.15, df = 5, p-value < 2.2e-16

pairwise.wilcox.test(Full_length_quant$Foci_Nucleus, Full_length_quant$Treat_collapsed, p.adjust.method = "BH",)


# Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
# 
# data:  Full_length_quant$Foci_Nucleus and Full_length_quant$Treat_collapsed 
# 
#           DMSO L+ DMSO L- OA L+   OA L-   PA L+ 
#   DMSO L- 2.1e-14 -       -       -       -     
#   OA L+   0.3149  8.6e-14 -       -       -     
#   OA L-   0.0230  5.3e-11 0.1860  -       -     
#   PA L+   2.0e-15 0.6424  2.1e-14 1.8e-11 -     
#   PA L-   < 2e-16 7.0e-05 < 2e-16 < 2e-16 0.0011
# 
# P value adjustment method: BH 

ggplot(Full_length_quant,aes(x=Treatment, y=Foci_Nucleus)) +
  geom_violin(aes(fill=Light,color=Light),position=position_dodge(1),bw=2,alpha=1,trim = T,width=1) +
  geom_boxplot(aes(fill=Light),position=position_dodge(1),width=0.15,outlier.shape = NA,show.legend = F) +
  ggpubr::theme_classic2() +
  theme(text = element_text(size=20),legend.position = "bottom",legend.title = element_blank(),axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed")) +
  scale_y_continuous(limits = c(-1, 80), breaks = c(seq(0, 60, by = 20))) +
  xlab(NULL) +
  geom_segment(aes(x = 0.77, y = 61, xend = 1.22, yend = 61)) + annotate(geom="text", x=1, y=62, label="***") +
  geom_segment(aes(x = 1.77, y = 61, xend = 2.22, yend = 61)) + annotate(geom="text", x=2, y=63, label="n.s.") +
  geom_segment(aes(x = 2.77, y = 61, xend = 3.22, yend = 61)) + annotate(geom="text", x=3, y=62, label="**") +
  geom_segment(aes(x = 0.77, y = 65, xend = 1.77, yend = 65)) + annotate(geom="text", x=1.27, y=66, label="***") +
  geom_segment(aes(x = 1.22, y = 69, xend = 2.22, yend = 69)) + annotate(geom="text", x=1.72, y=71, label="n.s.") +
  geom_segment(aes(x = 1.22, y = 73, xend = 3.22, yend = 73)) + annotate(geom="text", x=2.22, y=74, label="***") +
  scale_fill_manual(values = pal_jco()(10)[c(5,2)]) +
  scale_color_manual(values = pal_jco()(10)[c(5,2)],guide="none")
```

# Supplemental information

## Comparing predictors

This block compares the residues predicted as to be disordered across the entire proteomes of yeast and human using different disorders. A Venn diagram is generated to illustrate the overlap between different methods

```{r compare index ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=5}
# This blocks compares the residues predicted as to be disordered across the entire proteomes of yeas and human using different disorders. A Venn diagram is generated to illustrate the overlap between different methods 

# Yeast predictors
yeastSpotIndexes_vec <- base::strsplit(yeastDiso_full$spot_disoIndexes,',')
yeastSpotResidueIDs <- list()
for (i in 1:nrow(yeastDiso_full)) {
  yeastSpotResidueIDs[[i]] <- paste(yeastDiso_full$acc[[i]],yeastSpotIndexes_vec[[i]],sep = "_")
}
yeastSpotResidueIDs<-unlist(yeastSpotResidueIDs)

yeastVslIndexes_vec <- base::strsplit(yeastDiso_full$vsl_disoIndexes,',')
yeastVslResidueIDs <- list()
for (i in 1:nrow(yeastDiso_full)) {
  yeastVslResidueIDs[[i]] <- paste(yeastDiso_full$acc[[i]],yeastVslIndexes_vec[[i]],sep = "_")
}
yeastVslResidueIDs<-unlist(yeastVslResidueIDs)

yeastIUPredIndexes_vec <- base::strsplit(yeastDiso_full$iupl_disoIndexes,',')
yeastIUPredResidueIDs <- list()
for (i in 1:nrow(yeastDiso_full)) {
  yeastIUPredResidueIDs[[i]] <- paste(yeastDiso_full$acc[[i]],yeastIUPredIndexes_vec[[i]],sep = "_")
}
yeastIUPredResidueIDs<-unlist(yeastIUPredResidueIDs)

yeast_sets <- list("IUPred"=yeastIUPredResidueIDs,"SPOT"=yeastSpotResidueIDs,"VSL2B"=yeastVslResidueIDs)

eulerr_options(fills=list(fill=c("#f69220ff","#fedd10ff","#2aabe2ff"),alpha=0.6))
plot(euler(yeast_sets, shape = "ellipse"), quantities = TRUE)



# Human predictors

humanSpotIndexes_vec <- base::strsplit(humanDiso_full$spot_disoIndexes,',')
humanSpotResidueIDs <- list()
for (i in 1:nrow(humanDiso_full)) {
  humanSpotResidueIDs[[i]] <- paste(humanDiso_full$acc[[i]],humanSpotIndexes_vec[[i]],sep = "_")
}
humanSpotResidueIDs<-unlist(humanSpotResidueIDs)

humanVslIndexes_vec <- base::strsplit(humanDiso_full$vsl_disoIndexes,',')
humanVslResidueIDs <- list()
for (i in 1:nrow(humanDiso_full)) {
  humanVslResidueIDs[[i]] <- paste(humanDiso_full$acc[[i]],humanVslIndexes_vec[[i]],sep = "_")
}
humanVslResidueIDs<-unlist(humanVslResidueIDs)

humanIUPredIndexes_vec <- base::strsplit(humanDiso_full$iupl_disoIndexes,',')
humanIUPredResidueIDs <- list()
for (i in 1:nrow(humanDiso_full)) {
  humanIUPredResidueIDs[[i]] <- paste(humanDiso_full$acc[[i]],humanIUPredIndexes_vec[[i]],sep = "_")
}
humanIUPredResidueIDs<-unlist(humanIUPredResidueIDs)

human_sets <- list("IUPred"=humanIUPredResidueIDs,"SPOT"=humanSpotResidueIDs,"VSL2B"=humanVslResidueIDs)

eulerr_options(fills=list(fill=c("#f69220ff","#fedd10ff","#2aabe2ff"),alpha=0.6))
plot(euler(human_sets, shape = "ellipse"), quantities = TRUE)



```


## Supp. Fig. 3

### a


```{r DNA rep factors, echo=FALSE,message=FALSE,warning=FALSE,fig.height=8,fig.width=4}
#Load data

pSites <- read.csv('phosphorylation_data/Valverde_data/Phosphosites_XL_Extract_All_MedianNorm_SummaryStatistics_NoOA.csv', stringsAsFactors = F)


#Creating unique identifier

pSites_Unique_ID <- unite(pSites, 'Protein_Amino.acid_Site_NumberPhospho', c('Protein', 'Amino.acid', 'Position', 'Multiplicity'),
                          sep ='|', remove = FALSE)

#Filtering for ANOVA sig

pSites_Unique_ID_ANOVA <- pSites_Unique_ID[pSites_Unique_ID$ANOVA.Significant == '+', ]

#Df with only proteins of interest

DNA_Reps <- c('mcm2', 'mcm3', 'mcm4', 'mcm5', 'mcm6', 'mcm7', 'mcm10', 'rif1', 'orc1', 'orc2', 'cdt1', 'cdc6', 'dpola', 'rpa1',
              'rfc1', 'ctf4', 'wdhd1', 'mtbp', 'ticrr', 'recq4')

pSites_Unique_ID_ANOVA_DNARep <- filter(pSites_Unique_ID_ANOVA, Protein %in% DNA_Reps)

#Keeping only mean intensities and identifier

pSites_Unique_ID_ANOVA_DNARep_Means <- pSites_Unique_ID_ANOVA_DNARep[, c(21, 39:45)]

#Transforming to numeric matrix

pSites_Unique_ID_ANOVA_DNARep_Means_Matrix <-
  as.matrix(sapply(pSites_Unique_ID_ANOVA_DNARep_Means, as.numeric))

#Adding row names

rownames(pSites_Unique_ID_ANOVA_DNARep_Means_Matrix) <- pSites_Unique_ID_ANOVA_DNARep_Means[, 1]

#Removing useless column

pSites_Unique_ID_ANOVA_DNARep_Means_Matrix <- pSites_Unique_ID_ANOVA_DNARep_Means_Matrix[,-1]

#Scaling

pSites_Unique_ID_ANOVA_DNARep_Means_Matrix_Scaled <- t(scale(t(pSites_Unique_ID_ANOVA_DNARep_Means_Matrix)))

#Creating palette

library(circlize)
col_fun = colorRamp2(c(-2.2, 0, 2.2), c("cyan", "black", "yellow"))

#Creating heat map

ha = rowAnnotation(foo = anno_mark(at = c(2, 3, 8, 9, 10, 11, 15, 16, 17, 18, 19, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 40, 41, 42, 43, 44, 45,
                                          47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 61, 62, 63, 64, 66, 67, 68, 69, 70, 71), 
                                   labels = c('MCM10-S206', 'MCM10-S596', 'MCM4-S31', 'MCM4-S88', 'MCM4-S120', 'MCM4-S145',
                                              'TICRR-S978', 'TICRR-S1941', 'TICRR-S1763', 'TICRR-S1660', 'TICRR-S1833', 'CDC6-S70',
                                              'CDC6-S8', 'RECQ4-S151', 'RECQ4-S310', 'RPA1-S1490', 'RIF1-S904', 'RIF1-S1166', 'RIF1-S1439',
                                              'RIF1-S1515', 'RIF1-S1782', 'RIF1-S870', 'RIF1-S1728', 'RIF1-S1989', 'ORC2-T115', 'ORC2-T157',
                                              'TICRR-T1261', 'TICRR-T1167', 'TICRR-T1617', 'TICRR-T1728', 'ORC1-T332', 'ORC1-T401', 'RIF1-T690',
                                              'RIF1-T1385', 'RIF1-T453', 'MCM4-S31(2)', 'MCM4-S87(2)', 'MCM4-S3(2)', 'TICRR-S973(2)', 'TICRR-S978(2)',
                                              'RIF1-S1439(2)', 'RIF1-S1717(2)', 'RIF1-S1719(2)', 'MCM4-T23(2)', 'MCM4-T94(2)', 'MCM4-T7(2)',
                                              'TICRR-T1300(2)', 'MCM4-S87(3)', 'MCM4-S88(3)', 'RIF1-S1439(3)', 'RIF1-S1450(3)', 'MCM4-T94(3)', 'RIF1-S1430(3)'), labels_gp = gpar(fontsize = 6)))

Heatmap(pSites_Unique_ID_ANOVA_DNARep_Means_Matrix_Scaled, 
        column_order = colnames(pSites_Unique_ID_ANOVA_DNARep_Means_Matrix_Scaled),
        show_row_names = TRUE, clustering_distance_rows = 'pearson', show_row_dend = FALSE,
        clustering_method_rows = 'average', name = 'Z-Score', col = col_fun,
        row_names_gp = gpar(fontsize = 8), right_annotation = ha,
        column_names_rot = 90)

```

### c


```{r overlap embryo and extracts, echo=FALSE,message=FALSE,warning=FALSE,fig.height=8,fig.width=4}
col_fun = colorRamp2(c(-3, 0, 3), c("cyan", "black", "yellow"))

#Load data

pSites_Median_Norm_Full_Matrix_Statistics <- read.csv('phosphorylation_data/Valverde_data/All_pSites_Log2_Normalized_Imputed_ANOVA_Statistics.csv', stringsAsFactors = F)

#Creating unique useful identifier

pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID <- unite(pSites_Median_Norm_Full_Matrix_Statistics, 
                                                  'Protein_Amino.acid_Site_NumberPhospho', 
                                                  c('Protein', 'Amino.acid', 'Position', 'Multiplicity'),
                                                  sep ='|', remove = FALSE)

#Df with only intensities of time point of interest

pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int <- 
  pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID[, c(24, 52:57, 59, 62:67, 69)]

#reordering columns

col_order <- c('Mean.Interphase_000', "Mean.Interphase_030", 'Mean.Interphase_060', 'Mean.Interphase_090',
               'Mean.Interphase_120', 'Mean.Mitosis_CyB',  'Mean.Mitosis_CSF',
               'Median.Interphase_000', "Median.Interphase_030", 'Median.Interphase_060', 'Median.Interphase_090',
               'Median.Interphase_120', 'Median.Mitosis_CyB', 'Median.Mitosis_CSF', 
               'Protein_Amino.acid_Site_NumberPhospho')

pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered <- pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int[, col_order]

#Transforming to matrix

pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix <-
  as.matrix(sapply(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered, as.numeric))

#adding rownames

rownames(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix) <- pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered[, 15]

#removing useless column

pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy <- pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix[, -c(15)]

#Changing column names

colnames(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy) <- c("DNA replication - 00'(mean)", "DNA replication - 30'(mean)",
                                                                                           "DNA replication - 60'(mean)", "DNA replication - 90'(mean)",
                                                                                           "DNA replication - 120'(mean)", "Mitosis - Cyclin B - 60'(mean)",
                                                                                           "Mitosis - Metaphase arrested (CSF)(mean)",
                                                                                           "DNA replication - 00'(median)", "DNA replication - 30'(median)",
                                                                                           "DNA replication - 60'(median)", "DNA replication - 90'(median)",
                                                                                           "DNA replication - 120'(median)", "Mitosis - Cyclin B - 60'(median)",
                                                                                           "Mitosis - Metaphase arrested (CSF)(median)")

#Plotting with Complex Heat map
#Scaling is needed, not done automatically by Complex Heat Map

pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy_Scaled <-
  t(scale(t(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy)))

#Creating vector that includes values to plot#
#Loading overlap for each cluster

Embryo_A_Extract_ANOVA <- read.csv('phosphorylation_data/Valverde_data/Embryo_A_Extract_ANOVA.csv', stringsAsFactors = FALSE)

Embryo_B_Extract_ANOVA <- read.csv('phosphorylation_data/Valverde_data/Embryo_B_Extract_ANOVA.csv', stringsAsFactors = FALSE)

Embryo_C_Extract_ANOVA <- read.csv('phosphorylation_data/Valverde_data/Embryo_C_Extract_ANOVA.csv', stringsAsFactors = FALSE)

Embryo_D_Extract_ANOVA <- read.csv('phosphorylation_data/Valverde_data/Embryo_D_Extract_ANOVA.csv', stringsAsFactors = FALSE)


#Subsetting
Embryo_A_Extract_ANOVA_Averages <- subset(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy_Scaled, 
                                          rownames(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy_Scaled) %in% Embryo_A_Extract_ANOVA[,1])
Embryo_B_Extract_ANOVA_Averages <- subset(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy_Scaled, 
                                          rownames(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy_Scaled) %in% Embryo_B_Extract_ANOVA[,1])
Embryo_C_Extract_ANOVA_Averages <- subset(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy_Scaled, 
                                             rownames(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy_Scaled) %in% Embryo_C_Extract_ANOVA[,1])
Embryo_D_Extract_ANOVA_Averages <- subset(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy_Scaled, 
                                          rownames(pSites_Median_Norm_Full_Matrix_Statistics_Unique_ID_Int_ordered_Matrix_Tidy_Scaled) %in% Embryo_D_Extract_ANOVA[,1])

#Plotting#


Heatmap(Embryo_A_Extract_ANOVA_Averages[, c(1:7)], 
             column_order = colnames(Embryo_A_Extract_ANOVA_Averages[, c(1:7)]),
             show_row_names = FALSE, clustering_distance_rows = 'spearman',
             clustering_method_rows = 'complete', name = 'Z-Score', column_names_rot = 90, col = col_fun, column_title = "Cluster A")

Heatmap(Embryo_B_Extract_ANOVA_Averages[, c(1:7)], 
             column_order = colnames(Embryo_A_Extract_ANOVA_Averages[, c(1:7)]),
             show_row_names = FALSE, clustering_distance_rows = 'spearman',
             clustering_method_rows = 'complete', name = 'Z-Score', column_names_rot = 90, col = col_fun, column_title="Cluster B")

Heatmap(Embryo_C_Extract_ANOVA_Averages[, c(1:7)], 
             column_order = colnames(Embryo_A_Extract_ANOVA_Averages[, c(1:7)]),
             show_row_names = FALSE, clustering_distance_rows = 'spearman',
             clustering_method_rows = 'complete', name = 'Z-Score', column_names_rot = 90, col = col_fun, column_title="Cluster C")

Heatmap(Embryo_D_Extract_ANOVA_Averages[, c(1:7)], 
             column_order = colnames(Embryo_A_Extract_ANOVA_Averages[, c(1:7)]),
             show_row_names = FALSE, clustering_distance_rows = 'spearman',
             clustering_method_rows = 'complete', name = 'Z-Score', column_names_rot = 90, col = col_fun, column_title="Cluster D")
```

## Supp. Fig. 4

### c

```{r motifs logo, echo=FALSE,message=FALSE,warning=FALSE,fig.height=15,fig.width=8}

ggseqlogo(IC_matrix_list,seq_type='aa',method='custom',ncol = 3) + 
  scale_y_continuous(limits = c(-0.5, 1),breaks = c(seq(-0.5, 1, by = 0.5)),expand = c(0,0)) + 
  scale_x_continuous(breaks = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15),labels=c("-7","-6","-5","-4","-3","-2","-1","P","+1","+2","+3","+4","+5","+6","+7")) +
  theme( axis.line.y = element_line(size = 0.5, linetype = "solid"),axis.ticks.y = element_line(size = 0.5, linetype = "solid")) + 
  geom_hline(yintercept=0, linetype="solid",size = 0.5)

```


### d


```{r extracts line plots all kinases, echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=6}

# Interphase
interphasePsitesPlot_1 <- ggplot(data = subset(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_WithMotif, Cluster %in% c('cluster4', 'cluster5', 'cluster6') & Time_Or_Treatment<=120), 
             aes(x = Time_Or_Treatment, y = value, color = Motif)) +
  #geom_jitter(size = 2, shape = 21, stroke = 1.3)+
  geom_smooth(se = TRUE, alpha = 0.2, span = 0.65, method = 'loess')+
  theme_bw() + coord_cartesian(ylim=c(-2.7,1.1), xlim=c(0,130))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank(), 
        axis.line = element_line(colour = "black", size = 0.25),axis.ticks = element_line(size = 0.25),
        axis.text.x = element_text(size = 14, color = 'black'),axis.text.y = element_text(size = 14, color = 'black'),
        axis.title.y = element_text(size = 16, color = 'black'), axis.title.x = element_text(size = 16, color = 'black'), plot.title = element_text(size = 18), 
        legend.title = element_text(size = 9, color = 'black'), legend.text = element_text(size = 9, color = 'black'), legend.position = 'top') +  
  ylab('Normalized Intensity') +
  xlab('Time (min)') +
  # ggtitle('Interphase phosphorylations clusters 4, 5, 6)') + 
  scale_x_continuous(breaks=c(0,30,60,90,120))
interphasePsitesPlot_1 <- interphasePsitesPlot_1 + scale_color_tron()


interphasePsitesPlot_2 <- ggplot(data = subset(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_WithMotif, Cluster %in% c('cluster4', 'cluster5', 'cluster6') & Time_Or_Treatment>120),
             aes(x = Time_Or_Treatment, y = value, color = Motif)) +
  geom_point(stat = "summary",fun = "mean", size = 5,shape=2,stroke = 2) +
  theme_bw() + coord_cartesian(ylim=c(-2.7,1.1), xlim=c(130,170))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank(), 
        axis.line = element_line(colour = "black", size = 0.25),axis.line.y = element_blank(),axis.ticks = element_blank(),
        axis.text.x = element_text(size = 14, color = 'black'),axis.text.y = element_blank(),
        axis.title.y = element_blank(), axis.title.x = element_blank(), plot.title = element_text(size = 18), 
        legend.title = element_text(size = 9, color = 'black'), legend.text = element_text(size = 9, color = 'black'), legend.position = 'top') +  
  scale_x_continuous(breaks=c(150),labels = c("Mitosis"))
interphasePsitesPlot_2 <- interphasePsitesPlot_2 + scale_color_tron()

ggarrange(interphasePsitesPlot_1,interphasePsitesPlot_2,ncol = 2,nrow = 1,align = "h",widths = c(13,4),common.legend = T)

# Mitosis
mphasePsitesPlot_1 <- ggplot(data = subset(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_WithMotif, Cluster %in% c('cluster1', 'cluster2', 'cluster3') & Time_Or_Treatment<=120), 
                                 aes(x = Time_Or_Treatment, y = value, color = Motif)) +
  #geom_jitter(size = 2, shape = 21, stroke = 1.3)+
  geom_smooth(se = TRUE, alpha = 0.2, span = 0.65, method = 'loess')+
  theme_bw() + coord_cartesian(ylim=c(-2.7,1.1), xlim=c(0,130))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank(), 
        axis.line = element_line(colour = "black", size = 0.25),axis.ticks = element_line(size = 0.25),
        axis.text.x = element_text(size = 14, color = 'black'),axis.text.y = element_text(size = 14, color = 'black'),
        axis.title.y = element_text(size = 16, color = 'black'), axis.title.x = element_text(size = 16, color = 'black'), plot.title = element_text(size = 18), 
        legend.title = element_text(size = 9, color = 'black'), legend.text = element_text(size = 9, color = 'black'), legend.position = 'top') +  
  ylab('Normalized Intensity') +
  xlab('Time (min)') +
  # ggtitle('Interphase phosphorylations clusters 4, 5, 6)') + 
  scale_x_continuous(breaks=c(0,30,60,90,120))
mphasePsitesPlot_1 <- mphasePsitesPlot_1 + scale_color_tron()


mphasePsitesPlot_2 <- ggplot(data = subset(pSites_with_clusters_fromHM_Normalized_tidy_TxAsNumeric_v2_WithMotif, Cluster %in% c('cluster1', 'cluster2', 'cluster3') & Time_Or_Treatment>120),
                                 aes(x = Time_Or_Treatment, y = value, color = Motif)) +
  geom_point(stat = "summary",fun = "mean", size = 5,shape=2,stroke = 2) +
  theme_bw() + coord_cartesian(ylim=c(-2.7,1.1), xlim=c(130,170))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank(), 
        axis.line = element_line(colour = "black", size = 0.25),axis.line.y = element_blank(),axis.ticks = element_blank(),
        axis.text.x = element_text(size = 14, color = 'black'),axis.text.y = element_blank(),
        axis.title.y = element_blank(), axis.title.x = element_blank(), plot.title = element_text(size = 18), 
        legend.title = element_text(size = 9, color = 'black'), legend.text = element_text(size = 9, color = 'black'), legend.position = 'top') +  
  scale_x_continuous(breaks=c(150),labels = c("Mitosis"))
mphasePsitesPlot_2 <- mphasePsitesPlot_2 + scale_color_tron()

ggarrange(mphasePsitesPlot_1,mphasePsitesPlot_2,ncol = 2,nrow = 1,align = "h",widths = c(13,4),common.legend = T)

```


## Supp. Fig. 5

### a

#### Yeast

```{r yeast composition ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=7,fig.width=7}
##YEAST

yeastSeq_all<-yeastDiso$Sequence
names(yeastSeq_all)<-yeastDiso$acc
yeastSeq_all<-Biostrings::AAStringSet(yeastSeq_all)

yeastComp_all<-colSums(alphabetFrequency(yeastSeq_all))[1:20]/sum(colSums(alphabetFrequency(yeastSeq_all))[1:20])

######################Set the rainbow color palette for yeast #####################
colorPallete_YeastComp <- rainbow(20)
names(colorPallete_YeastComp) <- levels(reorder(names(yeastComp_all),-yeastComp_all))
########################################################################################

plotYeastComp <- ggplot()+geom_bar(aes(x=reorder(names(yeastComp_all),-yeastComp_all), fill=reorder(names(yeastComp_all),-yeastComp_all), y=yeastComp_all),stat="identity") + scale_fill_manual(values=colorPallete_YeastComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank(),axis.ticks.y = element_blank(),axis.text.y = element_blank()) + scale_y_continuous(breaks = seq(0,0.1,0.025),limits = c(0,0.1)) + ggtitle("Proteome Composition")

##Iupred##


yeastSeqIupred_diso <- Biostrings::AAStringSet(yeastDiso_full$iupl_disoSeq)
yeastCompIupred<-colSums(alphabetFrequency(yeastSeqIupred_diso))[1:20]/sum(colSums(alphabetFrequency(yeastSeqIupred_diso))[1:20])


yeastIupredDiffComp <- sort((yeastCompIupred-yeastComp_all)/yeastComp_all,decreasing = T)


plotYeastIupred_DC <- ggplot()+geom_bar(aes(x=reorder(names(yeastIupredDiffComp),-yeastIupredDiffComp), fill=reorder(names(yeastIupredDiffComp),-yeastIupredDiffComp), y=yeastIupredDiffComp),stat="identity") + scale_fill_manual(values=colorPallete_YeastComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank()) + scale_y_continuous(breaks = c(-0.8,-0.4,0,0.4,0.8),limits = c(-0.9,0.9)) + ggtitle("IUpred")


##SPOT##

yeastSeqSpot_diso <- Biostrings::AAStringSet(yeastDiso_full$spot_disoSeq)
yeastCompSpot<-colSums(alphabetFrequency(yeastSeqSpot_diso))[1:20]/sum(colSums(alphabetFrequency(yeastSeqSpot_diso))[1:20])


yeastSpotDiffComp <- sort((yeastCompSpot-yeastComp_all)/yeastComp_all,decreasing = T)


plotYeastSpot_DC <- ggplot()+geom_bar(aes(x=reorder(names(yeastSpotDiffComp),-yeastSpotDiffComp), fill=reorder(names(yeastSpotDiffComp),-yeastSpotDiffComp), y=yeastSpotDiffComp),stat="identity") + scale_fill_manual(values=colorPallete_YeastComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank()) + scale_y_continuous(breaks = c(-0.8,-0.4,0,0.4,0.8),limits = c(-0.9,0.9)) + ggtitle("SPOT")


##VSL##

yeastSeqVsl_diso <- Biostrings::AAStringSet(yeastDiso_full$vsl_disoSeq)
yeastCompVsl<-colSums(alphabetFrequency(yeastSeqVsl_diso))[1:20]/sum(colSums(alphabetFrequency(yeastSeqVsl_diso))[1:20])


yeastVslDiffComp <- sort((yeastCompVsl-yeastComp_all)/yeastComp_all,decreasing = T)


plotYeastVsl_DC <- ggplot()+geom_bar(aes(x=reorder(names(yeastVslDiffComp),-yeastVslDiffComp), fill=reorder(names(yeastVslDiffComp),-yeastVslDiffComp), y=yeastVslDiffComp),stat="identity") + scale_fill_manual(values=colorPallete_YeastComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank()) + scale_y_continuous(breaks = c(-0.8,-0.4,0,0.4,0.8),limits = c(-0.9,0.9)) + ggtitle("VSL2B")



fig<-ggarrange(plotYeastComp,plotYeastIupred_DC,plotYeastSpot_DC,plotYeastVsl_DC,ncol = 1)
annotate_figure(fig,left = text_grob("Comp. Disorder - Comp. Universe / Comp. Universe", rot = 90,size = 16))
```

#### Human

```{r human composition ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=10}
##HUMAN

humanSeq_all<-humanDiso$Sequence
names(humanSeq_all)<-humanDiso$acc
humanSeq_all<-Biostrings::AAStringSet(humanSeq_all)

humanComp_all<-colSums(alphabetFrequency(humanSeq_all))[1:20]/sum(colSums(alphabetFrequency(humanSeq_all))[1:20])

######################Set the rainbow color palette for human #####################
colorPallete_HumanComp <- rainbow(20)
names(colorPallete_HumanComp) <- levels(reorder(names(humanComp_all),-humanComp_all))
########################################################################################

plotHumanComp <- ggplot()+geom_bar(aes(x=reorder(names(humanComp_all),-humanComp_all), fill=reorder(names(humanComp_all),-humanComp_all), y=humanComp_all),stat="identity") + scale_fill_manual(values=colorPallete_HumanComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank(),axis.ticks.y = element_blank(),axis.text.y = element_blank()) + scale_y_continuous(breaks = seq(0,0.1,0.025),limits = c(0,0.1)) + ggtitle("Proteome Composition")

##Iupred##


humanSeqIupred_diso <- Biostrings::AAStringSet(humanDiso_full$iupl_disoSeq)
humanCompIupred<-colSums(alphabetFrequency(humanSeqIupred_diso))[1:20]/sum(colSums(alphabetFrequency(humanSeqIupred_diso))[1:20])


humanIupredDiffComp <- sort((humanCompIupred-humanComp_all)/humanComp_all,decreasing = T)


plotHumanIupred_DC <- ggplot()+geom_bar(aes(x=reorder(names(humanIupredDiffComp),-humanIupredDiffComp), fill=reorder(names(humanIupredDiffComp),-humanIupredDiffComp), y=humanIupredDiffComp),stat="identity") + scale_fill_manual(values=colorPallete_HumanComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank()) + scale_y_continuous(breaks = c(-0.8,-0.4,0,0.4,0.8),limits = c(-0.9,0.9)) + ggtitle("IUpred")


##SPOT##

humanSeqSpot_diso <- Biostrings::AAStringSet(humanDiso_full$spot_disoSeq[!is.na(humanDiso_full$spot_disoSeq)])
humanCompSpot<-colSums(alphabetFrequency(humanSeqSpot_diso))[1:20]/sum(colSums(alphabetFrequency(humanSeqSpot_diso))[1:20])


humanSpotDiffComp <- sort((humanCompSpot-humanComp_all)/humanComp_all,decreasing = T)


plotHumanSpot_DC <- ggplot()+geom_bar(aes(x=reorder(names(humanSpotDiffComp),-humanSpotDiffComp), fill=reorder(names(humanSpotDiffComp),-humanSpotDiffComp), y=humanSpotDiffComp),stat="identity") + scale_fill_manual(values=colorPallete_HumanComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank()) + scale_y_continuous(breaks = c(-0.8,-0.4,0,0.4,0.8),limits = c(-0.9,0.9)) + ggtitle("SPOT")


##VSL##

humanSeqVsl_diso <- Biostrings::AAStringSet(humanDiso_full$vsl_disoSeq[!is.na(humanDiso_full$vsl_disoSeq)])
humanCompVsl<-colSums(alphabetFrequency(humanSeqVsl_diso))[1:20]/sum(colSums(alphabetFrequency(humanSeqVsl_diso))[1:20])


humanVslDiffComp <- sort((humanCompVsl-humanComp_all)/humanComp_all,decreasing = T)


plotHumanVsl_DC <- ggplot()+geom_bar(aes(x=reorder(names(humanVslDiffComp),-humanVslDiffComp), fill=reorder(names(humanVslDiffComp),-humanVslDiffComp), y=humanVslDiffComp),stat="identity") + scale_fill_manual(values=colorPallete_HumanComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank()) + scale_y_continuous(breaks = c(-0.8,-0.4,0,0.4,0.8),limits = c(-0.9,0.9)) + ggtitle("VSL2B")



fig<-ggarrange(plotHumanComp,plotHumanIupred_DC,plotHumanSpot_DC,plotHumanVsl_DC,ncol = 1)
annotate_figure(fig,left = text_grob("Comp. Disorder - Comp. Universe / Comp. Universe", rot = 90,size = 16))
```

#### Xenopus

```{r xenopus composition ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=10}
##XENOPUS

xenopusSeq_all<-xenopus_data$Sequence
names(xenopusSeq_all)<-xenopus_data$acc
xenopusSeq_all<-Biostrings::AAStringSet(xenopusSeq_all)

xenopusComp_all<-colSums(alphabetFrequency(xenopusSeq_all))[1:20]/sum(colSums(alphabetFrequency(xenopusSeq_all))[1:20])

######################Set the rainbow color palette for xenopus #####################
colorPallete_XenopusComp <- rainbow(20)
names(colorPallete_XenopusComp) <- levels(reorder(names(xenopusComp_all),-xenopusComp_all))
########################################################################################

plotXenopusComp <- ggplot()+geom_bar(aes(x=reorder(names(xenopusComp_all),-xenopusComp_all), fill=reorder(names(xenopusComp_all),-xenopusComp_all), y=xenopusComp_all),stat="identity") + scale_fill_manual(values=colorPallete_XenopusComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank(),axis.ticks.y = element_blank(),axis.text.y = element_blank()) + scale_y_continuous(breaks = seq(0,0.1,0.025),limits = c(0,0.1)) + ggtitle("Proteome Composition")

##Iupred##


xenopusSeqIupred_diso <- Biostrings::AAStringSet(xenopus_data$iupl_disoSeq)
xenopusCompIupred<-colSums(alphabetFrequency(xenopusSeqIupred_diso))[1:20]/sum(colSums(alphabetFrequency(xenopusSeqIupred_diso))[1:20])


xenopusIupredDiffComp <- sort((xenopusCompIupred-xenopusComp_all)/xenopusComp_all,decreasing = T)


plotXenopusIupred_DC <- ggplot()+geom_bar(aes(x=reorder(names(xenopusIupredDiffComp),-xenopusIupredDiffComp), fill=reorder(names(xenopusIupredDiffComp),-xenopusIupredDiffComp), y=xenopusIupredDiffComp),stat="identity") + scale_fill_manual(values=colorPallete_XenopusComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank()) + scale_y_continuous(breaks = c(-0.8,-0.4,0,0.4,0.8),limits = c(-0.9,0.9)) + ggtitle("IUpred")


##SPOT##

xenopusSeqSpot_diso <- Biostrings::AAStringSet(xenopus_data$spot_disoSeq[!is.na(xenopus_data$spot_disoSeq)])
xenopusCompSpot<-colSums(alphabetFrequency(xenopusSeqSpot_diso))[1:20]/sum(colSums(alphabetFrequency(xenopusSeqSpot_diso))[1:20])


xenopusSpotDiffComp <- sort((xenopusCompSpot-xenopusComp_all)/xenopusComp_all,decreasing = T)


plotXenopusSpot_DC <- ggplot()+geom_bar(aes(x=reorder(names(xenopusSpotDiffComp),-xenopusSpotDiffComp), fill=reorder(names(xenopusSpotDiffComp),-xenopusSpotDiffComp), y=xenopusSpotDiffComp),stat="identity") + scale_fill_manual(values=colorPallete_XenopusComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank()) + scale_y_continuous(breaks = c(-0.8,-0.4,0,0.4,0.8),limits = c(-0.9,0.9)) + ggtitle("SPOT")


##VSL##

xenopusSeqVsl_diso <- Biostrings::AAStringSet(xenopus_data$vsl_disoSeq[!is.na(xenopus_data$vsl_disoSeq)])
xenopusCompVsl<-colSums(alphabetFrequency(xenopusSeqVsl_diso))[1:20]/sum(colSums(alphabetFrequency(xenopusSeqVsl_diso))[1:20])


xenopusVslDiffComp <- sort((xenopusCompVsl-xenopusComp_all)/xenopusComp_all,decreasing = T)


plotXenopusVsl_DC <- ggplot()+geom_bar(aes(x=reorder(names(xenopusVslDiffComp),-xenopusVslDiffComp), fill=reorder(names(xenopusVslDiffComp),-xenopusVslDiffComp), y=xenopusVslDiffComp),stat="identity") + scale_fill_manual(values=colorPallete_XenopusComp)+ theme_linedraw(base_size = 15) + theme(legend.position="none",axis.title.x =  element_blank(),axis.title.y =  element_blank(),panel.grid.major.x = element_blank(),panel.grid.minor.y = element_blank()) + scale_y_continuous(breaks = c(-0.8,-0.4,0,0.4,0.8),limits = c(-0.9,0.9)) + ggtitle("VSL2B")



fig<-ggarrange(plotXenopusComp,plotXenopusIupred_DC,plotXenopusSpot_DC,plotXenopusVsl_DC,ncol = 1)
annotate_figure(fig,left = text_grob("Comp. Disorder - Comp. Universe / Comp. Universe", rot = 90,size = 16))
```

### d

```{r Lollipop plot for all human kinases ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=10}
# Lollipop plots for other human kinases targets

subset(predictorsFisher_data,grepl(pattern = "MAPK|AURK|PLK|NEK|DYRK",organism)) %>%  
    mutate(organism=factor(organism,c("Human DYRK targets","Human NEK targets","Human PLK targets","Human AURK targets","Human MAPK targets"))) %>% 
    ggplot(aes(x=organism, y=OR)) +
    geom_segment( aes(xend=organism, yend=0)) +
    geom_point(aes(color=organism),size=12) +
    theme_linedraw(base_size = 22)+
    theme(panel.grid.major.y = element_blank())+
    scale_y_continuous(limits = c(0,5),minor_breaks = c(0:5),breaks = seq(0,5,1),expand = c(0,0)) +
    coord_flip() +
    facet_wrap(~predictors,ncol=3)+
    scale_color_manual(values = pal_tron()(7)[c(6,5,2,1,4,3)],guide=F)+
    xlab("")
```

### e

Calculate the porcentage of disorder of yeast and human CDK targets vs the rest of the phosphoproteome with all available predictors

```{r human disorder percentage for all predictors ,echo=FALSE,message=FALSE,warning=FALSE,,fig.height=5,fig.width=15}


ggplot(yeastIntersect_reshaped_disoPerc,aes(x=Predictor, y=`Percentage of disorder`)) +
    geom_violin(aes(fill=target),position=position_dodge(0.9),bw=7,alpha=1,trim = F) +
    geom_boxplot(aes(fill=target),position=position_dodge(0.9),width=0.10,outlier.size = 0.0001) +
ggpubr::theme_classic2() +
theme(
  text = element_text(size=20),
  legend.position = "none",legend.title = element_blank(),
  axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed"),axis.text.x =element_text(angle=45, hjust = 1)
  ) +
scale_y_continuous(limits = c(-20, 135), breaks = c(seq(0, 100, by = 20))) +
scale_x_discrete(labels = c("Pfilt", "SEG","DisEMBL-465","DisEMBL-HotLoops","ESpritz-DisProt","ESpritz-NMR","ESpritz-Xray","GlobPlot","IUPred-Long","IUPred-Short","JRONN","VSL2b","SPOT")) + xlab(NULL) +
scale_fill_manual(values = pal_jco()(10)[c(5,2)]) +
scale_color_manual(values = pal_jco()(10)[c(10,7)],guide=FALSE) + ggtitle("Yeast")

human_disoPerc_cols_idx <-grep("_perc",colnames(human_data))
human_reshaped_disoPerc <- melt(human_data[,human_disoPerc_cols_idx])
colnames(human_reshaped_disoPerc) <- c("Predictor","Percentage of disorder")
human_reshaped_disoPerc$target <- rep(human_data$target,length(human_disoPerc_cols_idx))
human_reshaped_disoPerc$target<- relevel(as.factor(human_reshaped_disoPerc$target),ref="Non Cdk1 target")
human_reshaped_disoPerc$Predictor <- factor(as.character(human_reshaped_disoPerc$Predictor),levels = c("pfilt_perc","seg_perc","dis465_perc","disHL_perc","espD_perc","espN_perc","espX_perc","glo_perc","iupl_perc","iups_perc","jronn_perc","vsl_perc","spot_perc"))



ggplot(human_reshaped_disoPerc,aes(x=Predictor, y=`Percentage of disorder`)) +
    geom_violin(aes(fill=target),position=position_dodge(0.9),bw=5,alpha=1,trim = F) +
    geom_boxplot(aes(fill=target),position=position_dodge(0.9),width=0.10,outlier.size = 0.0001) +
ggpubr::theme_classic2() +
theme(
  text = element_text(size=20),
  legend.position = "none",legend.title = element_blank(),
  axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed"),axis.text.x =element_text(angle=45, hjust = 1)
  ) +
scale_y_continuous(limits = c(-20, 135), breaks = c(seq(0, 100, by = 20))) +
scale_x_discrete(labels = c("Pfilt", "SEG","DisEMBL-465","DisEMBL-HotLoops","ESpritz-DisProt","ESpritz-NMR","ESpritz-Xray","GlobPlot","IUPred-Long","IUPred-Short","JRONN","VSL2b","SPOT")) + xlab(NULL) +
scale_fill_manual(values = pal_jco()(10)[c(5,2)]) +
scale_color_manual(values = pal_jco()(10)[c(10,7)],guide=FALSE) + ggtitle("Human")
```

## Supp. Fig 6

```{r PSAP,echo=FALSE,message=FALSE,warning=FALSE,fig.height=4,fig.width=9}

psap_human <- read_delim("PSAP_data/vanMierlo_prediction_data/prediction_RF_human_first_0to100_gero.tab", 
                         "\t", escape_double = FALSE, trim_ws = TRUE)

psap_xenopus <- read_delim("PSAP_data/vanMierlo_prediction_data/prediction_RF_xenopus_first_0to100_gero.tab", 
                         "\t", escape_double = FALSE, trim_ws = TRUE)

#Xenopus
xenopus_psap_data <- cbind(psap_xenopus,subset=rep("Proteome",nrow(psap_xenopus)))

phosphoproteomic_xenopus_psap <- psap_xenopus[psap_xenopus$acc %in% xenopus_data$ID,]
xenopus_psap_data <- rbind(xenopus_psap_data,cbind(phosphoproteomic_xenopus_psap,subset=rep("Phospho-\nproteome",nrow(phosphoproteomic_xenopus_psap))))

dynamic_xenopus_psap <- psap_xenopus[psap_xenopus$acc %in% unique(xenopus_ANOVA_data$Proteins),]
xenopus_psap_data <- rbind(xenopus_psap_data,cbind(dynamic_xenopus_psap,subset=rep(" Dynamic",nrow(dynamic_xenopus_psap))))

clusterA_xenopus_psap <- psap_xenopus[psap_xenopus$acc %in% clusterA,]
xenopus_psap_data <- rbind(xenopus_psap_data,cbind(clusterA_xenopus_psap,subset=rep("Cluster A",nrow(clusterA_xenopus_psap))))

clusterB_xenopus_psap <- psap_xenopus[psap_xenopus$acc %in% clusterB,]
xenopus_psap_data <- rbind(xenopus_psap_data,cbind(clusterB_xenopus_psap,subset=rep("Cluster B",nrow(clusterB_xenopus_psap))))

clusterC_xenopus_psap <- psap_xenopus[psap_xenopus$acc %in% clusterC,]
xenopus_psap_data <- rbind(xenopus_psap_data,cbind(clusterC_xenopus_psap,subset=rep("Cluster C",nrow(clusterC_xenopus_psap))))

clusterD_xenopus_psap <- psap_xenopus[psap_xenopus$acc %in% clusterD,]
xenopus_psap_data <- rbind(xenopus_psap_data,cbind(clusterD_xenopus_psap,subset=rep("Cluster D",nrow(clusterD_xenopus_psap))))

xenopus_psap_data$subset <- forcats::as_factor(xenopus_psap_data$subset)

ggplot(xenopus_psap_data) + geom_violin(aes(x=subset,y=average,fill=subset),trim = F,adjust = .7) + 
  geom_boxplot(aes(x=subset,y=average),width = 0.15) +
  ggpubr::theme_classic2() +
  theme(text = element_text(size=20),legend.position = "none",legend.title = element_blank(),axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed")) +
  scale_y_continuous(name = "PSAP score",limits = c(-0.20, 1.2), breaks = c(seq(0, 1, by = 0.2))) +
  scale_x_discrete() + xlab(NULL) +
  scale_fill_manual(values = c("#00adeeff","#ffdd15ff","#f6921eff","#f6921eff","#f6921eff","#f6921eff","#f6921eff"))


#Human
human_psap_data <- cbind(psap_human,subset=rep("Proteome",nrow(psap_human)))

phosphoproteomic_human_psap <- psap_human[psap_human$acc %in% human_data$`ACC#`,]
human_psap_data <- rbind(human_psap_data,cbind(phosphoproteomic_human_psap,subset=rep("Phospho-\nproteome",nrow(phosphoproteomic_human_psap))))

cdk_human_psap <- psap_human[psap_human$acc %in% subset(human_data,target=="Cdk1 target")$`ACC#`,]
human_psap_data <- rbind(human_psap_data,cbind(cdk_human_psap,subset=rep("CDK",nrow(cdk_human_psap))))

mapk_human_psap <- psap_human[psap_human$acc %in% subset(human_data,target_mapk=="mapk target")$`ACC#`,]
human_psap_data <- rbind(human_psap_data,cbind(mapk_human_psap,subset=rep("MAPK",nrow(mapk_human_psap))))

aurk_human_psap <- psap_human[psap_human$acc %in% subset(human_data,target_aurk=="aurk target")$`ACC#`,]
human_psap_data <- rbind(human_psap_data,cbind(aurk_human_psap,subset=rep("AURK",nrow(aurk_human_psap))))

plk_human_psap <- psap_human[psap_human$acc %in% subset(human_data,target_plk=="plk target")$`ACC#`,]
human_psap_data <- rbind(human_psap_data,cbind(plk_human_psap,subset=rep("PLK",nrow(plk_human_psap))))

nek_human_psap <- psap_human[psap_human$acc %in% subset(human_data,target_nek=="nek target")$`ACC#`,]
human_psap_data <- rbind(human_psap_data,cbind(nek_human_psap,subset=rep("NEK",nrow(nek_human_psap))))

dyrk_human_psap <- psap_human[psap_human$acc %in% subset(human_data,target_dyrk=="dyrk target")$`ACC#`,]
human_psap_data <- rbind(human_psap_data,cbind(dyrk_human_psap,subset=rep("DYRK",nrow(dyrk_human_psap))))

human_psap_data$subset <- forcats::as_factor(human_psap_data$subset)

ggplot(human_psap_data) + geom_violin(aes(x=subset,y=average,fill=subset),trim = F,adjust = .7) + 
  geom_boxplot(aes(x=subset,y=average),width = 0.15) +
  ggpubr::theme_classic2() +
  theme(text = element_text(size=20),legend.position = "none",legend.title = element_blank(),axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed")) +
  scale_y_continuous(name = "PSAP score",limits = c(-0.20, 1.2), breaks = c(seq(0, 1, by = 0.2))) +
  scale_x_discrete() + xlab(NULL) +
  scale_fill_manual(values = c("#00adeeff","#ffdd15ff",pal_tron()(7)[c(3,4,1,2,5,6)]))
```

## Supp. Fig. 7

### Deleted LR domain

```{r optodroplet quantification - dLR ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=6}

dLR_quant <- read_delim("Optodroplet_data/AlGhoul_data/SuppFig7_dLR.tab", 
                                delim = "\t", escape_double = FALSE, 
                                trim_ws = TRUE)
kruskal.test(Foci_Nucleus ~ Treat_collapsed, data = dLR_quant)

# Kruskal-Wallis rank sum test
# 
# data:  Foci_Nucleus by Treat_collapsed
# Kruskal-Wallis chi-squared = 204.5, df = 5, p-value < 2.2e-16

pairwise.wilcox.test(dLR_quant$Foci_Nucleus, dLR_quant$Treat_collapsed, p.adjust.method = "BH")


# Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
# 
# data:  dLR_quant$Foci_Nucleus and dLR_quant$Treat_collapsed 
# 
#           DMSO L+ DMSO L- OA L+   OA L-   PA L+ 
#   DMSO L- < 2e-16 -       -       -       -     
#   OA L+   0.8143  < 2e-16 -       -       -     
#   OA L-   1.5e-10 2.3e-05 2.9e-09 -       -     
#   PA L+   < 2e-16 0.3916  8.9e-16 0.0014  -     
#   PA L-   < 2e-16 0.7618  < 2e-16 2.9e-05 0.5839
# 
# P value adjustment method: BH 


ggplot(dLR_quant,aes(x=Treatment, y=Foci_Nucleus)) +
  geom_violin(aes(fill=Light,color=Light),position=position_dodge(1),bw=2,alpha=1,trim = T,width=1) +
  geom_boxplot(aes(fill=Light),position=position_dodge(1),width=0.15,outlier.shape = NA,show.legend = F) +
  ggpubr::theme_classic2() +
  theme(text = element_text(size=20),legend.position = "bottom",legend.title = element_blank(),axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed")) +
  scale_y_continuous(limits = c(-1, 80), breaks = c(seq(0, 60, by = 20))) +
  xlab(NULL) +
  geom_segment(aes(x = 0.77, y = 61, xend = 1.22, yend = 61)) + annotate(geom="text", x=1, y=62, label="***") +
  geom_segment(aes(x = 1.77, y = 61, xend = 2.22, yend = 61)) + annotate(geom="text", x=2, y=63, label="n.s.") +
  geom_segment(aes(x = 2.77, y = 61, xend = 3.22, yend = 61)) + annotate(geom="text", x=3, y=62, label="**") +
  geom_segment(aes(x = 0.77, y = 65, xend = 1.77, yend = 65)) + annotate(geom="text", x=1.27, y=66, label="***") +
  geom_segment(aes(x = 1.22, y = 69, xend = 2.22, yend = 69)) + annotate(geom="text", x=1.72, y=71, label="n.s.") +
  geom_segment(aes(x = 1.22, y = 73, xend = 3.22, yend = 73)) + annotate(geom="text", x=2.22, y=74, label="***") +
  scale_fill_manual(values = pal_jco()(10)[c(5,2)]) +
  scale_color_manual(values = pal_jco()(10)[c(5,2)],guide="none")
```


### Deleted N-terminus (Repeats + LR)


```{r optodroplet quantification - Deleted N-terminus (Repeats + LR) ,echo=FALSE,message=FALSE,warning=FALSE,fig.height=5,fig.width=6}

Repeats_LR_quant <- read_delim("Optodroplet_data/AlGhoul_data/SuppFig7_Repeats-LR.tab", 
                                delim = "\t", escape_double = FALSE, 
                                trim_ws = TRUE)
kruskal.test(Foci_Nucleus ~ Treat_collapsed, data = Repeats_LR_quant)

# Kruskal-Wallis rank sum test
# 
# data:  Foci_Nucleus by Treat_collapsed
# Kruskal-Wallis chi-squared = 204.5, df = 5, p-value < 2.2e-16

pairwise.wilcox.test(Repeats_LR_quant$Foci_Nucleus, Repeats_LR_quant$Treat_collapsed, p.adjust.method = "BH")


# Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
# 
# data:  dLR_quant$Foci_Nucleus and dLR_quant$Treat_collapsed 
# 
#           DMSO L+ DMSO L- OA L+   OA L-   PA L+ 
#   DMSO L- < 2e-16 -       -       -       -     
#   OA L+   0.8143  < 2e-16 -       -       -     
#   OA L-   1.5e-10 2.3e-05 2.9e-09 -       -     
#   PA L+   < 2e-16 0.3916  8.9e-16 0.0014  -     
#   PA L-   < 2e-16 0.7618  < 2e-16 2.9e-05 0.5839
# 
# P value adjustment method: BH 


ggplot(Repeats_LR_quant,aes(x=Treatment, y=Foci_Nucleus)) +
  geom_violin(aes(fill=Light,color=Light),position=position_dodge(1),bw=2,alpha=1,trim = T,width=1) +
  geom_boxplot(aes(fill=Light),position=position_dodge(1),width=0.15,outlier.shape = NA,show.legend = F) +
  ggpubr::theme_classic2() +
  theme(text = element_text(size=20),legend.position = "bottom",legend.title = element_blank(),axis.ticks.x = element_blank(),panel.grid.major.y = element_line(colour = "grey",linetype = "dashed")) +
  scale_y_continuous(limits = c(-1, 80), breaks = c(seq(0, 60, by = 20))) +
  xlab(NULL) +
  geom_segment(aes(x = 0.77, y = 61, xend = 1.22, yend = 61)) + annotate(geom="text", x=1, y=62, label="***") +
  geom_segment(aes(x = 1.77, y = 61, xend = 2.22, yend = 61)) + annotate(geom="text", x=2, y=63, label="n.s.") +
  geom_segment(aes(x = 2.77, y = 61, xend = 3.22, yend = 61)) + annotate(geom="text", x=3, y=62, label="**") +
  geom_segment(aes(x = 0.77, y = 65, xend = 1.77, yend = 65)) + annotate(geom="text", x=1.27, y=66, label="***") +
  geom_segment(aes(x = 1.22, y = 69, xend = 2.22, yend = 69)) + annotate(geom="text", x=1.72, y=71, label="n.s.") +
  geom_segment(aes(x = 1.22, y = 73, xend = 3.22, yend = 73)) + annotate(geom="text", x=2.22, y=74, label="***") +
  scale_fill_manual(values = pal_jco()(10)[c(5,2)]) +
  scale_color_manual(values = pal_jco()(10)[c(5,2)],guide="none")
```
